<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Emma Rand">
<title>Workshop – Data Analysis in R</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark-2fef5ea3f8957b3e4ecc936fc74692ca.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-a03e91e75e70fe074cb6f9e1686248b2.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark-efba4cb111cfd8b4774c90a6412e7cf1.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../../site_libs/bootstrap/bootstrap-a03e91e75e70fe074cb6f9e1686248b2.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-50FV9HPTRE"></script><script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-50FV9HPTRE', { 'anonymize_ip': true});
</script><script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script><script src="../../site_libs/kePrint-0.0.1/kePrint.js"></script><link href="../../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">
</head>
<body class="nav-sidebar floating nav-fixed quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Data Analysis in R</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Welcome!</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../r4babs1/r4babs1.html"> 
<span class="menu-text">BABS 1</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../r4babs2/r4babs2.html"> 
<span class="menu-text">BABS 2</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../r4babs4/r4babs4.html"> 
<span class="menu-text">BABS 4</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../pgt52m/pgt52m.html" aria-current="page"> 
<span class="menu-text">PGT 52M</span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
    <a href="https://github.com/3mmaRand" title="Emma on GitHub" class="quarto-navigation-tool px-1" aria-label="Emma on GitHub"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../pgt52m/week-7/overview.html">Week 7: Two-sample tests</a></li><li class="breadcrumb-item"><a href="../../pgt52m/week-7/workshop.html">Workshop</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/pgt52m.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">52M Data Analysis in R</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 1: Understanding file systems</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-1/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-1/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-1/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-1/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 2: Introduction to R and project organisation</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-2/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-2/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-2/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-2/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 3: Types of variable, summarising and plotting data</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-3/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-3/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-3/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-3/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 4: Summarising data with several variables</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-4/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-4/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-4/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-4/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 5: The logic of hypothesis testing and CI</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-5/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-5/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-5/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-5/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 6: Introduction to statistical models: Single regression</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-6/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-6/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-6/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-6/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 7: Two-sample tests</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-7" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-7" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-7/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-7/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-7/workshop.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-7/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 8: One-way ANOVA and Kruskal-Wallis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-8" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-8" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-8/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-8/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-8/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-8/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 9: Introduction to the Assessment</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-9" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-9" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-9/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-9/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-9/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pgt52m/week-9/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 10: Reproducible Reporting</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-10" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-10" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
 <span class="menu-text">About</span>
  </li>
          <li class="sidebar-item">
 <span class="menu-text">Prepare!</span>
  </li>
          <li class="sidebar-item">
 <span class="menu-text">Workshop</span>
  </li>
          <li class="sidebar-item">
 <span class="menu-text">Consolidate!</span>
  </li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li>
<a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul class="collapse">
<li><a href="#session-overview" id="toc-session-overview" class="nav-link" data-scroll-target="#session-overview">Session overview</a></li>
  <li><a href="#philosophy" id="toc-philosophy" class="nav-link" data-scroll-target="#philosophy">Philosophy</a></li>
  </ul>
</li>
  <li><a href="#getting-started" id="toc-getting-started" class="nav-link" data-scroll-target="#getting-started">Getting started</a></li>
  <li>
<a href="#exercises" id="toc-exercises" class="nav-link" data-scroll-target="#exercises">Exercises</a>
  <ul class="collapse">
<li>
<a href="#adiponectin-secretion" id="toc-adiponectin-secretion" class="nav-link" data-scroll-target="#adiponectin-secretion">Adiponectin secretion</a>
  <ul class="collapse">
<li><a href="#summarising-the-data" id="toc-summarising-the-data" class="nav-link" data-scroll-target="#summarising-the-data">Summarising the data</a></li>
  <li><a href="#selecting-a-test" id="toc-selecting-a-test" class="nav-link" data-scroll-target="#selecting-a-test">Selecting a test</a></li>
  <li><a href="#applying-interpreting-and-reporting" id="toc-applying-interpreting-and-reporting" class="nav-link" data-scroll-target="#applying-interpreting-and-reporting">Applying, interpreting and reporting</a></li>
  <li><a href="#check-assumptions" id="toc-check-assumptions" class="nav-link" data-scroll-target="#check-assumptions">Check assumptions</a></li>
  <li><a href="#illustrating" id="toc-illustrating" class="nav-link" data-scroll-target="#illustrating">Illustrating</a></li>
  </ul>
</li>
  <li>
<a href="#grouse-parasites" id="toc-grouse-parasites" class="nav-link" data-scroll-target="#grouse-parasites">Grouse Parasites</a>
  <ul class="collapse">
<li><a href="#selecting" id="toc-selecting" class="nav-link" data-scroll-target="#selecting">Selecting</a></li>
  <li><a href="#applying-interpreting-and-reporting-1" id="toc-applying-interpreting-and-reporting-1" class="nav-link" data-scroll-target="#applying-interpreting-and-reporting-1">Applying, interpreting and reporting</a></li>
  <li><a href="#illustrating-1" id="toc-illustrating-1" class="nav-link" data-scroll-target="#illustrating-1">Illustrating</a></li>
  </ul>
</li>
  <li>
<a href="#gene-expression" id="toc-gene-expression" class="nav-link" data-scroll-target="#gene-expression">Gene Expression</a>
  <ul class="collapse">
<li><a href="#selecting-1" id="toc-selecting-1" class="nav-link" data-scroll-target="#selecting-1">Selecting</a></li>
  <li><a href="#applying-interpreting-and-reporting-2" id="toc-applying-interpreting-and-reporting-2" class="nav-link" data-scroll-target="#applying-interpreting-and-reporting-2">Applying, interpreting and reporting</a></li>
  </ul>
</li>
  <li><a href="#look-after-future-you" id="toc-look-after-future-you" class="nav-link" data-scroll-target="#look-after-future-you">Look after future you!</a></li>
  </ul>
</li>
  <li><a href="#well-done" id="toc-well-done" class="nav-link" data-scroll-target="#well-done">🥳 Well Done! 🎉</a></li>
  <li><a href="#independent-study-following-the-workshop" id="toc-independent-study-following-the-workshop" class="nav-link" data-scroll-target="#independent-study-following-the-workshop">Independent study following the workshop</a></li>
  <li><a href="#the-code-file" id="toc-the-code-file" class="nav-link" data-scroll-target="#the-code-file">The Code file</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/3mmaRand/comp4biosci/edit/main/pgt52m/week-7/workshop.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li></ul></div><div class="quarto-other-links"><h2>Other Links</h2><ul><li><a href="#"><i class="bi bi-chat-left-dots"></i>Make a suggestion</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../pgt52m/week-7/overview.html">Week 7: Two-sample tests</a></li><li class="breadcrumb-item"><a href="../../pgt52m/week-7/workshop.html">Workshop</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Workshop</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">Two-sample tests</p>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><section id="introduction" class="level1"><h1>Introduction</h1>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="images/how-much-i-know.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Artwork by @allison_horst: “How much I think I know about R”"><img src="images/how-much-i-know.png" class="img-fluid figure-img" alt="Illustrated line plot of 'How much I think I know about R' on the y-axis, and 'Time' on the x-axis. Along the line are emoji-style faces, showing the non-linear progression of R knowledge over time. At first, a nervous face becomes a happy face early on in learning, then a grimace face at an intermediate peak before a steep decline (with an exhausted face at the local minimum). Then, a determined face charges back up a hill, reaching another peak with a mind-blown face and text annotation 'join R community on twitter' followed by another decline, but this time the faces look happy even though their 'How much I think I know about R' value is declining." width="800"></a></p>
<figcaption>Artwork by <span class="citation" data-cites="allison_horst">Horst (<a href="#ref-allison_horst" role="doc-biblioref">2023</a>)</span>: “How much I think I know about R”</figcaption></figure>
</div>
<section id="session-overview" class="level2"><h2 class="anchored" data-anchor-id="session-overview">Session overview</h2>
<p>In this workshop you will get practice in choosing between, performing, and presenting the results of, two-sample tests and their non-parametric equivalents in R.</p>
</section><section id="philosophy" class="level2"><h2 class="anchored" data-anchor-id="philosophy">Philosophy</h2>
<p>Workshops are not a test. It is expected that you often don’t know how to start, make a lot of mistakes and need help. It is expected that you are familiar with independent study content before the workshop. However, you need not remember or understand every detail as the workshop should build and consolidate your understanding. Tips</p>
<ul>
<li>don’t worry about making mistakes</li>
<li>don’t let what you can not do interfere with what you can do</li>
<li>discussing code with your neighbours will help</li>
<li>look things up in the independent study material</li>
<li>look things up in your own code from earlier</li>
<li>there are no stupid questions</li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Key
</div>
</div>
<div class="callout-body-container callout-body">
<p>These four symbols are used at the beginning of each instruction so you know where to carry out the instruction.</p>
<p><img src="images/do_on_your_computer.png" class="img-fluid"> Something you need to do on your computer. It may be opening programs or documents or locating a file.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Something you should do in RStudio. It will often be typing a command or using the menus but might also be creating folders, locating or moving files.</p>
<p><img src="images/do_on_internet.png" class="img-fluid"> Something you should do in your browser on the internet. It may be searching for information, going to the VLE or downloading a file.</p>
<p><img src="images/answer.png" class="img-fluid"> A question for you to think about and answer. Record your answers in your script for future reference.</p>
</div>
</div>
</section></section><section id="getting-started" class="level1"><h1>Getting started</h1>
<p><img src="images/do_on_your_computer.png" class="img-fluid"> Start RStudio from the Start menu.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Go the Files tab in the lower right pane and click on the <code>...</code> on the right. This will open a “Go to folder” window. Navigate to a place on your computer where you keep your work. Click Open.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Make an RStudio project for this workshop by clicking on the drop-down menu on top right where it says <code>Project: (None)</code> and choosing New Project, then New Directory, then New Project. Navigate to the <code>data-analysis-in-r-1</code> folder and name the RStudio Project <code>week-7</code>.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Make new folders called <code>data-raw</code> and <code>figures</code>. You can do this on the Files Pane by clicking New Folder and typing into the box that appears.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Make a new script then save it with a name like <code>two-sample-tests.R</code> to carry out the rest of the work.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Add a comment to the script: <code># Two-sample tests</code> and load the <strong><code>tidyverse</code></strong> <span class="citation" data-cites="tidyverse">(<a href="#ref-tidyverse" role="doc-biblioref">Wickham et al. 2019</a>)</span> package</p>
</section><section id="exercises" class="level1"><h1>Exercises</h1>
<section id="adiponectin-secretion" class="level2"><h2 class="anchored" data-anchor-id="adiponectin-secretion">Adiponectin secretion</h2>
<p>Adiponectin is exclusively secreted from adipose tissue and modulates a number of metabolic processes. Nicotinic acid can affect adiponectin secretion. 3T3-L1 adipocytes were treated with nicotinic acid or with a control treatment and adiponectin concentration (pg/mL) measured. The data are in <a href="data-raw/adipocytes.txt">adipocytes.txt</a>. Each row represents an independent sample of adipocytes and the first column gives the concentration adiponectin and the second column indicates whether they were treated with nicotinic acid or not.</p>
<p><img src="images/do_on_your_computer.png" class="img-fluid"> Save a copy of <a href="data-raw/adipocytes.txt">adipocytes.txt</a> to <code>data-raw</code></p>
<p><img src="images/do_in_R.png" class="img-fluid"> Read in the data and check the structure. I used the name <code>adip</code> for the dataframe/tibble.</p>
<p>We have a tibble containing two variables: <code>adiponectin</code> is the response and is continuous and <code>treatment</code> is explanatory. <code>treatment</code> is categorical with two levels (groups). The first task is visualise the data to get an overview. For continuous response variables with categorical explanatory variables you could use <code>geom_point()</code>, <code>geom_boxplot()</code> or a variety of other geoms. I often use <code>geom_violin()</code> which allows us to see the distribution - the violin is fatter where there are more data points.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Do a quick plot of the data:</p>
<div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">adip</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">treatment</span>, y <span class="op">=</span> <span class="va">adiponectin</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-3-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2"><img src="workshop_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<section id="summarising-the-data" class="level3"><h3 class="anchored" data-anchor-id="summarising-the-data">Summarising the data</h3>
<p>Summarising the data for each treatment group is the next sensible step. The most useful summary statistics are the means, standard deviations, sample sizes and standard errors.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Create a data frame called <code>adip_summary</code> that contains the means, standard deviations, sample sizes and standard errors for the control and nicotinic acid treated samples. You may need to the <a href="../week-4/workshop.html#summarise">Summarise</a> from the Week 4 workshop</p>
<p>You should get the following numbers:</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead><tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">treatment</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">mean</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">std</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">n</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">se</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">control</td>
<td style="text-align: right;">5.546000</td>
<td style="text-align: right;">1.475247</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">0.3809072</td>
</tr>
<tr class="even">
<td style="text-align: left;">nicotinic</td>
<td style="text-align: right;">7.508667</td>
<td style="text-align: right;">1.793898</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">0.4631824</td>
</tr>
</tbody>
</table>
</div>
</div>
</section><section id="selecting-a-test" class="level3"><h3 class="anchored" data-anchor-id="selecting-a-test">Selecting a test</h3>
<p><img src="images/answer.png" class="img-fluid"> Do you think this is a paired-sample test or two-sample test?</p>
<!-- #---THINKING ANSWER---  -->
<!-- Two-sample - the samples are either control or nicotinic acid treated, not both.  -->
<!-- There is no link between the first control value and the first nicotinic acid  -->
<!-- value. The two samples are independent and it is a two-sample -->
</section><section id="applying-interpreting-and-reporting" class="level3"><h3 class="anchored" data-anchor-id="applying-interpreting-and-reporting">Applying, interpreting and reporting</h3>
<p><img src="images/do_in_R.png" class="img-fluid"> Create a two-sample model like this:</p>
<div class="cell">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">adip</span>,</span>
<span>          <span class="va">adiponectin</span> <span class="op">~</span> <span class="va">treatment</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><img src="images/do_in_R.png" class="img-fluid"> Examine the model with:</p>
<div class="cell">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = adiponectin ~ treatment, data = adip)

Residuals:
    Min      1Q  Median      3Q     Max 
-4.3787 -1.0967  0.1927  1.0245  3.1113 

Coefficients:
                   Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)          5.5460     0.4240  13.079  1.9e-13 ***
treatmentnicotinic   1.9627     0.5997   3.273  0.00283 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.642 on 28 degrees of freedom
Multiple R-squared:  0.2767,    Adjusted R-squared:  0.2509 
F-statistic: 10.71 on 1 and 28 DF,  p-value: 0.00283</code></pre>
</div>
</div>
<p><img src="images/answer.png" class="img-fluid"> What do you conclude from the test? Write your conclusion in a form suitable for a report.</p>
<!-- #---THINKING ANSWER---  -->
<!-- Adipoctyes treated with nicotinic acid (mean +/- s.e.: 7.51 +/- 1.10 pg/mL)  -->
<!-- secrete significantly more adiponectin than controls (5.55  +/- 0.12 pg/mL)  -->
<!-- (t = 3.27; d.f. = 28; p = 0.003). -->
</section><section id="check-assumptions" class="level3"><h3 class="anchored" data-anchor-id="check-assumptions">Check assumptions</h3>
<p>The assumptions of the general linear model are that the residuals – the difference between predicted value (i.e., the group mean) and observed values - are normally distributed and have homogeneous variance. To check these we can examine the <code>mod$residuals</code> variable. You may want to refer to <a href="../week-6/workshop.html#checking-assumptions">Checking assumptions</a> in the “Single regression” workshop.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Plot the model residuals against the fitted values.</p>
<p><img src="images/answer.png" class="img-fluid"> What to you conclude?</p>
<!-- #---THINKING ANSWER--- -->
<!-- The residuals look 'homogeneously' distributed. That is, the variance is the same  -->
<!-- for the three groups  -->
<p>To examine normality of the model residuals we can plot them as a histogram and do a normality test on them.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Plot a histogram of the residuals.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Use the <code><a href="https://rdrr.io/r/stats/shapiro.test.html">shapiro.test()</a></code> to test the normality of the model residuals</p>
<p><img src="images/answer.png" class="img-fluid"> What to you conclude?</p>
<!-- #---THINKING ANSWER--- -->
<!-- the residuals appear roughly normally distributed.  -->
<!-- The histogram is symmetrical and the shapiro test is NS. -->
<!-- Overall, it seems the assumptions are met -->
</section><section id="illustrating" class="level3"><h3 class="anchored" data-anchor-id="illustrating">Illustrating</h3>
<p><img src="images/do_in_R.png" class="img-fluid"> Create a figure like the one below. You may need to refer to <a href="../week-4/workshop.html#visualise">Visualise</a> from the “Summarising data with several variables” workshop <span class="citation" data-cites="rand2023">(<a href="#ref-rand2023" role="doc-biblioref">Rand 2023</a>)</span></p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-11-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="workshop_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<p>We now need to annotate the figure with the results from the statistical test. This most commonly done with a line linking the means being compared and the <em>p</em>-value. The <code>annotate()</code> function can be used to draw the line and then to add the value. The line is a <code>segment</code> and the <em>p</em>-value is a <code>text</code>.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Add annotation to the figure by adding:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">......</span> <span class="op">+</span></span>
<span>  <span class="fu">annotate</span><span class="op">(</span><span class="st">"segment"</span>, x <span class="op">=</span> <span class="fl">1</span>, xend <span class="op">=</span> <span class="fl">2</span>, </span>
<span>           y <span class="op">=</span> <span class="fl">11.3</span>, yend <span class="op">=</span> <span class="fl">11.3</span>,</span>
<span>           colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">annotate</span><span class="op">(</span><span class="st">"text"</span>, x <span class="op">=</span> <span class="fl">1.5</span>,  y <span class="op">=</span> <span class="fl">11.7</span>, </span>
<span>           label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="fu">italic</span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">~</span><span class="st">"= 0.003"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-12-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4"><img src="workshop_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<p>For the segment, <code>annotate()</code> needs the <em>x</em> and <em>y</em> coordinates for the start and the finish of the line.</p>
<p>The use of <code><a href="https://rdrr.io/r/base/expression.html">expression()</a></code> allows you to specify formatting or special characters. <code><a href="https://rdrr.io/r/base/expression.html">expression()</a></code> takes strings or <a href="https://en.wikipedia.org/wiki/LaTeX">LaTeX</a> formatting. Each string or piece of LaTeX is separated by a <code>*</code> or a <code>~</code>. The <code>*</code> concatenates the strings without a space, <code>~</code> does so with a space. It will generate a warning message “In is.na(x) : is.na() applied to non-(list or vector) of type ‘expression’” which can be ignored.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Save your figure to your figures folder.</p>
</section></section><section id="grouse-parasites" class="level2"><h2 class="anchored" data-anchor-id="grouse-parasites">Grouse Parasites</h2>
<p>Grouse livers were dissected and the number of individuals of a parasitic nematode were counted for two estates ‘Gordon’ and ‘Moss’. We want to know if the two estates have different infection rates. The data are in <a href="data-raw/grouse.csv">grouse.csv</a></p>
<p><img src="images/do_on_your_computer.png" class="img-fluid"> Save a copy of <a href="data-raw/grouse.csv">grouse.csv</a> to <code>data-raw</code></p>
<p><img src="images/do_in_R.png" class="img-fluid"> Read in the data and check the structure. I used the name <code>grouse</code> for the dataframe/tibble.</p>
<section id="selecting" class="level3"><h3 class="anchored" data-anchor-id="selecting">Selecting</h3>
<p><img src="images/answer.png" class="img-fluid"> Using your common sense, do these data look normally distributed?</p>
<!-- #---THINKING ANSWER---  -->
<!-- No and little need to test/plot - they're counts, skewed and  -->
<!-- have unequal variances -->
<p><img src="images/answer.png" class="img-fluid"> What test do you suggest?</p>
<!-- #---THINKING ANSWER---  -->
<!-- two-sample Wilcoxon (unpaired) AKA Mann-Whitney -->
</section><section id="applying-interpreting-and-reporting-1" class="level3"><h3 class="anchored" data-anchor-id="applying-interpreting-and-reporting-1">Applying, interpreting and reporting</h3>
<p><img src="images/do_in_R.png" class="img-fluid"> Summarise the data by finding the median of each group:</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Carry out a two-sample Wilcoxon test (also known as a Mann-Whitney):</p>
<div class="cell">
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/wilcox.test.html">wilcox.test</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">grouse</span>, <span class="va">nematodes</span> <span class="op">~</span> <span class="va">estate</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Wilcoxon rank sum exact test

data:  nematodes by estate
W = 78, p-value = 0.03546
alternative hypothesis: true location shift is not equal to 0</code></pre>
</div>
</div>
<p><img src="images/answer.png" class="img-fluid"> What do you conclude from the test? Write your conclusion in a form suitable for a report.</p>
<!-- #---THINKING ANSWER---  -->
<!-- Grouse on Gordon had significantly more nematodes (median = 13.5) than those  -->
<!-- on Moss (median = 5) (W = 78; n1 = 10, n2 = 10; p = 0.03546). -->
</section><section id="illustrating-1" class="level3"><h3 class="anchored" data-anchor-id="illustrating-1">Illustrating</h3>
<p>A box plot is a usually good choice for illustrating a two-sample Wilcoxon test because it shows the median and interquartile range.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> We can create a simple boxplot with:</p>
<div class="cell">
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">grouse</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">estate</span>, y <span class="op">=</span> <span class="va">nematodes</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-18-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5"><img src="workshop_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<p><img src="images/do_in_R.png" class="img-fluid"> Annotate and format the figure so it is more suitable for a report and save it to your figures folder.</p>
</section></section><section id="gene-expression" class="level2"><h2 class="anchored" data-anchor-id="gene-expression">Gene Expression</h2>
<p>Bambara groundnut (<em>Vigna subterranea</em>) is an African legume with good nutritional value which can be influenced by low temperature stress. Researchers are interested in the expression levels of a particular set of 35 genes (<code>probe_id</code>) in response to temperature stress. They measure the expression of the genes at 23 and 18 degrees C (high and low temperature). These samples are <strong>not</strong> independent because we have two measure from one gene. The data are in <a href="data-raw/expr.xlsx">expr.xlxs</a>.</p>
<section id="selecting-1" class="level3"><h3 class="anchored" data-anchor-id="selecting-1">Selecting</h3>
<p><img src="images/answer.png" class="img-fluid"> What is the null hypothesis?</p>
<!-- #---THINKING ANSWER---  -->
<!-- expression is not affected by temperature / for each gene,  -->
<!-- the expression at the higher t   = expression at the lower t -->
<p><img src="images/do_in_R.png" class="img-fluid"> Save a copy of <a href="data-raw/expr.xlsx">expr.xlxs</a> and import the data. I named the dataframe <code>bambara</code></p>
<p><img src="images/answer.png" class="img-fluid"> What is the appropriate parametric test?</p>
<!-- #---THINKING ANSWER--- -->
<!-- the appropriate parametric test is the paired sample lm  -->
</section><section id="applying-interpreting-and-reporting-2" class="level3"><h3 class="anchored" data-anchor-id="applying-interpreting-and-reporting-2">Applying, interpreting and reporting</h3>
<p>A paired test requires us to test whether the difference in expression between high and low temperatures is zero on average. One handy way to achieve this is to organise our groups into two columns. The <code>pivot_wider()</code> function will do this for us. We need to tell it what column gives the identifiers (i.e., matches the the pairs) - the probe_ids in this case. We also need to say which variable contains what will become the column names and which contains the values.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Pivot the data so there is a column for each temperature:</p>
<div class="cell">
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">bambara</span> <span class="op">&lt;-</span> <span class="va">bambara</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">temperature</span>, </span>
<span>              values_from <span class="op">=</span> <span class="va">expression</span>, </span>
<span>              id_cols <span class="op">=</span> <span class="va">probe_id</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><img src="images/do_in_R.png" class="img-fluid"> Click on the <code>bambara</code> dataframe in the environment to open a view of it so that you understand what <code>pivot_wider()</code> has done.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Create a paired-sample model like this:</p>
<div class="cell">
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">bambara</span>, </span>
<span>          <span class="va">highert</span> <span class="op">-</span> <span class="va">lowert</span> <span class="op">~</span> <span class="fl">1</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Since we have done <code>highert - lowert</code>, the “(Intercept) Estimate” will be the average of the higher temperature expression minus the lower temperature expression for each gene.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Examine the model with:</p>
<div class="cell">
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = highert - lowert ~ 1, data = bambara)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.05478 -0.46058  0.09682  0.33342  1.06892 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)   
(Intercept)  0.30728    0.09591   3.204  0.00294 **
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.5674 on 34 degrees of freedom</code></pre>
</div>
</div>
<p><img src="images/answer.png" class="img-fluid"> State your conclusion from the test in a form suitable for including in a report. Make sure you give the direction of any significant effect.</p>
<!-- #---THINKING ANSWER--- -->
<!-- The expression of individual genes is significantly higher at higher temperatures  -->
<!-- than lower temperatures (t = 3.20, d.f. = 34, p = 0.0029) with an average difference -->
<!-- of 0.307 units. -->
</section></section><section id="look-after-future-you" class="level2"><h2 class="anchored" data-anchor-id="look-after-future-you">Look after future you!</h2>
<p>The code required to summarise, test, and plot data for any two-sample test AND for any for any one-way ANOVA is <strong>exactly the same</strong> except for the names of the dataframe, variables and the axis labels and limits. Take some time to comment it your code so that you can make use of it next week.</p>
<p><a href="images/future_you.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6"><img src="images/future_you.png" class="img-fluid" alt="xxx" width="300"></a></p>
<p>You’re finished!</p>
</section></section><section id="well-done" class="level1"><h1>🥳 Well Done! 🎉</h1>
</section><section id="independent-study-following-the-workshop" class="level1"><h1>Independent study following the workshop</h1>
<p><a href="../../pgt52m/week-7/study_after_workshop.html">Consolidate</a></p>
</section><section id="the-code-file" class="level1"><h1>The Code file</h1>
<p>This contains all the code needed in the workshop even where it is not visible on the webpage.</p>
<p>The <code>workshop.qmd</code> file is the file I use to compile the practical. Qmd stands for Quarto markdown. It allows code and ordinary text to be interweaved to produce well-formatted reports including webpages. <a href="https://github.com/3mmaRand/R4BABS/blob/main/pgt52m/week-7/workshop.qmd">View the Qmd in Browser</a>. Coding and thinking answers are marked with <code>#---CODING ANSWER---</code> and <code>#---THINKING ANSWER---</code></p>
<p>Pages made with R <span class="citation" data-cites="R-core">(<a href="#ref-R-core" role="doc-biblioref">R Core Team 2024</a>)</span>, Quarto <span class="citation" data-cites="allaire2022">(<a href="#ref-allaire2022" role="doc-biblioref">Allaire et al. 2022</a>)</span>, <code>knitr</code> <span class="citation" data-cites="knitr">(<a href="#ref-knitr" role="doc-biblioref"><strong>knitr?</strong></a>)</span>, <code>kableExtra</code> <span class="citation" data-cites="kableExtra">(<a href="#ref-kableExtra" role="doc-biblioref">Zhu 2024</a>)</span></p>
</section><section id="references" class="level1">


<!-- -->


</section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-allaire2022" class="csl-entry" role="listitem">
Allaire, J. J., Charles Teague, Carlos Scheidegger, Yihui Xie, and Christophe Dervieux. 2022. <em>Quarto</em>. <a href="https://doi.org/10.5281/zenodo.5960048">https://doi.org/10.5281/zenodo.5960048</a>.
</div>
<div id="ref-allison_horst" class="csl-entry" role="listitem">
Horst, Allison. 2023. <span>“Data Science Illustrations.”</span> <a href="https://allisonhorst.com/allison-horst">https://allisonhorst.com/allison-horst</a>.
</div>
<div id="ref-R-core" class="csl-entry" role="listitem">
R Core Team. 2024. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-rand2023" class="csl-entry" role="listitem">
Rand, Emma. 2023. <em>Data Analysis in r for Becoming a Bioscientist</em>. <a href="https://3mmarand.github.io/R4BABS/">https://3mmarand.github.io/R4BABS/</a>.
</div>
<div id="ref-tidyverse" class="csl-entry" role="listitem">
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. <span>“Welcome to the <span></span>Tidyverse<span></span>”</span> 4: 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>.
</div>
<div id="ref-kableExtra" class="csl-entry" role="listitem">
Zhu, Hao. 2024. <em>kableExtra: Construct Complex Table with ’Kable’ and Pipe Syntax</em>. <a href="https://CRAN.R-project.org/package=kableExtra">https://CRAN.R-project.org/package=kableExtra</a>.
</div>
</div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("^(?:http:|https:)\/\/3mmarand\.github\.io\/R4BABS");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb13" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workshop"</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Two-sample tests"</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="an">toc:</span><span class="co"> true</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="an">toc-location:</span><span class="co"> right</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="fu"># Introduction</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a><span class="al">![Artwork by @allison_horst: "How much I think I know about R"](images/how-much-i-know.png)</span>{fig-alt="Illustrated line plot of 'How much I think I know about R' on the y-axis, and 'Time' on the x-axis. Along the line are emoji-style faces, showing the non-linear progression of R knowledge over time. At first, a nervous face becomes a happy face early on in learning, then a grimace face at an intermediate peak before a steep decline (with an exhausted face at the local minimum). Then, a determined face charges back up a hill, reaching another peak with a mind-blown face and text annotation 'join R community on twitter' followed by another decline, but this time the faces look happy even though their 'How much I think I know about R' value is declining." width="800"}</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a><span class="fu">## Session overview</span></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>In this workshop you will get practice in choosing between, performing, and presenting the results of, two-sample tests and their non-parametric equivalents in R.</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a><span class="fu">## Philosophy</span></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>Workshops are not a test. It is expected that you often don't know how to start, make a lot of mistakes and need help. It is expected that you are familiar with independent study content before the workshop. However, you need not remember or understand every detail as the workshop should build and consolidate your understanding. Tips</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>don't worry about making mistakes</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>don't let what you can not do interfere with what you can do</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>discussing code with your neighbours will help</span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>look things up in the independent study material</span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>look things up in your own code from earlier </span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>there are no stupid questions</span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a><span class="fu">## Key</span></span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>These four symbols are used at the beginning of each instruction so you know where to carry out the instruction.</span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_on_your_computer.png)</span> Something you need to do on your computer. It may be opening programs or documents or locating a file.</span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Something you should do in RStudio. It will often be typing a command or using the menus but might also be creating folders, locating or moving files.</span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_on_internet.png)</span> Something you should do in your browser on the internet. It may be searching for information, going to the VLE or downloading a file.</span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-46"><a href="#cb13-46" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> A question for you to think about and answer. Record your answers in your script for future reference.</span>
<span id="cb13-47"><a href="#cb13-47" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb13-48"><a href="#cb13-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-49"><a href="#cb13-49" aria-hidden="true" tabindex="-1"></a><span class="fu"># Getting started</span></span>
<span id="cb13-50"><a href="#cb13-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-51"><a href="#cb13-51" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_on_your_computer.png)</span> Start RStudio from the Start menu.</span>
<span id="cb13-52"><a href="#cb13-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-53"><a href="#cb13-53" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Go the Files tab in the lower right pane and click on the <span class="in">`...`</span> on the right. This will open a "Go to folder" window. Navigate to a place on your computer where you keep your work. Click Open.</span>
<span id="cb13-54"><a href="#cb13-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-55"><a href="#cb13-55" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Make an RStudio project for this workshop by clicking on the drop-down menu on top right where it says <span class="in">`Project: (None)`</span> and choosing New Project, then New Directory, then New Project. Navigate to the <span class="in">`data-analysis-in-r-1`</span> folder and name the RStudio Project <span class="in">`week-7`</span>.</span>
<span id="cb13-56"><a href="#cb13-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-57"><a href="#cb13-57" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Make new folders called <span class="in">`data-raw`</span> and <span class="in">`figures`</span>. You can do this on the Files Pane by clicking New Folder and typing into the box that appears.</span>
<span id="cb13-58"><a href="#cb13-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-59"><a href="#cb13-59" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Make a new script then save it with a name like <span class="in">`two-sample-tests.R`</span> to carry out the rest of the work.</span>
<span id="cb13-60"><a href="#cb13-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-61"><a href="#cb13-61" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Add a comment to the script: <span class="in">`# Two-sample tests`</span> and load the **`tidyverse`** <span class="co">[</span><span class="ot">@tidyverse</span><span class="co">]</span> package</span>
<span id="cb13-62"><a href="#cb13-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-63"><a href="#cb13-63" aria-hidden="true" tabindex="-1"></a><span class="fu"># Exercises </span></span>
<span id="cb13-64"><a href="#cb13-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-65"><a href="#cb13-65" aria-hidden="true" tabindex="-1"></a><span class="fu">## Adiponectin secretion</span></span>
<span id="cb13-66"><a href="#cb13-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-67"><a href="#cb13-67" aria-hidden="true" tabindex="-1"></a>Adiponectin is exclusively secreted from adipose tissue and modulates a number of metabolic processes. Nicotinic acid can affect adiponectin secretion. 3T3-L1 adipocytes were treated with nicotinic acid or with a control treatment and adiponectin concentration (pg/mL) measured. The data are in <span class="co">[</span><span class="ot">adipocytes.txt</span><span class="co">](data-raw/adipocytes.txt)</span>. Each row represents an independent sample of adipocytes and the first column gives the concentration adiponectin and the second column indicates whether they were treated with nicotinic acid or not. </span>
<span id="cb13-68"><a href="#cb13-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-69"><a href="#cb13-69" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_on_your_computer.png)</span> Save a copy of <span class="co">[</span><span class="ot">adipocytes.txt</span><span class="co">](data-raw/adipocytes.txt)</span> to <span class="in">`data-raw`</span></span>
<span id="cb13-70"><a href="#cb13-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-71"><a href="#cb13-71" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Read in the data and check the structure. I used the name <span class="in">`adip`</span> for the dataframe/tibble.</span>
<span id="cb13-72"><a href="#cb13-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-75"><a href="#cb13-75" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-76"><a href="#cb13-76" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false  </span></span>
<span id="cb13-77"><a href="#cb13-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-78"><a href="#cb13-78" aria-hidden="true" tabindex="-1"></a><span class="co">#| #---CODING ANSWER---</span></span>
<span id="cb13-79"><a href="#cb13-79" aria-hidden="true" tabindex="-1"></a>adip  <span class="ot">&lt;-</span>  <span class="fu">read_table</span>(<span class="st">"data-raw/adipocytes.txt"</span>)</span>
<span id="cb13-80"><a href="#cb13-80" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(adip)</span>
<span id="cb13-81"><a href="#cb13-81" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-82"><a href="#cb13-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-83"><a href="#cb13-83" aria-hidden="true" tabindex="-1"></a>We have a tibble containing two variables: <span class="in">`adiponectin`</span> is the response and is continuous and <span class="in">`treatment`</span> is explanatory. <span class="in">`treatment`</span> is categorical with two levels (groups). The first task is visualise the data to get an overview. For continuous response variables with categorical explanatory variables you could use <span class="in">`geom_point()`</span>, <span class="in">`geom_boxplot()`</span> or a variety of other geoms. I often use <span class="in">`geom_violin()`</span> which allows us to see the distribution  - the violin is fatter where there are more data points.</span>
<span id="cb13-84"><a href="#cb13-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-85"><a href="#cb13-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-86"><a href="#cb13-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-87"><a href="#cb13-87" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Do a quick plot of the data:</span>
<span id="cb13-90"><a href="#cb13-90" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-91"><a href="#cb13-91" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> adip, <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> adiponectin)) <span class="sc">+</span></span>
<span id="cb13-92"><a href="#cb13-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>()</span>
<span id="cb13-93"><a href="#cb13-93" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-94"><a href="#cb13-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-95"><a href="#cb13-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-96"><a href="#cb13-96" aria-hidden="true" tabindex="-1"></a><span class="fu">### Summarising the data</span></span>
<span id="cb13-97"><a href="#cb13-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-98"><a href="#cb13-98" aria-hidden="true" tabindex="-1"></a>Summarising the data for each treatment group is the next sensible step. The most useful summary statistics are the means, standard deviations, sample sizes and standard errors.</span>
<span id="cb13-99"><a href="#cb13-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-100"><a href="#cb13-100" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Create a data frame called <span class="in">`adip_summary`</span> that contains the means, standard deviations, sample sizes and standard errors for the control and nicotinic acid treated samples. You may need to the <span class="co">[</span><span class="ot">Summarise</span><span class="co">](../week-4/workshop.html#summarise)</span> from the Week 4 workshop</span>
<span id="cb13-101"><a href="#cb13-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-104"><a href="#cb13-104" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-105"><a href="#cb13-105" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false  </span></span>
<span id="cb13-106"><a href="#cb13-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-107"><a href="#cb13-107" aria-hidden="true" tabindex="-1"></a><span class="co">#| #---CODING ANSWER---</span></span>
<span id="cb13-108"><a href="#cb13-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-109"><a href="#cb13-109" aria-hidden="true" tabindex="-1"></a>adip_summary <span class="ot">&lt;-</span> adip <span class="sc">%&gt;%</span></span>
<span id="cb13-110"><a href="#cb13-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(treatment) <span class="sc">%&gt;%</span></span>
<span id="cb13-111"><a href="#cb13-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(adiponectin),</span>
<span id="cb13-112"><a href="#cb13-112" aria-hidden="true" tabindex="-1"></a>            <span class="at">std =</span> <span class="fu">sd</span>(adiponectin),</span>
<span id="cb13-113"><a href="#cb13-113" aria-hidden="true" tabindex="-1"></a>            <span class="at">n =</span> <span class="fu">length</span>(adiponectin),</span>
<span id="cb13-114"><a href="#cb13-114" aria-hidden="true" tabindex="-1"></a>            <span class="at">se =</span> std<span class="sc">/</span><span class="fu">sqrt</span>(n))</span>
<span id="cb13-115"><a href="#cb13-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-116"><a href="#cb13-116" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-117"><a href="#cb13-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-118"><a href="#cb13-118" aria-hidden="true" tabindex="-1"></a>You should get the following numbers:</span>
<span id="cb13-119"><a href="#cb13-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-122"><a href="#cb13-122" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-123"><a href="#cb13-123" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb13-124"><a href="#cb13-124" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(adip_summary) <span class="sc">%&gt;%</span> </span>
<span id="cb13-125"><a href="#cb13-125" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>()</span>
<span id="cb13-126"><a href="#cb13-126" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-127"><a href="#cb13-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-128"><a href="#cb13-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-129"><a href="#cb13-129" aria-hidden="true" tabindex="-1"></a><span class="fu">### Selecting a test</span></span>
<span id="cb13-130"><a href="#cb13-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-131"><a href="#cb13-131" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> Do you think this is a paired-sample test or two-sample test?</span>
<span id="cb13-132"><a href="#cb13-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-133"><a href="#cb13-133" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER---  --&gt;</span></span>
<span id="cb13-134"><a href="#cb13-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-135"><a href="#cb13-135" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Two-sample - the samples are either control or nicotinic acid treated, not both.  --&gt;</span></span>
<span id="cb13-136"><a href="#cb13-136" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- There is no link between the first control value and the first nicotinic acid  --&gt;</span></span>
<span id="cb13-137"><a href="#cb13-137" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- value. The two samples are independent and it is a two-sample --&gt;</span></span>
<span id="cb13-138"><a href="#cb13-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-139"><a href="#cb13-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-140"><a href="#cb13-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-141"><a href="#cb13-141" aria-hidden="true" tabindex="-1"></a><span class="fu">### Applying, interpreting and reporting</span></span>
<span id="cb13-142"><a href="#cb13-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-143"><a href="#cb13-143" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Create a two-sample model like this:</span>
<span id="cb13-144"><a href="#cb13-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-147"><a href="#cb13-147" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-148"><a href="#cb13-148" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="at">data =</span> adip,</span>
<span id="cb13-149"><a href="#cb13-149" aria-hidden="true" tabindex="-1"></a>          adiponectin <span class="sc">~</span> treatment)</span>
<span id="cb13-150"><a href="#cb13-150" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-151"><a href="#cb13-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-152"><a href="#cb13-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-153"><a href="#cb13-153" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Examine the model with:</span>
<span id="cb13-154"><a href="#cb13-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-157"><a href="#cb13-157" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-158"><a href="#cb13-158" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(mod)</span>
<span id="cb13-159"><a href="#cb13-159" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-160"><a href="#cb13-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-161"><a href="#cb13-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-162"><a href="#cb13-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-163"><a href="#cb13-163" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> What do you conclude from the test? Write your conclusion in a form suitable for a report. </span>
<span id="cb13-164"><a href="#cb13-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-165"><a href="#cb13-165" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER---  --&gt;</span></span>
<span id="cb13-166"><a href="#cb13-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-167"><a href="#cb13-167" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Adipoctyes treated with nicotinic acid (mean +/- s.e.: 7.51 +/- 1.10 pg/mL)  --&gt;</span></span>
<span id="cb13-168"><a href="#cb13-168" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- secrete significantly more adiponectin than controls (5.55  +/- 0.12 pg/mL)  --&gt;</span></span>
<span id="cb13-169"><a href="#cb13-169" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- (t = 3.27; d.f. = 28; p = 0.003). --&gt;</span></span>
<span id="cb13-170"><a href="#cb13-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-171"><a href="#cb13-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-172"><a href="#cb13-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-173"><a href="#cb13-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-174"><a href="#cb13-174" aria-hidden="true" tabindex="-1"></a><span class="fu">### Check assumptions</span></span>
<span id="cb13-175"><a href="#cb13-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-176"><a href="#cb13-176" aria-hidden="true" tabindex="-1"></a>The assumptions of the general linear model are that the residuals – the difference between predicted value (i.e., the group mean) and observed values - are normally distributed and have homogeneous variance. To check these we can examine the <span class="in">`mod$residuals`</span> variable. You may want to refer to <span class="co">[</span><span class="ot">Checking assumptions</span><span class="co">](../week-6/workshop.html#checking-assumptions)</span> in the "Single regression" workshop.</span>
<span id="cb13-177"><a href="#cb13-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-178"><a href="#cb13-178" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span>  Plot the model residuals against the fitted values.</span>
<span id="cb13-181"><a href="#cb13-181" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-182"><a href="#cb13-182" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb13-183"><a href="#cb13-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-184"><a href="#cb13-184" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER-</span></span>
<span id="cb13-185"><a href="#cb13-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-186"><a href="#cb13-186" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mod, <span class="at">which =</span> <span class="dv">1</span>)</span>
<span id="cb13-187"><a href="#cb13-187" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-188"><a href="#cb13-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-189"><a href="#cb13-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-190"><a href="#cb13-190" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span>  What to you conclude?</span>
<span id="cb13-191"><a href="#cb13-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-192"><a href="#cb13-192" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER--- --&gt;</span></span>
<span id="cb13-193"><a href="#cb13-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-194"><a href="#cb13-194" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- The residuals look 'homogeneously' distributed. That is, the variance is the same  --&gt;</span></span>
<span id="cb13-195"><a href="#cb13-195" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- for the three groups  --&gt;</span></span>
<span id="cb13-196"><a href="#cb13-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-197"><a href="#cb13-197" aria-hidden="true" tabindex="-1"></a>To examine normality of the model residuals we can plot them as a histogram and do a normality test on them.</span>
<span id="cb13-198"><a href="#cb13-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-199"><a href="#cb13-199" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span>  Plot a histogram of the residuals.</span>
<span id="cb13-202"><a href="#cb13-202" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-203"><a href="#cb13-203" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb13-204"><a href="#cb13-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-205"><a href="#cb13-205" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER-</span></span>
<span id="cb13-206"><a href="#cb13-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-207"><a href="#cb13-207" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> mod<span class="sc">$</span>residuals)) <span class="sc">+</span> </span>
<span id="cb13-208"><a href="#cb13-208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">10</span>)</span>
<span id="cb13-209"><a href="#cb13-209" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-210"><a href="#cb13-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-211"><a href="#cb13-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-212"><a href="#cb13-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-213"><a href="#cb13-213" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Use the <span class="in">`shapiro.test()`</span> to test the normality of the model residuals</span>
<span id="cb13-216"><a href="#cb13-216" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-217"><a href="#cb13-217" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb13-218"><a href="#cb13-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-219"><a href="#cb13-219" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER-</span></span>
<span id="cb13-220"><a href="#cb13-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-221"><a href="#cb13-221" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(mod<span class="sc">$</span>residuals)</span>
<span id="cb13-222"><a href="#cb13-222" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-223"><a href="#cb13-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-224"><a href="#cb13-224" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span>  What to you conclude?</span>
<span id="cb13-225"><a href="#cb13-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-226"><a href="#cb13-226" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER--- --&gt;</span></span>
<span id="cb13-227"><a href="#cb13-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-228"><a href="#cb13-228" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- the residuals appear roughly normally distributed.  --&gt;</span></span>
<span id="cb13-229"><a href="#cb13-229" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- The histogram is symmetrical and the shapiro test is NS. --&gt;</span></span>
<span id="cb13-230"><a href="#cb13-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-231"><a href="#cb13-231" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Overall, it seems the assumptions are met --&gt;</span></span>
<span id="cb13-232"><a href="#cb13-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-233"><a href="#cb13-233" aria-hidden="true" tabindex="-1"></a><span class="fu">### Illustrating</span></span>
<span id="cb13-234"><a href="#cb13-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-235"><a href="#cb13-235" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Create a figure like the one below. You may need to refer to <span class="co">[</span><span class="ot">Visualise</span><span class="co">](../week-4/workshop.html#visualise)</span> from the "Summarising data with several variables" workshop <span class="co">[</span><span class="ot">@rand2023</span><span class="co">]</span></span>
<span id="cb13-236"><a href="#cb13-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-239"><a href="#cb13-239" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-240"><a href="#cb13-240" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb13-241"><a href="#cb13-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-242"><a href="#cb13-242" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER---</span></span>
<span id="cb13-243"><a href="#cb13-243" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb13-244"><a href="#cb13-244" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> adip, <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> adiponectin),</span>
<span id="cb13-245"><a href="#cb13-245" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">height =</span> <span class="dv">0</span>),</span>
<span id="cb13-246"><a href="#cb13-246" aria-hidden="true" tabindex="-1"></a>             <span class="at">colour =</span> <span class="st">"gray50"</span>) <span class="sc">+</span></span>
<span id="cb13-247"><a href="#cb13-247" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> adip_summary, </span>
<span id="cb13-248"><a href="#cb13-248" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">ymin =</span> mean <span class="sc">-</span> se, <span class="at">ymax =</span> mean <span class="sc">+</span> se),</span>
<span id="cb13-249"><a href="#cb13-249" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb13-250"><a href="#cb13-250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> adip_summary, </span>
<span id="cb13-251"><a href="#cb13-251" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">ymin =</span> mean, <span class="at">ymax =</span> mean),</span>
<span id="cb13-252"><a href="#cb13-252" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb13-253"><a href="#cb13-253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Adiponectin (pg/mL)"</span>, </span>
<span id="cb13-254"><a href="#cb13-254" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>), </span>
<span id="cb13-255"><a href="#cb13-255" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb13-256"><a href="#cb13-256" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Treatment"</span>, </span>
<span id="cb13-257"><a href="#cb13-257" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Control"</span>, <span class="st">"Nicotinic acid"</span>)) <span class="sc">+</span></span>
<span id="cb13-258"><a href="#cb13-258" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb13-259"><a href="#cb13-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-260"><a href="#cb13-260" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-261"><a href="#cb13-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-262"><a href="#cb13-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-263"><a href="#cb13-263" aria-hidden="true" tabindex="-1"></a>We now need to annotate the figure with the results from the statistical test. This most commonly done with a line linking the means being compared and the *p*-value. The `annotate()` function can be used to draw the line and then to add the value. The line is a `segment` and the *p*-value is a <span class="in">`text`</span>.</span>
<span id="cb13-264"><a href="#cb13-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-265"><a href="#cb13-265" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Add annotation to the figure by adding:</span>
<span id="cb13-266"><a href="#cb13-266" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb13-267"><a href="#cb13-267" aria-hidden="true" tabindex="-1"></a>...... <span class="sc">+</span></span>
<span id="cb13-268"><a href="#cb13-268" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>, </span>
<span id="cb13-269"><a href="#cb13-269" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="fl">11.3</span>, <span class="at">yend =</span> <span class="fl">11.3</span>,</span>
<span id="cb13-270"><a href="#cb13-270" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb13-271"><a href="#cb13-271" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>,  <span class="at">y =</span> <span class="fl">11.7</span>, </span>
<span id="cb13-272"><a href="#cb13-272" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">expression</span>(<span class="fu">italic</span>(p)<span class="sc">~</span><span class="st">"= 0.003"</span>)) <span class="sc">+</span></span>
<span id="cb13-273"><a href="#cb13-273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb13-274"><a href="#cb13-274" aria-hidden="true" tabindex="-1"></a><span class="in">```</span>  </span>
<span id="cb13-275"><a href="#cb13-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-278"><a href="#cb13-278" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-279"><a href="#cb13-279" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb13-280"><a href="#cb13-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-281"><a href="#cb13-281" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb13-282"><a href="#cb13-282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> adip, <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> adiponectin),</span>
<span id="cb13-283"><a href="#cb13-283" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">height =</span> <span class="dv">0</span>),</span>
<span id="cb13-284"><a href="#cb13-284" aria-hidden="true" tabindex="-1"></a>             <span class="at">colour =</span> <span class="st">"gray50"</span>) <span class="sc">+</span></span>
<span id="cb13-285"><a href="#cb13-285" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> adip_summary, </span>
<span id="cb13-286"><a href="#cb13-286" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">ymin =</span> mean <span class="sc">-</span> se, <span class="at">ymax =</span> mean <span class="sc">+</span> se),</span>
<span id="cb13-287"><a href="#cb13-287" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb13-288"><a href="#cb13-288" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> adip_summary, </span>
<span id="cb13-289"><a href="#cb13-289" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">ymin =</span> mean, <span class="at">ymax =</span> mean),</span>
<span id="cb13-290"><a href="#cb13-290" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb13-291"><a href="#cb13-291" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Adiponectin (pg/mL)"</span>, </span>
<span id="cb13-292"><a href="#cb13-292" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>), </span>
<span id="cb13-293"><a href="#cb13-293" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb13-294"><a href="#cb13-294" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Treatment"</span>, </span>
<span id="cb13-295"><a href="#cb13-295" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Control"</span>, <span class="st">"Nicotinic acid"</span>)) <span class="sc">+</span></span>
<span id="cb13-296"><a href="#cb13-296" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>, </span>
<span id="cb13-297"><a href="#cb13-297" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="fl">11.3</span>, <span class="at">yend =</span> <span class="fl">11.3</span>,</span>
<span id="cb13-298"><a href="#cb13-298" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb13-299"><a href="#cb13-299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>,  <span class="at">y =</span> <span class="fl">11.7</span>, </span>
<span id="cb13-300"><a href="#cb13-300" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">expression</span>(<span class="fu">italic</span>(p)<span class="sc">~</span><span class="st">"= 0.003"</span>)) <span class="sc">+</span></span>
<span id="cb13-301"><a href="#cb13-301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb13-302"><a href="#cb13-302" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-303"><a href="#cb13-303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-304"><a href="#cb13-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-305"><a href="#cb13-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-306"><a href="#cb13-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-307"><a href="#cb13-307" aria-hidden="true" tabindex="-1"></a>For the segment, <span class="in">`annotate()`</span> needs the *x* and *y* coordinates for the start and the finish of the line.</span>
<span id="cb13-308"><a href="#cb13-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-309"><a href="#cb13-309" aria-hidden="true" tabindex="-1"></a>The use of <span class="in">`expression()`</span> allows you to specify formatting or special characters. <span class="in">`expression()`</span> takes strings or <span class="co">[</span><span class="ot">LaTeX</span><span class="co">](https://en.wikipedia.org/wiki/LaTeX)</span> formatting. Each string or piece of LaTeX is separated by a <span class="in">`*`</span> or a <span class="in">`~`</span>. The <span class="in">`*`</span> concatenates the strings without a space, <span class="in">`~`</span> does so with a space.</span>
<span id="cb13-310"><a href="#cb13-310" aria-hidden="true" tabindex="-1"></a>It will generate a warning message "In is.na(x) : is.na() applied to non-(list or vector) of type 'expression'" which can be ignored.</span>
<span id="cb13-311"><a href="#cb13-311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-312"><a href="#cb13-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-313"><a href="#cb13-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-314"><a href="#cb13-314" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Save your figure to your figures folder.</span>
<span id="cb13-317"><a href="#cb13-317" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-318"><a href="#cb13-318" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb13-319"><a href="#cb13-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-320"><a href="#cb13-320" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER---</span></span>
<span id="cb13-321"><a href="#cb13-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-322"><a href="#cb13-322" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign the figure to a variable</span></span>
<span id="cb13-323"><a href="#cb13-323" aria-hidden="true" tabindex="-1"></a>fig1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb13-324"><a href="#cb13-324" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> adip, <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> adiponectin),</span>
<span id="cb13-325"><a href="#cb13-325" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">height =</span> <span class="dv">0</span>),</span>
<span id="cb13-326"><a href="#cb13-326" aria-hidden="true" tabindex="-1"></a>             <span class="at">colour =</span> <span class="st">"gray50"</span>) <span class="sc">+</span></span>
<span id="cb13-327"><a href="#cb13-327" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> adip_summary, </span>
<span id="cb13-328"><a href="#cb13-328" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">ymin =</span> mean <span class="sc">-</span> se, <span class="at">ymax =</span> mean <span class="sc">+</span> se),</span>
<span id="cb13-329"><a href="#cb13-329" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb13-330"><a href="#cb13-330" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> adip_summary, </span>
<span id="cb13-331"><a href="#cb13-331" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">ymin =</span> mean, <span class="at">ymax =</span> mean),</span>
<span id="cb13-332"><a href="#cb13-332" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb13-333"><a href="#cb13-333" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Adiponectin (pg/mL)"</span>, </span>
<span id="cb13-334"><a href="#cb13-334" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>), </span>
<span id="cb13-335"><a href="#cb13-335" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb13-336"><a href="#cb13-336" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Treatment"</span>, </span>
<span id="cb13-337"><a href="#cb13-337" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Control"</span>, <span class="st">"Nicotinic acid"</span>)) <span class="sc">+</span></span>
<span id="cb13-338"><a href="#cb13-338" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>, </span>
<span id="cb13-339"><a href="#cb13-339" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="fl">11.3</span>, <span class="at">yend =</span> <span class="fl">11.3</span>,</span>
<span id="cb13-340"><a href="#cb13-340" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb13-341"><a href="#cb13-341" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>,  <span class="at">y =</span> <span class="fl">11.7</span>, </span>
<span id="cb13-342"><a href="#cb13-342" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">expression</span>(<span class="fu">italic</span>(p)<span class="sc">~</span><span class="st">"= 0.003"</span>)) <span class="sc">+</span></span>
<span id="cb13-343"><a href="#cb13-343" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb13-344"><a href="#cb13-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-345"><a href="#cb13-345" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-346"><a href="#cb13-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-347"><a href="#cb13-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-350"><a href="#cb13-350" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-351"><a href="#cb13-351" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb13-352"><a href="#cb13-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-353"><a href="#cb13-353" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER---</span></span>
<span id="cb13-354"><a href="#cb13-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-355"><a href="#cb13-355" aria-hidden="true" tabindex="-1"></a><span class="co"># save figure to figures/adipo.png</span></span>
<span id="cb13-356"><a href="#cb13-356" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"figures/adipo.png"</span>,</span>
<span id="cb13-357"><a href="#cb13-357" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> fig1,</span>
<span id="cb13-358"><a href="#cb13-358" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="fl">3.5</span>,</span>
<span id="cb13-359"><a href="#cb13-359" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="fl">3.5</span>,</span>
<span id="cb13-360"><a href="#cb13-360" aria-hidden="true" tabindex="-1"></a>       <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb13-361"><a href="#cb13-361" aria-hidden="true" tabindex="-1"></a>       <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb13-362"><a href="#cb13-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-363"><a href="#cb13-363" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-364"><a href="#cb13-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-365"><a href="#cb13-365" aria-hidden="true" tabindex="-1"></a><span class="fu">## Grouse Parasites</span></span>
<span id="cb13-366"><a href="#cb13-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-367"><a href="#cb13-367" aria-hidden="true" tabindex="-1"></a>Grouse livers were dissected and the number of individuals of a parasitic nematode were counted for two estates 'Gordon' and 'Moss'. We want to know if the two estates have different infection rates. The data are in <span class="co">[</span><span class="ot">grouse.csv</span><span class="co">](data-raw/grouse.csv)</span></span>
<span id="cb13-368"><a href="#cb13-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-369"><a href="#cb13-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-370"><a href="#cb13-370" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_on_your_computer.png)</span> Save a copy of <span class="co">[</span><span class="ot">grouse.csv</span><span class="co">](data-raw/grouse.csv)</span> to <span class="in">`data-raw`</span></span>
<span id="cb13-371"><a href="#cb13-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-372"><a href="#cb13-372" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Read in the data and check the structure. I used the name <span class="in">`grouse`</span> for the dataframe/tibble.</span>
<span id="cb13-373"><a href="#cb13-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-376"><a href="#cb13-376" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-377"><a href="#cb13-377" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false  </span></span>
<span id="cb13-378"><a href="#cb13-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-379"><a href="#cb13-379" aria-hidden="true" tabindex="-1"></a><span class="co">#| #---CODING ANSWER---</span></span>
<span id="cb13-380"><a href="#cb13-380" aria-hidden="true" tabindex="-1"></a>grouse  <span class="ot">&lt;-</span>  <span class="fu">read_csv</span>(<span class="st">"data-raw/grouse.csv"</span>)</span>
<span id="cb13-381"><a href="#cb13-381" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(grouse)</span>
<span id="cb13-382"><a href="#cb13-382" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-383"><a href="#cb13-383" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb13-384"><a href="#cb13-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-385"><a href="#cb13-385" aria-hidden="true" tabindex="-1"></a><span class="fu">### Selecting</span></span>
<span id="cb13-386"><a href="#cb13-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-387"><a href="#cb13-387" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> Using your common sense, do these data look normally distributed? </span>
<span id="cb13-388"><a href="#cb13-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-389"><a href="#cb13-389" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER---  --&gt;</span></span>
<span id="cb13-390"><a href="#cb13-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-391"><a href="#cb13-391" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- No and little need to test/plot - they're counts, skewed and  --&gt;</span></span>
<span id="cb13-392"><a href="#cb13-392" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- have unequal variances --&gt;</span></span>
<span id="cb13-393"><a href="#cb13-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-394"><a href="#cb13-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-395"><a href="#cb13-395" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> What test do you suggest?</span>
<span id="cb13-396"><a href="#cb13-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-397"><a href="#cb13-397" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER---  --&gt;</span></span>
<span id="cb13-398"><a href="#cb13-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-399"><a href="#cb13-399" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- two-sample Wilcoxon (unpaired) AKA Mann-Whitney --&gt;</span></span>
<span id="cb13-400"><a href="#cb13-400" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-401"><a href="#cb13-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-402"><a href="#cb13-402" aria-hidden="true" tabindex="-1"></a><span class="fu">### Applying, interpreting and reporting</span></span>
<span id="cb13-403"><a href="#cb13-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-404"><a href="#cb13-404" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Summarise the data by finding the median of each group:</span>
<span id="cb13-407"><a href="#cb13-407" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-408"><a href="#cb13-408" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false  </span></span>
<span id="cb13-409"><a href="#cb13-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-410"><a href="#cb13-410" aria-hidden="true" tabindex="-1"></a><span class="co"># ---CODING ANSWER---</span></span>
<span id="cb13-411"><a href="#cb13-411" aria-hidden="true" tabindex="-1"></a>grouse <span class="sc">%&gt;%</span> </span>
<span id="cb13-412"><a href="#cb13-412" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(estate) <span class="sc">%&gt;%</span> </span>
<span id="cb13-413"><a href="#cb13-413" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">median</span>(nematodes))</span>
<span id="cb13-414"><a href="#cb13-414" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-415"><a href="#cb13-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-416"><a href="#cb13-416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-417"><a href="#cb13-417" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span>  Carry out a two-sample Wilcoxon test (also known as a Mann-Whitney):</span>
<span id="cb13-420"><a href="#cb13-420" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-421"><a href="#cb13-421" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(<span class="at">data =</span> grouse, nematodes <span class="sc">~</span> estate)</span>
<span id="cb13-422"><a href="#cb13-422" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-423"><a href="#cb13-423" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> What do you conclude from the test? Write your conclusion in a form suitable for a report. </span>
<span id="cb13-424"><a href="#cb13-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-425"><a href="#cb13-425" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER---  --&gt;</span></span>
<span id="cb13-426"><a href="#cb13-426" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-427"><a href="#cb13-427" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Grouse on Gordon had significantly more nematodes (median = 13.5) than those  --&gt;</span></span>
<span id="cb13-428"><a href="#cb13-428" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- on Moss (median = 5) (W = 78; n1 = 10, n2 = 10; p = 0.03546). --&gt;</span></span>
<span id="cb13-429"><a href="#cb13-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-430"><a href="#cb13-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-431"><a href="#cb13-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-432"><a href="#cb13-432" aria-hidden="true" tabindex="-1"></a><span class="fu">### Illustrating</span></span>
<span id="cb13-433"><a href="#cb13-433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-434"><a href="#cb13-434" aria-hidden="true" tabindex="-1"></a>A box plot is a usually good choice for illustrating a two-sample Wilcoxon test because it shows the median and interquartile range.</span>
<span id="cb13-435"><a href="#cb13-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-436"><a href="#cb13-436" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> We can create a simple boxplot with:  </span>
<span id="cb13-439"><a href="#cb13-439" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-440"><a href="#cb13-440" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> grouse, <span class="fu">aes</span>(<span class="at">x =</span> estate, <span class="at">y =</span> nematodes) ) <span class="sc">+</span></span>
<span id="cb13-441"><a href="#cb13-441" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() </span>
<span id="cb13-442"><a href="#cb13-442" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-443"><a href="#cb13-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-444"><a href="#cb13-444" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Annotate and format the figure so it is more suitable for a report and save it to your figures folder.  </span>
<span id="cb13-447"><a href="#cb13-447" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-448"><a href="#cb13-448" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false  </span></span>
<span id="cb13-449"><a href="#cb13-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-450"><a href="#cb13-450" aria-hidden="true" tabindex="-1"></a><span class="co"># ---CODING ANSWER---</span></span>
<span id="cb13-451"><a href="#cb13-451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-452"><a href="#cb13-452" aria-hidden="true" tabindex="-1"></a>fig2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> grouse, <span class="fu">aes</span>(<span class="at">x =</span> estate, <span class="at">y =</span> nematodes) ) <span class="sc">+</span></span>
<span id="cb13-453"><a href="#cb13-453" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb13-454"><a href="#cb13-454" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Estate"</span>, </span>
<span id="cb13-455"><a href="#cb13-455" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Gordon"</span>, <span class="st">"Moss"</span>)) <span class="sc">+</span></span>
<span id="cb13-456"><a href="#cb13-456" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Number of nematodes"</span>,</span>
<span id="cb13-457"><a href="#cb13-457" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>),</span>
<span id="cb13-458"><a href="#cb13-458" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">75</span>)) <span class="sc">+</span></span>
<span id="cb13-459"><a href="#cb13-459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>, </span>
<span id="cb13-460"><a href="#cb13-460" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="dv">67</span>, <span class="at">yend =</span> <span class="dv">67</span>,</span>
<span id="cb13-461"><a href="#cb13-461" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb13-462"><a href="#cb13-462" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>,  <span class="at">y =</span> <span class="dv">69</span>, </span>
<span id="cb13-463"><a href="#cb13-463" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">expression</span>(<span class="fu">italic</span>(p)<span class="sc">~</span><span class="st">"= 0.035"</span>)) <span class="sc">+</span></span>
<span id="cb13-464"><a href="#cb13-464" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb13-465"><a href="#cb13-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-466"><a href="#cb13-466" aria-hidden="true" tabindex="-1"></a><span class="co"># save figure to figures/grouse.png</span></span>
<span id="cb13-467"><a href="#cb13-467" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"figures/grouse.png"</span>,</span>
<span id="cb13-468"><a href="#cb13-468" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> fig2,</span>
<span id="cb13-469"><a href="#cb13-469" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">3</span>,</span>
<span id="cb13-470"><a href="#cb13-470" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">3</span>,</span>
<span id="cb13-471"><a href="#cb13-471" aria-hidden="true" tabindex="-1"></a>       <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb13-472"><a href="#cb13-472" aria-hidden="true" tabindex="-1"></a>       <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb13-473"><a href="#cb13-473" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-474"><a href="#cb13-474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-475"><a href="#cb13-475" aria-hidden="true" tabindex="-1"></a><span class="fu">## Gene Expression</span></span>
<span id="cb13-476"><a href="#cb13-476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-477"><a href="#cb13-477" aria-hidden="true" tabindex="-1"></a>Bambara groundnut (*Vigna subterranea*) is an African legume with good  nutritional value which can be influenced by low temperature stress. Researchers are interested in the expression levels of a particular set of 35 genes (`probe_id`) in response to temperature stress. They measure the expression of the genes at 23 and 18 degrees C (high and low temperature). These samples are **not** independent because we have two measure from one gene. The data are in <span class="co">[</span><span class="ot">expr.xlxs</span><span class="co">](data-raw/expr.xlsx)</span>.</span>
<span id="cb13-478"><a href="#cb13-478" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-479"><a href="#cb13-479" aria-hidden="true" tabindex="-1"></a><span class="fu">### Selecting</span></span>
<span id="cb13-480"><a href="#cb13-480" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> What is the null hypothesis?</span>
<span id="cb13-481"><a href="#cb13-481" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-482"><a href="#cb13-482" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER---  --&gt;</span></span>
<span id="cb13-483"><a href="#cb13-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-484"><a href="#cb13-484" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- expression is not affected by temperature / for each gene,  --&gt;</span></span>
<span id="cb13-485"><a href="#cb13-485" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- the expression at the higher t   = expression at the lower t --&gt;</span></span>
<span id="cb13-486"><a href="#cb13-486" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-487"><a href="#cb13-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-488"><a href="#cb13-488" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Save a copy of <span class="co">[</span><span class="ot">expr.xlxs</span><span class="co">](data-raw/expr.xlsx)</span> and import the data. I named the dataframe <span class="in">`bambara`</span></span>
<span id="cb13-489"><a href="#cb13-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-492"><a href="#cb13-492" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-493"><a href="#cb13-493" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false  </span></span>
<span id="cb13-494"><a href="#cb13-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-495"><a href="#cb13-495" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER---</span></span>
<span id="cb13-496"><a href="#cb13-496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-497"><a href="#cb13-497" aria-hidden="true" tabindex="-1"></a><span class="co"># we need the readxl package</span></span>
<span id="cb13-498"><a href="#cb13-498" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb13-499"><a href="#cb13-499" aria-hidden="true" tabindex="-1"></a>bambara  <span class="ot">&lt;-</span>  <span class="fu">read_excel</span>(<span class="st">"data-raw/expr.xlsx"</span>)</span>
<span id="cb13-500"><a href="#cb13-500" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-501"><a href="#cb13-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-502"><a href="#cb13-502" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> What is the appropriate parametric test?</span>
<span id="cb13-503"><a href="#cb13-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-504"><a href="#cb13-504" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER--- --&gt;</span></span>
<span id="cb13-505"><a href="#cb13-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-506"><a href="#cb13-506" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- the appropriate parametric test is the paired sample lm  --&gt;</span></span>
<span id="cb13-507"><a href="#cb13-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-508"><a href="#cb13-508" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-509"><a href="#cb13-509" aria-hidden="true" tabindex="-1"></a><span class="fu">### Applying, interpreting and reporting</span></span>
<span id="cb13-510"><a href="#cb13-510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-511"><a href="#cb13-511" aria-hidden="true" tabindex="-1"></a>A paired test requires us to test whether the difference in expression between high and low temperatures is zero on average. One handy way to achieve this is to organise our groups into two columns. The <span class="in">`pivot_wider()`</span> function will do this for us. We need to tell it what column gives the identifiers (i.e., matches the the pairs) - the probe_ids in this case. We also need to say which variable contains what will become the column names and which contains the values.</span>
<span id="cb13-512"><a href="#cb13-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-513"><a href="#cb13-513" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Pivot the data so there is a column for each temperature: </span>
<span id="cb13-516"><a href="#cb13-516" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-517"><a href="#cb13-517" aria-hidden="true" tabindex="-1"></a>bambara <span class="ot">&lt;-</span> bambara <span class="sc">|&gt;</span> </span>
<span id="cb13-518"><a href="#cb13-518" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> temperature, </span>
<span id="cb13-519"><a href="#cb13-519" aria-hidden="true" tabindex="-1"></a>              <span class="at">values_from =</span> expression, </span>
<span id="cb13-520"><a href="#cb13-520" aria-hidden="true" tabindex="-1"></a>              <span class="at">id_cols =</span> probe_id)</span>
<span id="cb13-521"><a href="#cb13-521" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-522"><a href="#cb13-522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-523"><a href="#cb13-523" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Click on the <span class="in">`bambara`</span> dataframe in the environment to open a view of it so that you understand what <span class="in">`pivot_wider()`</span> has done.</span>
<span id="cb13-524"><a href="#cb13-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-525"><a href="#cb13-525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-526"><a href="#cb13-526" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Create a paired-sample model like this:</span>
<span id="cb13-527"><a href="#cb13-527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-530"><a href="#cb13-530" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-531"><a href="#cb13-531" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="at">data =</span> bambara, </span>
<span id="cb13-532"><a href="#cb13-532" aria-hidden="true" tabindex="-1"></a>          highert <span class="sc">-</span> lowert <span class="sc">~</span> <span class="dv">1</span>)</span>
<span id="cb13-533"><a href="#cb13-533" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-534"><a href="#cb13-534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-535"><a href="#cb13-535" aria-hidden="true" tabindex="-1"></a>Since we have done <span class="in">`highert - lowert`</span>, the "(Intercept) Estimate" will be the average of the higher temperature expression minus the lower temperature expression for each gene.</span>
<span id="cb13-536"><a href="#cb13-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-537"><a href="#cb13-537" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Examine the model with:</span>
<span id="cb13-538"><a href="#cb13-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-541"><a href="#cb13-541" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb13-542"><a href="#cb13-542" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(mod)</span>
<span id="cb13-543"><a href="#cb13-543" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb13-544"><a href="#cb13-544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-545"><a href="#cb13-545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-546"><a href="#cb13-546" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> State your conclusion from the test in a form suitable for including in a report. Make sure you give the direction of any significant effect.</span>
<span id="cb13-547"><a href="#cb13-547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-548"><a href="#cb13-548" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER--- --&gt;</span></span>
<span id="cb13-549"><a href="#cb13-549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-550"><a href="#cb13-550" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- The expression of individual genes is significantly higher at higher temperatures  --&gt;</span></span>
<span id="cb13-551"><a href="#cb13-551" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- than lower temperatures (t = 3.20, d.f. = 34, p = 0.0029) with an average difference --&gt;</span></span>
<span id="cb13-552"><a href="#cb13-552" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- of 0.307 units. --&gt;</span></span>
<span id="cb13-553"><a href="#cb13-553" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-554"><a href="#cb13-554" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-555"><a href="#cb13-555" aria-hidden="true" tabindex="-1"></a><span class="fu">## Look after future you!</span></span>
<span id="cb13-556"><a href="#cb13-556" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-557"><a href="#cb13-557" aria-hidden="true" tabindex="-1"></a>The code required to summarise, test, and plot data for any two-sample test AND for any for any one-way ANOVA is **exactly the same** except for the names of the dataframe, variables and the axis labels and limits. Take some time to comment it your code so that you can make use of it next week.</span>
<span id="cb13-558"><a href="#cb13-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-559"><a href="#cb13-559" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/future_you.png)</span>{fig-alt="xxx" width="300"}</span>
<span id="cb13-560"><a href="#cb13-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-561"><a href="#cb13-561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-562"><a href="#cb13-562" aria-hidden="true" tabindex="-1"></a>You're finished!</span>
<span id="cb13-563"><a href="#cb13-563" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-564"><a href="#cb13-564" aria-hidden="true" tabindex="-1"></a><span class="fu"># 🥳 Well Done! 🎉</span></span>
<span id="cb13-565"><a href="#cb13-565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-566"><a href="#cb13-566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-567"><a href="#cb13-567" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-568"><a href="#cb13-568" aria-hidden="true" tabindex="-1"></a><span class="fu"># Independent study following the workshop</span></span>
<span id="cb13-569"><a href="#cb13-569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-570"><a href="#cb13-570" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">Consolidate</span><span class="co">](study_after_workshop.qmd)</span></span>
<span id="cb13-571"><a href="#cb13-571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-572"><a href="#cb13-572" aria-hidden="true" tabindex="-1"></a><span class="fu"># The Code file</span></span>
<span id="cb13-573"><a href="#cb13-573" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-574"><a href="#cb13-574" aria-hidden="true" tabindex="-1"></a>This contains all the code needed in the workshop even where it is not visible on the webpage.</span>
<span id="cb13-575"><a href="#cb13-575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-576"><a href="#cb13-576" aria-hidden="true" tabindex="-1"></a>The <span class="in">`workshop.qmd`</span> file is the file I use to compile the practical. Qmd stands for Quarto markdown. It allows code and ordinary text to be interweaved to produce well-formatted reports including webpages. <span class="co">[</span><span class="ot">View the Qmd in Browser</span><span class="co">](https://github.com/3mmaRand/R4BABS/blob/main/pgt52m/week-7/workshop.qmd)</span>. Coding and thinking answers are marked with <span class="in">`#---CODING ANSWER---`</span> and <span class="in">`#---THINKING ANSWER---`</span></span>
<span id="cb13-577"><a href="#cb13-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-578"><a href="#cb13-578" aria-hidden="true" tabindex="-1"></a>Pages made with R <span class="co">[</span><span class="ot">@R-core</span><span class="co">]</span>, Quarto <span class="co">[</span><span class="ot">@allaire2022</span><span class="co">]</span>, <span class="in">`knitr`</span> <span class="co">[</span><span class="ot">@knitr</span><span class="co">]</span>, <span class="in">`kableExtra`</span> <span class="co">[</span><span class="ot">@kableExtra</span><span class="co">]</span></span>
<span id="cb13-579"><a href="#cb13-579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-580"><a href="#cb13-580" aria-hidden="true" tabindex="-1"></a><span class="fu"># References</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>Rand, E. (2025). Data Analysis in R for Becoming a Bioscientist (Version 0.2) <a href="https://github.com/3mmaRand/R4BABS" class="uri">https://github.com/3mmaRand/R4BABS</a></p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/3mmaRand/comp4biosci/edit/main/pgt52m/week-7/workshop.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li></ul></div></div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
<li class="nav-item">
    <a class="nav-link" href="https://github.com/3mmaRand">
<p>Emma on GitHub</p>
</a>
  </li>  
</ul>
</div>
  </div>
</footer><script>
document.addEventListener("DOMContentLoaded", function() {
  // grab all links under “Other Links” in the right-hand TOC
  const links = document.querySelectorAll('.quarto-other-links ul li a');
  links.forEach(function(a) {
    // find the one whose text is exactly “Make a suggestion”
    if (a.textContent.trim() === 'Make a suggestion') {
      const pageUrl   = encodeURIComponent(window.location.href);
      const pageTitle = encodeURIComponent(document.title);
      a.href = 'https://github.com/3mmaRand/comp4biosci/issues/new'
             + '?title=' + pageTitle
             + '&body='  + pageUrl;
      a.target = '_blank';
    }
  });
});
</script><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>


</body></html>