<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.27">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Emma Rand">
<title>Workshop ‚Äì Data Analysis in R</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-ed96de9b727972fe78a7b5d16c58bf87.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark-4d9afe2b8d18ee9fa5d0d57b5ed4214d.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-ed96de9b727972fe78a7b5d16c58bf87.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-bd31597842da8e70aa6de660987a7f7b.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark-0724b92bd66eafb47a2b216599690d49.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../../site_libs/bootstrap/bootstrap-bd31597842da8e70aa6de660987a7f7b.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-50FV9HPTRE"></script><script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-50FV9HPTRE', { 'anonymize_ip': true});
</script><script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<meta property="og:title" content="Workshop ‚Äì Data Analysis in R">
<meta property="og:description" content="Data Analysis 1: Core">
<meta property="og:image" content="images/future_you.png">
<meta property="og:site_name" content="Data Analysis in R">
<meta property="og:image:alt" content="Person working at a computer with an offstage person asking 'How is the analysis going?' The person at the computer replies 'Can't understand the date...and the data collector does not answer my emails or calls' Person offstage: 'That's terrible! So cruel! Who did collect the data? I will sack them!' Person at the computer: 'um...I did, 3 years ago.'">
</head>
<body class="nav-sidebar floating nav-fixed quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Data Analysis in R</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Welcome!</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../r4babs1/r4babs1.html"> 
<span class="menu-text">BABS 1</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../r4babs2/r4babs2.html"> 
<span class="menu-text">BABS 2</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../r4babs4/r4babs4.html" aria-current="page"> 
<span class="menu-text">BABS 4</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../pgt52m/pgt52m.html"> 
<span class="menu-text">PGT 52M</span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
    <a href="https://github.com/3mmaRand" title="Emma on GitHub" class="quarto-navigation-tool px-1" aria-label="Emma on GitHub"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../r4babs4/week-1/overview.html">Week 1: DA 1 Core</a></li><li class="breadcrumb-item"><a href="../../r4babs4/week-1/workshop.html">Workshop</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/r4babs4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Analysis in R for BABS 4</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 1: DA 1 Core</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-1/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-1/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-1/workshop.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-1/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 2: DA 2 Biomed. Sci</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-2/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-2/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-2/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-2/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 4: DA 3 Biomed. Sci</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-4/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-4/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-4/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-4/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 6: DA 4 Biomed. Sci</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-6/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-6/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-6/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs4/week-6/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li>
<a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul class="collapse">
<li><a href="#session-overview" id="toc-session-overview" class="nav-link" data-scroll-target="#session-overview">Session overview</a></li>
  </ul>
</li>
  <li>
<a href="#reproducibility" id="toc-reproducibility" class="nav-link" data-scroll-target="#reproducibility">Reproducibility</a>
  <ul class="collapse">
<li><a href="#why-does-it-matter" id="toc-why-does-it-matter" class="nav-link" data-scroll-target="#why-does-it-matter">Why does it matter?</a></li>
  <li><a href="#how-to-achieve-reproducibility" id="toc-how-to-achieve-reproducibility" class="nav-link" data-scroll-target="#how-to-achieve-reproducibility">How to achieve reproducibility</a></li>
  </ul>
</li>
  <li><a href="#project-oriented-workflow" id="toc-project-oriented-workflow" class="nav-link" data-scroll-target="#project-oriented-workflow">Project-oriented workflow</a></li>
  <li><a href="#naming-things" id="toc-naming-things" class="nav-link" data-scroll-target="#naming-things">Naming things</a></li>
  <li>
<a href="#comment-your-code" id="toc-comment-your-code" class="nav-link" data-scroll-target="#comment-your-code">Comment your code</a>
  <ul class="collapse">
<li><a href="#example-comments" id="toc-example-comments" class="nav-link" data-scroll-target="#example-comments">Example comments</a></li>
  </ul>
</li>
  <li><a href="#critical-curation" id="toc-critical-curation" class="nav-link" data-scroll-target="#critical-curation">Critical Curation</a></li>
  <li><a href="#generative-ai-for-the-rstudio-project" id="toc-generative-ai-for-the-rstudio-project" class="nav-link" data-scroll-target="#generative-ai-for-the-rstudio-project">Generative AI for the RStudio Project</a></li>
  <li><a href="#big-data" id="toc-big-data" class="nav-link" data-scroll-target="#big-data">Big data</a></li>
  <li>
<a href="#exercises" id="toc-exercises" class="nav-link" data-scroll-target="#exercises">Exercises</a>
  <ul class="collapse">
<li><a href="#set-up-a-project" id="toc-set-up-a-project" class="nav-link" data-scroll-target="#set-up-a-project">Set up a Project</a></li>
  <li><a href="#load-packages" id="toc-load-packages" class="nav-link" data-scroll-target="#load-packages">Load packages</a></li>
  <li><a href="#look-at-the-data" id="toc-look-at-the-data" class="nav-link" data-scroll-target="#look-at-the-data">Look at the data!</a></li>
  <li><a href="#import-the-data" id="toc-import-the-data" class="nav-link" data-scroll-target="#import-the-data">Import the data</a></li>
  <li>
<a href="#getting-an-overview-1" id="toc-getting-an-overview-1" class="nav-link" data-scroll-target="#getting-an-overview-1">Getting an overview 1</a>
  <ul class="collapse">
<li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary"><code>summary()</code></a></li>
  </ul>
</li>
  <li>
<a href="#quality-control" id="toc-quality-control" class="nav-link" data-scroll-target="#quality-control">Quality Control</a>
  <ul class="collapse">
<li><a href="#filtering-rows" id="toc-filtering-rows" class="nav-link" data-scroll-target="#filtering-rows">Filtering rows</a></li>
  <li><a href="#selecting-columns" id="toc-selecting-columns" class="nav-link" data-scroll-target="#selecting-columns">Selecting columns</a></li>
  <li><a href="#visualisation" id="toc-visualisation" class="nav-link" data-scroll-target="#visualisation">Visualisation</a></li>
  </ul>
</li>
  </ul>
</li>
  <li><a href="#well-done" id="toc-well-done" class="nav-link" data-scroll-target="#well-done">ü•≥ Well Done! üéâ</a></li>
  <li><a href="#independent-study-following-the-workshop" id="toc-independent-study-following-the-workshop" class="nav-link" data-scroll-target="#independent-study-following-the-workshop">Independent study following the workshop</a></li>
  <li><a href="#the-code-file" id="toc-the-code-file" class="nav-link" data-scroll-target="#the-code-file">The Code file</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/3mmaRand/comp4biosci/edit/main/r4babs4/week-1/workshop.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li></ul></div><div class="quarto-other-links"><h2>Other Links</h2><ul><li><a href="#"><i class="bi bi-chat-left-dots"></i>Make a suggestion</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../r4babs4/week-1/overview.html">Week 1: DA 1 Core</a></li><li class="breadcrumb-item"><a href="../../r4babs4/week-1/workshop.html">Workshop</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Workshop</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">Data Analysis 1: Core</p>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><section id="introduction" class="level1"><h1>Introduction</h1>
<section id="session-overview" class="level2"><h2 class="anchored" data-anchor-id="session-overview">Session overview</h2>
<p>In the first part of the workshop I will talk about Project organisation for reproducible analysis and data that has many variables and observations. In the second part of the workshop you will practice getting an overview of such data with summaries and distribution plots, filtering rows and columns.</p>
</section></section><section id="reproducibility" class="level1"><h1>Reproducibility</h1>
<section id="why-does-it-matter" class="level2"><h2 class="anchored" data-anchor-id="why-does-it-matter">Why does it matter?</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="images/future_you.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="futureself, CC-BY-NC, by Julen Colomb"><img src="images/future_you.png" class="img-fluid figure-img" alt="Person working at a computer with an offstage person asking 'How is the analysis going?' The person at the computer replies 'Can't understand the date...and the data collector does not answer my emails or calls' Person offstage: 'That's terrible! So cruel! Who did collect the data? I will sack them!' Person at the computer: 'um...I did, 3 years ago.'" width="400"></a></p>
<figcaption>futureself, CC-BY-NC, by Julen Colomb</figcaption></figure>
</div>
<ul>
<li><p>Five selfish reasons to work reproducibly <span class="citation" data-cites="markowetz2015">(<a href="#ref-markowetz2015" role="doc-biblioref">Markowetz 2015</a>)</span>. Alternatively, see the very entertaining <a href="https://youtu.be/yVT07Sukv9Q">talk</a> which covers the ‚ÄúDuke Scandal‚Äù.</p></li>
<li><p>Many high profile cases of work which did not reproduce e.g.&nbsp;Anil Potti‚Äôs work unravelled by <span class="citation" data-cites="baggerly2009">Baggerly and Coombes (<a href="#ref-baggerly2009" role="doc-biblioref">2009</a>)</span> in the ‚ÄúDuke Scandal‚Äù</p></li>
<li><p><strong>Will</strong> become standard in Science and publishing e.g OECD Global Science Forum Building digital workforce capacity and skills for data-intensive science <span class="citation" data-cites="oecdglobalscienceforum2020">(<a href="#ref-oecdglobalscienceforum2020" role="doc-biblioref">OECD Global Science Forum 2020</a>)</span></p></li>
</ul></section><section id="how-to-achieve-reproducibility" class="level2"><h2 class="anchored" data-anchor-id="how-to-achieve-reproducibility">How to achieve reproducibility</h2>
<ul>
<li><p>Scripting</p></li>
<li><p>Organisation: Project-oriented workflows with file and folder structure, naming things</p></li>
<li><p>Documentation: Comment your code.</p></li>
<li><p>Critical curation of your project and code</p></li>
</ul></section></section><section id="project-oriented-workflow" class="level1"><h1>Project-oriented workflow</h1>
<ul>
<li><p>use folders to organise your work</p></li>
<li><p>you are aiming for structured, systematic and repeatable.</p></li>
<li><p>inputs and outputs should be clearly identifiable from structure and/or naming</p></li>
</ul>
<p>Example</p>
<pre><code>-- stem-cells
   |__stem-cells.Rproj
   |__analysis.R
   |__data-raw
      |__2019-03-21_donor_1.csv
      |__2019-03-21_donor_2.csv
      |__2019-03-21_donor_3.csv
   |__figures
      |__01_volcano_donor_1_vs_donor_2.png
      |__02_volcano_donor_1_vs_donor_3.png</code></pre>
</section><section id="naming-things" class="level1"><h1>Naming things</h1>
<ul>
<li><p>systematic and consistent</p></li>
<li><p>informative</p></li>
<li><p>use same character (e.g., <code>_</code> or <code>-</code>) for separating parts of the information</p></li>
<li><p>machine readable</p></li>
<li><p>human readable</p></li>
<li><p>play nicely with sorting</p></li>
</ul>
<p>I suggest</p>
<ul>
<li><p>no spaces in names</p></li>
<li><p>use snake_case or kebab-case rather than CamelCase or dot.case</p></li>
<li><p>use all lower case</p></li>
<li><p>ordering: use left-padded numbers e.g., 01, 02‚Ä¶.99 or 001, 002‚Ä¶.999</p></li>
<li><p>dates <a href="https://en.wikipedia.org/wiki/ISO_8601">ISO 8601</a> format: 2020-10-16</p></li>
</ul>
<p>Example</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ECOLIGreen_ISOTYPE.fcs" "ECOLIGreen_TNFAPC.fcs"  "LPS_ISOTYPE.fcs"       
[4] "LPS_TNFAPC.fcs"         "MEDIA_ISOTYPE.fcs"      "MEDIA_TNFAPC.fcs"      </code></pre>
</div>
</div>
</section><section id="comment-your-code" class="level1"><h1>Comment your code</h1>
<p>Effective commenting is crucial for making your analysis reproducible and understandable. Good commenting is about quality, not quantity. You do not need to teach a person to program in the comments, but you should explain the logic of the code and the decisions you make based on what you see in your analysis. If you remove the code, the comments should still make sense. You need not interpret the results in any depth, but you should comment on interpretation where it impacts what you do in the code.</p>
<p>Comments should</p>
<ul>
<li><p>Give an overview of the work and analysis. This should give enough information for someone to understand what the analysis is about. It should be concise and only give information needed to understand the data and analysis - you don‚Äôt need biological information unless it is essential to understand the analysis. - Structure the code into sections with section headers</p></li>
<li><p>Say what was done where it is not obvious</p></li>
<li><p>Explain the reasoning behind analytical and technical decisions relevant to the direction of the analysis.</p></li>
</ul>
<section id="example-comments" class="level2"><h2 class="anchored" data-anchor-id="example-comments">Example comments</h2>
<p><strong>Bad example</strong> No structure. Comments are self-explanatory or try to explain R (variable names are also meaningless).</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Load the readr package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org">readr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Read the data file</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_csv</a></span><span class="op">(</span><span class="st">"experiment_data.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Make a new variable named average_value using the standard R </span></span>
<span><span class="co"># assignment operator (&lt;-) to store the result. We use the mean() </span></span>
<span><span class="co"># function, which is used to calculate the arithmetic average of </span></span>
<span><span class="co"># a numeric vector. Inside the function parentheses, we give the </span></span>
<span><span class="co"># data frame object called data and use the dollar sign ($) to </span></span>
<span><span class="co"># specifically select the column named measurement. </span></span>
<span><span class="va">average_value</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">measurement</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p><strong>Good example</strong> Structure. We say why we do things and what the code is for.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Packages -----------------------------------------------------------</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">flowCore</span><span class="op">)</span>      <span class="co"># for importing flow cytometry format (FCS) </span></span>
<span>                       <span class="co"># files. Analysis is carried out with tidyverse</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span>     <span class="co"># for importing, processing and plotting</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://sjmgarnier.github.io/viridisLite/">viridisLite</a></span><span class="op">)</span>   <span class="co"># for colour scales on Figure 2 only</span></span>
<span></span>
<span><span class="co"># Import -------------------------------------------------------------</span></span>
<span></span>
<span><span class="co"># List all 6 FCS files in the data-raw folder</span></span>
<span><span class="va">myfiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="st">"data-raw"</span>, pattern <span class="op">=</span> <span class="st">".fcs$"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Read the FCS files into a flowSet object (6 flowFrames)</span></span>
<span><span class="va">fs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/flowCore/man/read.flowSet.html">read.flowSet</a></span><span class="op">(</span><span class="va">myfiles</span>, </span>
<span>                   path <span class="op">=</span> <span class="st">"data-raw"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Reformat -----------------------------------------------------------</span></span>
<span></span>
<span><span class="co"># Put flowFrames into a single data frame for ease of use in tidyverse</span></span>
<span><span class="co"># workflows</span></span>
<span><span class="va">macrophages</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/flowCore/man/fsApply.html">fsApply</a></span><span class="op">(</span><span class="va">fs</span>, <span class="va">exprs</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section></section><section id="critical-curation" class="level1"><h1>Critical Curation</h1>
<ul>
<li>Regularly review your project</li>
<li>Remove unnecessary files</li>
<li>Ensure your code is clear and well commented</li>
</ul>
<p>You might need to:</p>
<ul>
<li>collect together library statements at the top of the script</li>
<li>edit your comments for clarity</li>
<li>rename variables for consistency or clarity</li>
<li>remove house keeping or exploratory code or mark it for later removal</li>
<li>restyle code, add code section headers etc</li>
</ul></section><section id="generative-ai-for-the-rstudio-project" class="level1"><h1>Generative AI for the RStudio Project</h1>
<p><a href="https://gemini.google.com/app">Google Gemini</a> is the University‚Äôs preferred generative AI, but <a href="https://claude.ai/new">Claude AI</a> is also good.</p>
<p>Each time you use a generative AI tool, you should consider whether generative AI is the right tool.</p>
<ul>
<li>You want to avoid <strong>false authorship</strong> or anything else that is against the University‚Äôs guidelines on using AI tools (<a href="https://www.york.ac.uk/students/studying/assessment-and-examination/ai/">student guidance on using AI and translation tools</a>)</li>
<li>You might be able to do the task quicker without generative AI by reading workshop materials because you won‚Äôt have to evaluate and edit so much.</li>
</ul>
<table class="caption-top table">
<thead><tr class="header">
<th>This is OK</th>
<th>This is NOT ok</th>
</tr></thead>
<tbody><tr class="odd">
<td>
<p>It is OK to ask an AI to <strong>explain</strong> code and coding concepts.</p>
<p>It is OK to use an AI to <strong>modify</strong> your code but you must understand the inputs, the outputs and what the code does. This is important because AIs like to add things you don‚Äôt need and the marking scheme takes account relevance.</p>
<p>Do <a href="https://%20sub%20jectg%20uides.york.ac.uk/referen%20%20c%20ing-style-guides/generative-ai">Acknowledge and re ference</a> generative AI (you should also acknowledge and reference other sources of code such as workshop, text books, online tutorials)</p>
</td>
<td>
<p>Do not use an AI to <strong>generate</strong> code</p>
<p>Do not <strong>uncritically accept</strong> code modifications without evaluating their accuracy or relevance.</p>
<p>Do not use an AI to <strong>generate</strong> comments.</p>
</td>
</tr></tbody>
</table></section><section id="big-data" class="level1"><h1>Big data</h1>
<p>People argue about what quantity of data constitutes big data. In my view, it is relative to what you are used to and it is not just about the size of the data but also the complexity and the approach to analysis.</p>
<p>You have been used to working with data that has a few explanatory variables, one or two response variables 10 to 100 observations. Typically your approach has been to test one or two specific hypotheses. There is usually a very direct relationship between the experiment you design, the data you collect and the hypotheses you test.</p>
<p>Example: You want to know the effect of an antimicrobial agent on the growth of a bacterium. You design an experiment to measure the growth of the bacterium in the presence and absence of the antimicrobial agent. You collect the data and test the hypothesis that the growth of the bacterium is different in the presence and absence of the antimicrobial agent. You have a dataset with one response variable (growth) and one explanatory variable (presence or absence of the antimicrobial agent) and n observations (replicates of the experiment).</p>
<p>When working with big data we often mean:</p>
<ul>
<li><p>having many variables (100s or 1000s), many observations or both</p></li>
<li><p>visualising patterns in the data such as the relationship between variables or observations using ‚Äúdata reduction‚Äù techniques such as Principle Components Analysis (PCA)</p></li>
<li><p>applying hundreds or thousands to tests to identify patterns or relationships amongst the significant results rather than drawing conclusions from individual tests</p></li>
</ul>
<p>Example: You want to know the effect of a drug on the expression of genes in a cell. You design an experiment to measure the expression of <em>all</em> the genes in the cell in the presence and absence of the drug. You collect the data and test the hypothesis that the expression of any gene is different in the presence and absence of the drug. You then look for patterns in the data such as groups of genes that are co-expressed and the relationship between the expression of genes and the effect of the drug. You have a dataset with many response variables (expression of genes) and one explanatory variable (presence or absence of the drug) and n observations (replicates of the experiment).</p>
</section><section id="exercises" class="level1"><h1>Exercises</h1>
<section id="set-up-a-project" class="level2"><h2 class="anchored" data-anchor-id="set-up-a-project">Set up a Project</h2>
<p>üé¨ Start RStudio from the Start menu</p>
<p>üé¨ Make an RStudio project. Be deliberate about where you create it so that it is a good place for you</p>
<p>üé¨ Use the Files pane to make a folder for the data. I suggest <code>data-raw/</code></p>
<p>üé¨ Make a new script called <code>core-data-analysis.R</code> to carry out the rest of the work.</p>
</section><section id="load-packages" class="level2"><h2 class="anchored" data-anchor-id="load-packages">Load packages</h2>
<p>We need the following packages for this workshop:</p>
<ul>
<li>
<strong><code>tidyverse</code></strong> <span class="citation" data-cites="tidyverse">(<a href="#ref-tidyverse" role="doc-biblioref">Wickham et al. 2019</a>)</span>: importing the meta data which is in a text file, working with the data once it is in a dataframe to filter, summarise and plot.</li>
</ul>
<p>üé¨ Load <strong><code>tidyverse</code></strong></p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="look-at-the-data" class="level2"><h2 class="anchored" data-anchor-id="look-at-the-data">Look at the data!</h2>
<p>It is almost always useful to take a look at your data (or at least think about its format) before you attempt to read it in.</p>
<p>We will be working with the following data files in this workshop</p>
<ul>
<li><p><a href="data-raw/biotech-cts.txt">biotech-cts.txt</a>. These are RNA-Seq data from three replicates for each of two of wheat varieties (<u>S</u>usceptible and <u>T</u>olerant) grown at two potassium conditions (<u>C</u>ontrol <u>K</u> and <u>L</u>ow <u>K</u>). The data are counts of the number of reads.</p></li>
<li><p><a href="data-raw/cell-bio.tsv">cell-bio.tsv</a>. These are measurements from a Livecyte microscope, which performs live cell imaging, tracking individual cells, and measuring cell shape and size parameters for thousands of cells as they grow and divide. Measurements are taken for three replicates from each of two cell lines A and B.</p></li>
<li><p><a href="data-raw/immuno.csv">immuno.csv</a>. These are flow cytometry data for three treatments with each of two antibodies. The measures are Forward scatter, side scatter, red fluorescence and green fluorescence.</p></li>
<li><p><a href="data-raw/xlaevis_counts_S20.csv">xlaevis_counts_S20.csv</a>. These are RNA-Seq data from frogs. There are 2 siblings from each of three fertilisations and one sibling was treated with FGF and the other was a control. The data are counts of the number of reads.</p></li>
</ul>
<p>üé¨ Save these data files to the <code>data-raw/</code> folder:</p>
<p>üé¨ Consider the file extensions. What you do think the extensions indicate?</p>
<!-- #---THINKING ANSWER--- -->
<!-- txt usually indicates a plain text file, where the columns are  -->
<!-- often separated by a space but might also be separated by a tab,  -->
<!-- a backslash or forward slash, or some other character.  -->
<!-- Plain text files can be opened in notepad or other similar -->
<!-- editor and still be readable. -->
<!-- csv stands for comma separated values and tsv stands for tab  -->
<!-- separated values. Both of these are plain text format. -->
<p>üé¨ Go to the Files pane click on the <code>xlaevis_counts_S20.csv</code> file and choose View File<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="images/rstudio-filepane-viewfile.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="RStudio Files Pane"><img src="images/rstudio-filepane-viewfile.png" class="img-fluid figure-img" alt="Rstudio Files pane showing the data files and the View File option that appears when you click on the a particular file"></a></p>
<figcaption>RStudio Files Pane</figcaption></figure>
</div>
<p>Any plain text file will open in the top left pane. Note that this is NOT importing the data, it‚Äôs just viewing the file.</p>
<p>üé¨ Does it seem to be a csv file?</p>
<p>üé¨ Try opening the other files from the Files pane. What happens?</p>
<p>üé¨ Open each file in Excel. For the <code>.txt</code> and <code>.tsv</code> files you will need to show ‚ÄúAll files (<em>.</em>)‚Äù in the file type dropdown to see them and you will need to work through the ‚ÄúText Import Wizard‚Äù to open them.</p>
<p>We are opening them each in Excel to see what they look like and to get a sense of the structure of the data. We are not going to use Excel to do anything.</p>
<p>üé¨ What do you notice? Does it look like this data will be easy to import and work with?</p>
</section><section id="import-the-data" class="level2"><h2 class="anchored" data-anchor-id="import-the-data">Import the data</h2>
<p>üé¨ Import each of the data files into R:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">biotech</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_tsv</a></span><span class="op">(</span><span class="st">"data-raw/biotech-cts.txt"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cell_bio</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_tsv</a></span><span class="op">(</span><span class="st">"data-raw/cell-bio.tsv"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">immuno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_csv</a></span><span class="op">(</span><span class="st">"data-raw/immuno.csv"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">frogs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_csv</a></span><span class="op">(</span><span class="st">"data-raw/xlaevis_counts_S20.csv"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>üé¨ Click on each of the dataframes in the Environment pane to see what they look like. An alternative to clicking is to use the <code><a href="https://rdrr.io/r/utils/View.html">View()</a></code> function:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/View.html">View</a></span><span class="op">(</span><span class="va">cell_bio</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>üé¨ In each case, describe what is in the rows and the columns of the dataframe. Where are the replicates/samples and where are the variables? Are there any things you‚Äôd like to fix/change?</p>
<!-- #---THINKING ANSWER--- -->
<p>üé¨ Name of the first column in the <code>biotech</code> dataframe:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">biotech</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"transcript"</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>üé¨ Column names in the <code>cell_bio</code> dataframe:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cell_bio</span> <span class="op">&lt;-</span> <span class="fu">janitor</span><span class="fu">::</span><span class="fu"><a href="https://sfirke.github.io/janitor/reference/clean_names.html">clean_names</a></span><span class="op">(</span><span class="va">cell_bio</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="getting-an-overview-1" class="level2"><h2 class="anchored" data-anchor-id="getting-an-overview-1">Getting an overview 1</h2>
<section id="summary" class="level3"><h3 class="anchored" data-anchor-id="summary"><code>summary()</code></h3>
<p>R‚Äôs <code><a href="https://rdrr.io/r/base/summary.html">summary()</a></code> function is a quick way to get an overview of datasets. It gives you a six number summary for every numeric variable (the minimum, lower quartile, median, mean, upper quartile, and maximum). It also gives you a count of the number of missing values.</p>
<p>üé¨ Use the <code><a href="https://rdrr.io/r/base/summary.html">summary()</a></code> function to get an overview of each dataframe. It works well for these datasets. What types of dataset might this be less useful for? Which dataframes and columns have missing values? Which dataframes and variables seem very skewed? Which dataframe has more than one character variable?</p>
<!-- #---THINKING ANSWER--- -->
<!-- It is less useful for datasets with very many variables  -->
<!-- (columns) because we aren't able to take that in all  -->
<!-- at once. It fills up the screen buffer and we can't -->
<!-- see all the output at once. -->
<!-- It is also less useful for datasets with many  -->
<!-- character variables -->
<!-- cell_bio has a perimeter column with 19948 NAs -->
<!-- skew is revealed by the difference between the mean and  -->
<!-- the median and a maximum that is much larger than all  -->
<!-- the other values. The frogs and biotech dataframes have -->
<!-- skewed variables. Skew is typical in RNA-Seq counts. -->
<!-- immuno is the only dataset with more than one character -->
<!-- variable (it has three) -->
</section></section><section id="quality-control" class="level2"><h2 class="anchored" data-anchor-id="quality-control">Quality Control</h2>
<section id="filtering-rows" class="level3"><h3 class="anchored" data-anchor-id="filtering-rows">Filtering rows</h3>
<p>The filter function selects/drops a whole row on a condition. The condition can be based on the value in one, or a combination, of columns. We specify what we want to keep with the <code><a href="https://dplyr.tidyverse.org/reference/filter.html">filter()</a></code> function. This means you might often want to negate a condition with <code>!</code>.</p>
<p>üé¨ To remove the rows in <code>cell_bio</code> with missing values in the <code>perimeter</code> column:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cell_bio</span> <span class="op">&lt;-</span> <span class="va">cell_bio</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">perimeter</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p><code>is.na(perimeter)</code> returns a logical vector (a vector of <code>TRUE</code>s and <code>FALSE</code>s) of the same length as the column. The <code>!</code> negates the logical vector so the <code>TRUE</code>s become <code>FALSE</code>s and the <code>FALSE</code>s become TRUE. This means that the <code><a href="https://dplyr.tidyverse.org/reference/filter.html">filter()</a></code> function <em>keeps</em> the rows where the perimeter column is not missing.</p>
<p>üé¨ There‚Äôs actually a tidyverse function that does the same thing as <code>filter(!is.na())</code></p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cell_bio</span> <span class="op">&lt;-</span> <span class="va">cell_bio</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://tidyr.tidyverse.org/reference/drop_na.html">drop_na</a></span><span class="op">(</span><span class="va">perimeter</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Sometimes you might want to work with a subset of data. For example, you might want to examine just the Media treated cells in the <code>immuno</code> dataset.</p>
<p>üé¨ Create a new dataframe called <code>immuno_media</code> that contains only the rows where <code>treatment</code> is ‚ÄúMEDIA‚Äù</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">immuno_media</span> <span class="op">&lt;-</span> <span class="va">immuno</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">treatment</span> <span class="op">==</span> <span class="st">"MEDIA"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Or just the Media treated cells with a <code>FS_Lin</code> between two values. You can apply two filters and the <code><a href="https://dplyr.tidyverse.org/reference/between.html">between()</a></code> function to achieve this:</p>
<p>üé¨ Create a new dataframe called <code>immuno_media_live</code> that contains only the rows where <code>treatment</code> is ‚ÄúMEDIA‚Äù and <code>FS_Lin</code> is between 7500 and 28000:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">immuno_media_live</span> <span class="op">&lt;-</span> <span class="va">immuno</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">treatment</span> <span class="op">==</span> <span class="st">"MEDIA"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/between.html">between</a></span><span class="op">(</span><span class="va">FS_Lin</span>, <span class="fl">7500</span>, <span class="fl">28000</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>üé¨ Now you try. Create a dataframe called <code>immuno_live</code> that contains only the rows where <code>FS_Lin</code> is between 7500 and 28000 and <code>SS_Lin</code> is between 15000 and 35000:</p>
</section><section id="selecting-columns" class="level3"><h3 class="anchored" data-anchor-id="selecting-columns">Selecting columns</h3>
<p>Whilst we can always specify the columns we want to use when working with data, sometimes we find it less overwhelming to create a new dataframe with only the columns we want. The <code><a href="https://dplyr.tidyverse.org/reference/select.html">select()</a></code> function helps us here.</p>
<p>üé¨ Suppose we only want to work with the Susceptible varieties of wheat in the <code>biotech</code> dataframe. We can create a new dataframe with only the columns we want:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">biotech_susceptible</span> <span class="op">&lt;-</span> <span class="va">biotech</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">SCK14_1</span>,</span>
<span>         <span class="va">SCK14_2</span>,</span>
<span>         <span class="va">SCK14_3</span>,</span>
<span>         <span class="va">SLK14_1</span>,</span>
<span>         <span class="va">SLK14_2</span>,</span>
<span>         <span class="va">SLK14_3</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>In fact, there are a couple of alternatives to this. We can use the <code><a href="https://tidyselect.r-lib.org/reference/starts_with.html">starts_with()</a></code> function to select all the columns that start with a certain string:</p>
<p>üé¨ Select columns starting with <code>S</code></p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">biotech_susceptible</span> <span class="op">&lt;-</span> <span class="va">biotech</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html">starts_with</a></span><span class="op">(</span><span class="st">"S"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>There is an <code><a href="https://tidyselect.r-lib.org/reference/starts_with.html">ends_with()</a></code> function too!</p>
<p>üé¨ The colon notation allows us to select a range of columns. Select columns from <code>SCK14_1</code> to <code>SLK14_3</code></p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">biotech_susceptible</span> <span class="op">&lt;-</span> <span class="va">biotech</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">SCK14_1</span><span class="op">:</span><span class="va">SLK14_3</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Note that you need to pay attention to the order of the columns in your dataframe to use this.</p>
<p>üé¨ You can use select and filter together. Try creating a dataframe from <code>frogs</code> which has only the columns from sibling ‚Äú_A‚Äù and only the rows where <code>S20_C_5</code> is above 20.</p>
</section><section id="visualisation" class="level3"><h3 class="anchored" data-anchor-id="visualisation">Visualisation</h3>
<p>Histograms and density plots are useful for visualising the distribution of variables when we have many observations. They allow you to see features in the data that are not obvious from the summary statistics.</p>
<ul>
<li><p>A histogram shows the number of values in each range (bin), it is composed of touching bars. How a histogram looks depends on the number of bins used.</p></li>
<li><p>A density plot shows the proportions of values in a range. The appearance of the distribution is not affected by the number of bins.</p></li>
</ul>
<p>üé¨ To plot a histogram of <code>S20_C_5</code> in the <code>frogs</code> dataframe:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">frogs</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">S20_C_5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-23-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="workshop_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<p>This data is very skewed - there are so many low values that we can‚Äôt see the tiny bars for the higher values. Logging the counts is a way to make the distribution more visible.</p>
<p>üé¨ To plot the distribution of log <code>S20_C_5</code> in the <code>frogs</code> dataframe as a histogram:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">frogs</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">S20_C_5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-24-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4"><img src="workshop_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<p>üé¨ Or a density plot:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">frogs</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">S20_C_5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-25-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5"><img src="workshop_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<p>I‚Äôve used base 10 only because it is easy to convert to the original scale (1 is 10, 2 is 100, 3 is 1000 etc). The warning about rows being removed is expected - these are the counts of 0 since you can‚Äôt log a value of 0. The peak at zero suggests quite a few counts of 1. We would expect the distribution of counts to be roughly log normal because this is expression of all the genes in the genome<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>. That peak near the low end is not expected for this distribution and suggests that these lower counts might be anomalies.</p>
<p>The excess number of low counts indicates we might want to create a cut off for quality control. The removal of low counts is a common processing step in ‚Äôomic data.</p>
<p>If you have two groups of data, you can compare their distributions on the same plot using the <code>fill</code> aesthetic and making the fill transparent with the <code>alpha</code> argument.</p>
<p>üé¨ To plot the distribution of <code>area</code> in the <code>cell_bio</code> dataframe as a density plot, with the <code>clone</code> variable as the fill:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cell_bio</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">area</span>, fill <span class="op">=</span> <span class="va">clone</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-26-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6"><img src="workshop_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<p>Boxplots and violin plots are also useful for visualising the distribution of a variable, especially when there are more than about 5 groups.</p>
<p>üé¨ To plot the distribution of <code>area</code> in the <code>cell_bio</code> dataframe as a boxplot, with the <code>clone</code> variable on the x-axis:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">cell_bio</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">clone</span>, y <span class="op">=</span> <span class="va">area</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-27-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7"><img src="workshop_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<p>Suppose you want to see the distribution of many similar variables such as we have in <code>biotech</code>. You cannot map a variable to <code>x</code> or <code>fill</code> as we did in the two previous cases because the values are in <em>separate</em> columns rather than than in <em>one</em> column with a second column giving a group. However, we can use the <code><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer()</a></code> function to reformat the data and then pipe it into ggplot.</p>
<p>üé¨ Plot the distribution of the logged counts in the <code>biotech</code> dataframe as a boxplot, the samples on the x axis:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">biotech</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="op">-</span><span class="va">transcript</span>,</span>
<span>               names_to <span class="op">=</span> <span class="st">"sample"</span>,</span>
<span>               values_to <span class="op">=</span> <span class="st">"count"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">sample</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span><span class="op">(</span><span class="va">count</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-28-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8"><img src="workshop_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>This code takes the <code>biotech</code> dataframe and transforms it dataset from wide format to long format. It takes all columns except <code>transcript</code> (<code>cols = -transcript</code>) and gathers them into two new columns:</p>
<ul>
<li>
<code>"sample"</code> ‚Üí Stores the original column names (<code>SCK14_1</code>, <code>SCK14_2</code>, etc.).</li>
<li>
<code>"count"</code> ‚Üí Stores the values from those columns.</li>
</ul>
<p>Now, instead of multiple sample columns, we have a table like this:</p>
<table class="caption-top table">
<thead><tr class="header">
<th>transcript</th>
<th>sample</th>
<th>count</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>MSTRG.1</td>
<td>SCK14_1</td>
<td>560</td>
</tr>
<tr class="even">
<td>MSTRG.1</td>
<td>SCK14_2</td>
<td>1526</td>
</tr>
<tr class="odd">
<td>MSTRG.1</td>
<td>SCK14_3</td>
<td>4168</td>
</tr>
<tr class="even">
<td>MSTRG.10</td>
<td>SCK14_1</td>
<td>0</td>
</tr>
</tbody>
</table>
<p>which is piped straight into ggplot. The nice thing about the pipe, is that you need not create a new dataframe to do this. You can pipe the output of <code><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer()</a></code> directly into <code><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot()</a></code>. This is very useful when you are exploring the data and want to try out different visualisations.</p>
<p>You‚Äôre finished!</p>
</section></section></section><section id="well-done" class="level1"><h1>ü•≥ Well Done! üéâ</h1>
</section><section id="independent-study-following-the-workshop" class="level1"><h1>Independent study following the workshop</h1>
<p><a href="../../r4babs4/week-1/study_after_workshop.html">Consolidate</a></p>
</section><section id="the-code-file" class="level1"><h1>The Code file</h1>
<p>All the answers (coding and thinking) are in the source code for the workshop. You can view it using the <code>&lt;/&gt; Code</code> button at the top of the page. Coding and thinking answers are marked with <code>#---CODING ANSWER---</code> and <code>#---THINKING ANSWER---</code></p>
<p>Pages made with R <span class="citation" data-cites="R-core">(<a href="#ref-R-core" role="doc-biblioref">R Core Team 2024</a>)</span>, Quarto <span class="citation" data-cites="allaire2022">(<a href="#ref-allaire2022" role="doc-biblioref">Allaire et al. 2022</a>)</span>, <code>knitr</code> <span class="citation" data-cites="knitr1 knitr2 knitr3">(<a href="#ref-knitr1" role="doc-biblioref">Xie 2024</a>, <a href="#ref-knitr2" role="doc-biblioref">2015</a>, <a href="#ref-knitr3" role="doc-biblioref">2014</a>)</span>, <code>kableExtra</code> <span class="citation" data-cites="kableExtra">(<a href="#ref-kableExtra" role="doc-biblioref">Zhu 2024</a>)</span></p>
</section><section id="references" class="level1">


<!-- -->


</section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-allaire2022" class="csl-entry" role="listitem">
Allaire, J. J., Charles Teague, Carlos Scheidegger, Yihui Xie, and Christophe Dervieux. 2022. <em>Quarto</em>. <a href="https://doi.org/10.5281/zenodo.5960048">https://doi.org/10.5281/zenodo.5960048</a>.
</div>
<div id="ref-baggerly2009" class="csl-entry" role="listitem">
Baggerly, Keith A, and Kevin R Coombes. 2009. <span>‚ÄúDERIVING CHEMOSENSITIVITY FROM CELL LINES: FORENSIC BIOINFORMATICS AND REPRODUCIBLE RESEARCH IN HIGH-THROUGHPUT BIOLOGY.‚Äù</span> <em>Ann. Appl. Stat.</em> 3 (4): 1309‚Äì34. <a href="https://doi.org/10.2307/27801549">https://doi.org/10.2307/27801549</a>.
</div>
<div id="ref-markowetz2015" class="csl-entry" role="listitem">
Markowetz, Florian. 2015. <span>‚ÄúFive Selfish Reasons to Work Reproducibly.‚Äù</span> <em>Genome Biol.</em> 16 (December): 274. <a href="https://doi.org/10.1186/s13059-015-0850-7">https://doi.org/10.1186/s13059-015-0850-7</a>.
</div>
<div id="ref-oecdglobalscienceforum2020" class="csl-entry" role="listitem">
OECD Global Science Forum. 2020. <span>‚ÄúBuilding Digital Workforce Capacity and Skills for Data-Intensive Science.‚Äù</span> <a href="http://www.oecd.org/officialdocuments/publicdisplaydocumentpdf/?cote=DSTI/STP/GSF(2020)6/FINAL&amp;docLanguage=En">http://www.oecd.org/officialdocuments/publicdisplaydocumentpdf/?cote=DSTI/STP/GSF(2020)6/FINAL&amp;docLanguage=En</a>.
</div>
<div id="ref-R-core" class="csl-entry" role="listitem">
R Core Team. 2024. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-tidyverse" class="csl-entry" role="listitem">
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D‚ÄôAgostino McGowan, Romain Fran√ßois, Garrett Grolemund, et al. 2019. <span>‚ÄúWelcome to the <span></span>Tidyverse<span></span>‚Äù</span> 4: 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>.
</div>
<div id="ref-knitr3" class="csl-entry" role="listitem">
Xie, Yihui. 2014. <span>‚ÄúKnitr: A Comprehensive Tool for Reproducible Research in <span>R</span>.‚Äù</span> In <em>Implementing Reproducible Computational Research</em>, edited by Victoria Stodden, Friedrich Leisch, and Roger D. Peng. Chapman; Hall/CRC.
</div>
<div id="ref-knitr2" class="csl-entry" role="listitem">
‚Äî‚Äî‚Äî. 2015. <em>Dynamic Documents with <span>R</span> and Knitr</em>. 2nd ed. Boca Raton, Florida: Chapman; Hall/CRC. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
<div id="ref-knitr1" class="csl-entry" role="listitem">
‚Äî‚Äî‚Äî. 2024. <em>Knitr: A General-Purpose Package for Dynamic Report Generation in r</em>. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
<div id="ref-kableExtra" class="csl-entry" role="listitem">
Zhu, Hao. 2024. <em>kableExtra: Construct Complex Table with ‚ÄôKable‚Äô and Pipe Syntax</em>. <a href="https://CRAN.R-project.org/package=kableExtra">https://CRAN.R-project.org/package=kableExtra</a>.
</div>
</div></section><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>
<ol>
<li id="fn1"><p><strong>Do not</strong> be tempted to import data this way. Unless you are careful, your data import will not be scripted or will not be scripted correctly. This means your work will not be reproducible.<a href="#fnref1" class="footnote-back" role="doc-backlink">‚Ü©Ô∏é</a></p></li>
<li id="fn2"><p>This a result of the&nbsp;<a href="https://en.wikipedia.org/wiki/Central_limit_theorem">Central limit theorem</a>, one consequence of which is that adding together lots of distributions - whatever distributions they are - will tend to a normal distribution.<a href="#fnref2" class="footnote-back" role="doc-backlink">‚Ü©Ô∏é</a></p></li>
</ol></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "Óßã";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("^(?:http:|https:)\/\/3mmarand\.github\.io\/R4BABS");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb27" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workshop"</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Data Analysis 1: Core"</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="an">toc:</span><span class="co"> true</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="an">toc-location:</span><span class="co"> right</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> </span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="co">  markdown: </span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="co">    wrap: 72</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a><span class="fu"># Introduction</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a><span class="fu">## Session overview</span></span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>In the first part of the workshop I will talk about Project organisation</span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>for reproducible analysis and data that has many variables and</span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a>observations. In the second part of the workshop you will practice</span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>getting an overview of such data with summaries and distribution plots,</span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>filtering rows and columns.</span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a><span class="fu"># Reproducibility</span></span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a><span class="fu">## Why does it matter?</span></span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a>![futureself, CC-BY-NC, by Julen</span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a>Colomb](images/future_you.png){fig-alt="Person working at a computer with an offstage person asking 'How is the analysis going?' The person at the computer replies 'Can't understand the date...and the data collector does not answer my emails or calls' Person offstage: 'That's terrible! So cruel! Who did collect the data? I will sack them!' Person at the computer: 'um...I did, 3 years ago.'"</span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>width="400"}</span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Five selfish reasons to work reproducibly <span class="co">[</span><span class="ot">@markowetz2015</span><span class="co">]</span>.</span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a>    Alternatively, see the very entertaining</span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a>    <span class="co">[</span><span class="ot">talk</span><span class="co">](https://youtu.be/yVT07Sukv9Q)</span> which covers the "Duke</span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a>    Scandal".</span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Many high profile cases of work which did not reproduce e.g. Anil</span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a>    Potti's work unravelled by @baggerly2009 in the "Duke Scandal"</span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-45"><a href="#cb27-45" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Will** become standard in Science and publishing e.g OECD Global</span>
<span id="cb27-46"><a href="#cb27-46" aria-hidden="true" tabindex="-1"></a>    Science Forum Building digital workforce capacity and skills for</span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a>    data-intensive science <span class="co">[</span><span class="ot">@oecdglobalscienceforum2020</span><span class="co">]</span></span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a><span class="fu">## How to achieve reproducibility</span></span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Scripting</span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Organisation: Project-oriented workflows with file and folder</span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a>    structure, naming things</span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Documentation: Comment your code.</span>
<span id="cb27-57"><a href="#cb27-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-58"><a href="#cb27-58" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Critical curation of your project and code</span>
<span id="cb27-59"><a href="#cb27-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-60"><a href="#cb27-60" aria-hidden="true" tabindex="-1"></a><span class="fu"># Project-oriented workflow</span></span>
<span id="cb27-61"><a href="#cb27-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-62"><a href="#cb27-62" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>use folders to organise your work</span>
<span id="cb27-63"><a href="#cb27-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-64"><a href="#cb27-64" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>you are aiming for structured, systematic and repeatable.</span>
<span id="cb27-65"><a href="#cb27-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-66"><a href="#cb27-66" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>inputs and outputs should be clearly identifiable from structure</span>
<span id="cb27-67"><a href="#cb27-67" aria-hidden="true" tabindex="-1"></a>    and/or naming</span>
<span id="cb27-68"><a href="#cb27-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-69"><a href="#cb27-69" aria-hidden="true" tabindex="-1"></a>Example</span>
<span id="cb27-70"><a href="#cb27-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-71"><a href="#cb27-71" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb27-72"><a href="#cb27-72" aria-hidden="true" tabindex="-1"></a><span class="in">-- stem-cells</span></span>
<span id="cb27-73"><a href="#cb27-73" aria-hidden="true" tabindex="-1"></a><span class="in">   |__stem-cells.Rproj</span></span>
<span id="cb27-74"><a href="#cb27-74" aria-hidden="true" tabindex="-1"></a><span class="in">   |__analysis.R</span></span>
<span id="cb27-75"><a href="#cb27-75" aria-hidden="true" tabindex="-1"></a><span class="in">   |__data-raw</span></span>
<span id="cb27-76"><a href="#cb27-76" aria-hidden="true" tabindex="-1"></a><span class="in">      |__2019-03-21_donor_1.csv</span></span>
<span id="cb27-77"><a href="#cb27-77" aria-hidden="true" tabindex="-1"></a><span class="in">      |__2019-03-21_donor_2.csv</span></span>
<span id="cb27-78"><a href="#cb27-78" aria-hidden="true" tabindex="-1"></a><span class="in">      |__2019-03-21_donor_3.csv</span></span>
<span id="cb27-79"><a href="#cb27-79" aria-hidden="true" tabindex="-1"></a><span class="in">   |__figures</span></span>
<span id="cb27-80"><a href="#cb27-80" aria-hidden="true" tabindex="-1"></a><span class="in">      |__01_volcano_donor_1_vs_donor_2.png</span></span>
<span id="cb27-81"><a href="#cb27-81" aria-hidden="true" tabindex="-1"></a><span class="in">      |__02_volcano_donor_1_vs_donor_3.png</span></span>
<span id="cb27-82"><a href="#cb27-82" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-83"><a href="#cb27-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-84"><a href="#cb27-84" aria-hidden="true" tabindex="-1"></a><span class="fu"># Naming things</span></span>
<span id="cb27-85"><a href="#cb27-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-86"><a href="#cb27-86" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>systematic and consistent</span>
<span id="cb27-87"><a href="#cb27-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-88"><a href="#cb27-88" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>informative</span>
<span id="cb27-89"><a href="#cb27-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-90"><a href="#cb27-90" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>use same character (e.g., <span class="in">`_`</span> or <span class="in">`-`</span>) for separating parts of the</span>
<span id="cb27-91"><a href="#cb27-91" aria-hidden="true" tabindex="-1"></a>    information</span>
<span id="cb27-92"><a href="#cb27-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-93"><a href="#cb27-93" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>machine readable</span>
<span id="cb27-94"><a href="#cb27-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-95"><a href="#cb27-95" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>human readable</span>
<span id="cb27-96"><a href="#cb27-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-97"><a href="#cb27-97" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>play nicely with sorting</span>
<span id="cb27-98"><a href="#cb27-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-99"><a href="#cb27-99" aria-hidden="true" tabindex="-1"></a>I suggest</span>
<span id="cb27-100"><a href="#cb27-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-101"><a href="#cb27-101" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>no spaces in names</span>
<span id="cb27-102"><a href="#cb27-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-103"><a href="#cb27-103" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>use snake_case or kebab-case rather than CamelCase or dot.case</span>
<span id="cb27-104"><a href="#cb27-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-105"><a href="#cb27-105" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>use all lower case</span>
<span id="cb27-106"><a href="#cb27-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-107"><a href="#cb27-107" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>ordering: use left-padded numbers e.g., 01, 02....99 or 001,</span>
<span id="cb27-108"><a href="#cb27-108" aria-hidden="true" tabindex="-1"></a>    002....999</span>
<span id="cb27-109"><a href="#cb27-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-110"><a href="#cb27-110" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>dates <span class="co">[</span><span class="ot">ISO 8601</span><span class="co">](https://en.wikipedia.org/wiki/ISO_8601)</span> format:</span>
<span id="cb27-111"><a href="#cb27-111" aria-hidden="true" tabindex="-1"></a>    2020-10-16</span>
<span id="cb27-112"><a href="#cb27-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-113"><a href="#cb27-113" aria-hidden="true" tabindex="-1"></a>Example</span>
<span id="cb27-114"><a href="#cb27-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-117"><a href="#cb27-117" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-118"><a href="#cb27-118" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb27-119"><a href="#cb27-119" aria-hidden="true" tabindex="-1"></a><span class="fu">dir</span>(<span class="st">"../week-2/data-raw"</span>)</span>
<span id="cb27-120"><a href="#cb27-120" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-121"><a href="#cb27-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-122"><a href="#cb27-122" aria-hidden="true" tabindex="-1"></a><span class="fu"># Comment your code</span></span>
<span id="cb27-123"><a href="#cb27-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-124"><a href="#cb27-124" aria-hidden="true" tabindex="-1"></a>Effective commenting is crucial for making your analysis reproducible</span>
<span id="cb27-125"><a href="#cb27-125" aria-hidden="true" tabindex="-1"></a>and understandable. Good commenting is about quality, not quantity. You</span>
<span id="cb27-126"><a href="#cb27-126" aria-hidden="true" tabindex="-1"></a>do not need to teach a person to program in the comments, but you should</span>
<span id="cb27-127"><a href="#cb27-127" aria-hidden="true" tabindex="-1"></a>explain the logic of the code and the decisions you make based on what</span>
<span id="cb27-128"><a href="#cb27-128" aria-hidden="true" tabindex="-1"></a>you see in your analysis. If you remove the code, the comments should</span>
<span id="cb27-129"><a href="#cb27-129" aria-hidden="true" tabindex="-1"></a>still make sense. You need not interpret the results in any depth, but</span>
<span id="cb27-130"><a href="#cb27-130" aria-hidden="true" tabindex="-1"></a>you should comment on interpretation where it impacts what you do in the</span>
<span id="cb27-131"><a href="#cb27-131" aria-hidden="true" tabindex="-1"></a>code.</span>
<span id="cb27-132"><a href="#cb27-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-133"><a href="#cb27-133" aria-hidden="true" tabindex="-1"></a>Comments should</span>
<span id="cb27-134"><a href="#cb27-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-135"><a href="#cb27-135" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Give an overview of the work and analysis. This should give enough</span>
<span id="cb27-136"><a href="#cb27-136" aria-hidden="true" tabindex="-1"></a>    information for someone to understand what the analysis is about. It</span>
<span id="cb27-137"><a href="#cb27-137" aria-hidden="true" tabindex="-1"></a>    should be concise and only give information needed to understand the</span>
<span id="cb27-138"><a href="#cb27-138" aria-hidden="true" tabindex="-1"></a>    data and analysis - you don't need biological information unless it</span>
<span id="cb27-139"><a href="#cb27-139" aria-hidden="true" tabindex="-1"></a>    is essential to understand the analysis. - Structure the code into</span>
<span id="cb27-140"><a href="#cb27-140" aria-hidden="true" tabindex="-1"></a>    sections with section headers</span>
<span id="cb27-141"><a href="#cb27-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-142"><a href="#cb27-142" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Say what was done where it is not obvious</span>
<span id="cb27-143"><a href="#cb27-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-144"><a href="#cb27-144" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Explain the reasoning behind analytical and technical decisions</span>
<span id="cb27-145"><a href="#cb27-145" aria-hidden="true" tabindex="-1"></a>    relevant to the direction of the analysis.</span>
<span id="cb27-146"><a href="#cb27-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-147"><a href="#cb27-147" aria-hidden="true" tabindex="-1"></a><span class="fu">## Example comments</span></span>
<span id="cb27-148"><a href="#cb27-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-149"><a href="#cb27-149" aria-hidden="true" tabindex="-1"></a>**Bad example** No structure. Comments are self-explanatory or try to</span>
<span id="cb27-150"><a href="#cb27-150" aria-hidden="true" tabindex="-1"></a>explain R (variable names are also meaningless).</span>
<span id="cb27-151"><a href="#cb27-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-154"><a href="#cb27-154" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-155"><a href="#cb27-155" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb27-156"><a href="#cb27-156" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the readr package</span></span>
<span id="cb27-157"><a href="#cb27-157" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb27-158"><a href="#cb27-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-159"><a href="#cb27-159" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the data file</span></span>
<span id="cb27-160"><a href="#cb27-160" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"experiment_data.csv"</span>)</span>
<span id="cb27-161"><a href="#cb27-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-162"><a href="#cb27-162" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a new variable named average_value using the standard R </span></span>
<span id="cb27-163"><a href="#cb27-163" aria-hidden="true" tabindex="-1"></a><span class="co"># assignment operator (&lt;-) to store the result. We use the mean() </span></span>
<span id="cb27-164"><a href="#cb27-164" aria-hidden="true" tabindex="-1"></a><span class="co"># function, which is used to calculate the arithmetic average of </span></span>
<span id="cb27-165"><a href="#cb27-165" aria-hidden="true" tabindex="-1"></a><span class="co"># a numeric vector. Inside the function parentheses, we give the </span></span>
<span id="cb27-166"><a href="#cb27-166" aria-hidden="true" tabindex="-1"></a><span class="co"># data frame object called data and use the dollar sign ($) to </span></span>
<span id="cb27-167"><a href="#cb27-167" aria-hidden="true" tabindex="-1"></a><span class="co"># specifically select the column named measurement. </span></span>
<span id="cb27-168"><a href="#cb27-168" aria-hidden="true" tabindex="-1"></a>average_value <span class="ot">&lt;-</span> <span class="fu">mean</span>(data<span class="sc">$</span>measurement)</span>
<span id="cb27-169"><a href="#cb27-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-170"><a href="#cb27-170" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-171"><a href="#cb27-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-172"><a href="#cb27-172" aria-hidden="true" tabindex="-1"></a>**Good example** Structure. We say why we do things and what the code is</span>
<span id="cb27-173"><a href="#cb27-173" aria-hidden="true" tabindex="-1"></a>for.</span>
<span id="cb27-174"><a href="#cb27-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-177"><a href="#cb27-177" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-178"><a href="#cb27-178" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb27-179"><a href="#cb27-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-180"><a href="#cb27-180" aria-hidden="true" tabindex="-1"></a><span class="co"># Packages -----------------------------------------------------------</span></span>
<span id="cb27-181"><a href="#cb27-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-182"><a href="#cb27-182" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(flowCore)      <span class="co"># for importing flow cytometry format (FCS) </span></span>
<span id="cb27-183"><a href="#cb27-183" aria-hidden="true" tabindex="-1"></a>                       <span class="co"># files. Analysis is carried out with tidyverse</span></span>
<span id="cb27-184"><a href="#cb27-184" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)     <span class="co"># for importing, processing and plotting</span></span>
<span id="cb27-185"><a href="#cb27-185" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridisLite)   <span class="co"># for colour scales on Figure 2 only</span></span>
<span id="cb27-186"><a href="#cb27-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-187"><a href="#cb27-187" aria-hidden="true" tabindex="-1"></a><span class="co"># Import -------------------------------------------------------------</span></span>
<span id="cb27-188"><a href="#cb27-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-189"><a href="#cb27-189" aria-hidden="true" tabindex="-1"></a><span class="co"># List all 6 FCS files in the data-raw folder</span></span>
<span id="cb27-190"><a href="#cb27-190" aria-hidden="true" tabindex="-1"></a>myfiles <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">"data-raw"</span>, <span class="at">pattern =</span> <span class="st">".fcs$"</span>)</span>
<span id="cb27-191"><a href="#cb27-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-192"><a href="#cb27-192" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the FCS files into a flowSet object (6 flowFrames)</span></span>
<span id="cb27-193"><a href="#cb27-193" aria-hidden="true" tabindex="-1"></a>fs <span class="ot">&lt;-</span> <span class="fu">read.flowSet</span>(myfiles, </span>
<span id="cb27-194"><a href="#cb27-194" aria-hidden="true" tabindex="-1"></a>                   <span class="at">path =</span> <span class="st">"data-raw"</span>)</span>
<span id="cb27-195"><a href="#cb27-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-196"><a href="#cb27-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-197"><a href="#cb27-197" aria-hidden="true" tabindex="-1"></a><span class="co"># Reformat -----------------------------------------------------------</span></span>
<span id="cb27-198"><a href="#cb27-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-199"><a href="#cb27-199" aria-hidden="true" tabindex="-1"></a><span class="co"># Put flowFrames into a single data frame for ease of use in tidyverse</span></span>
<span id="cb27-200"><a href="#cb27-200" aria-hidden="true" tabindex="-1"></a><span class="co"># workflows</span></span>
<span id="cb27-201"><a href="#cb27-201" aria-hidden="true" tabindex="-1"></a>macrophages <span class="ot">&lt;-</span> <span class="fu">fsApply</span>(fs, exprs) <span class="sc">|&gt;</span> <span class="fu">data.frame</span>() </span>
<span id="cb27-202"><a href="#cb27-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-203"><a href="#cb27-203" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-204"><a href="#cb27-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-207"><a href="#cb27-207" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-208"><a href="#cb27-208" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb27-209"><a href="#cb27-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-210"><a href="#cb27-210" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-211"><a href="#cb27-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-212"><a href="#cb27-212" aria-hidden="true" tabindex="-1"></a><span class="fu"># Critical Curation</span></span>
<span id="cb27-213"><a href="#cb27-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-214"><a href="#cb27-214" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Regularly review your project</span>
<span id="cb27-215"><a href="#cb27-215" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Remove unnecessary files</span>
<span id="cb27-216"><a href="#cb27-216" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Ensure your code is clear and well commented</span>
<span id="cb27-217"><a href="#cb27-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-218"><a href="#cb27-218" aria-hidden="true" tabindex="-1"></a>You might need to:</span>
<span id="cb27-219"><a href="#cb27-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-220"><a href="#cb27-220" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>collect together library statements at the top of the script</span>
<span id="cb27-221"><a href="#cb27-221" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>edit your comments for clarity</span>
<span id="cb27-222"><a href="#cb27-222" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>rename variables for consistency or clarity</span>
<span id="cb27-223"><a href="#cb27-223" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>remove house keeping or exploratory code or mark it for later</span>
<span id="cb27-224"><a href="#cb27-224" aria-hidden="true" tabindex="-1"></a>    removal</span>
<span id="cb27-225"><a href="#cb27-225" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>restyle code, add code section headers etc</span>
<span id="cb27-226"><a href="#cb27-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-227"><a href="#cb27-227" aria-hidden="true" tabindex="-1"></a><span class="fu"># Generative AI for the RStudio Project</span></span>
<span id="cb27-228"><a href="#cb27-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-229"><a href="#cb27-229" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">Google Gemini</span><span class="co">](https://gemini.google.com/app)</span> is the University's</span>
<span id="cb27-230"><a href="#cb27-230" aria-hidden="true" tabindex="-1"></a>preferred generative AI, but <span class="co">[</span><span class="ot">Claude AI</span><span class="co">](https://claude.ai/new)</span> is also</span>
<span id="cb27-231"><a href="#cb27-231" aria-hidden="true" tabindex="-1"></a>good.</span>
<span id="cb27-232"><a href="#cb27-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-233"><a href="#cb27-233" aria-hidden="true" tabindex="-1"></a>Each time you use a generative AI tool, you should consider whether</span>
<span id="cb27-234"><a href="#cb27-234" aria-hidden="true" tabindex="-1"></a>generative AI is the right tool.</span>
<span id="cb27-235"><a href="#cb27-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-236"><a href="#cb27-236" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>You want to avoid **false authorship** or anything else that is</span>
<span id="cb27-237"><a href="#cb27-237" aria-hidden="true" tabindex="-1"></a>    against the University's guidelines on using AI tools ([student</span>
<span id="cb27-238"><a href="#cb27-238" aria-hidden="true" tabindex="-1"></a>    guidance on using AI and translation</span>
<span id="cb27-239"><a href="#cb27-239" aria-hidden="true" tabindex="-1"></a>    tools](https://www.york.ac.uk/students/studying/assessment-and-examination/ai/))</span>
<span id="cb27-240"><a href="#cb27-240" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>You might be able to do the task quicker without generative AI by</span>
<span id="cb27-241"><a href="#cb27-241" aria-hidden="true" tabindex="-1"></a>    reading workshop materials because you won't have to evaluate and</span>
<span id="cb27-242"><a href="#cb27-242" aria-hidden="true" tabindex="-1"></a>    edit so much.</span>
<span id="cb27-243"><a href="#cb27-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-244"><a href="#cb27-244" aria-hidden="true" tabindex="-1"></a>+---------------------------------+---------------------------------+</span>
<span id="cb27-245"><a href="#cb27-245" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> This is OK                      <span class="pp">|</span> This is NOT ok                  <span class="pp">|</span></span>
<span id="cb27-246"><a href="#cb27-246" aria-hidden="true" tabindex="-1"></a>+===============================<span class="al">==+==</span>===============================+</span>
<span id="cb27-247"><a href="#cb27-247" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> It is OK to ask an AI to        <span class="pp">|</span> Do not use an AI to             <span class="pp">|</span></span>
<span id="cb27-248"><a href="#cb27-248" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> **explain** code and coding     | **generate** code               <span class="pp">|</span></span>
<span id="cb27-249"><a href="#cb27-249" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> concepts.                       <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-250"><a href="#cb27-250" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span>                                 <span class="pp">|</span> Do not **uncritically accept**  <span class="pp">|</span></span>
<span id="cb27-251"><a href="#cb27-251" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> It is OK to use an AI to        <span class="pp">|</span> code modifications without      <span class="pp">|</span></span>
<span id="cb27-252"><a href="#cb27-252" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> **modify** your code but you    <span class="pp">|</span> evaluating their accuracy or    <span class="pp">|</span></span>
<span id="cb27-253"><a href="#cb27-253" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> must understand the inputs, the <span class="pp">|</span> relevance.                      <span class="pp">|</span></span>
<span id="cb27-254"><a href="#cb27-254" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> outputs and what the code does. <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-255"><a href="#cb27-255" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> This is important because AIs   <span class="pp">|</span> Do not use an AI to             <span class="pp">|</span></span>
<span id="cb27-256"><a href="#cb27-256" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> like to add things you don't    <span class="pp">|</span> **generate** comments.          <span class="pp">|</span></span>
<span id="cb27-257"><a href="#cb27-257" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> need and the marking scheme     <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-258"><a href="#cb27-258" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> takes account relevance.        <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-259"><a href="#cb27-259" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span>                                 <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-260"><a href="#cb27-260" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Do [Acknowledge and             <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-261"><a href="#cb27-261" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> re                              <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-262"><a href="#cb27-262" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> ference](https://%20sub%20jectg <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-263"><a href="#cb27-263" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> uides.york.ac.uk/referen%20%20c <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-264"><a href="#cb27-264" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> ing-style-guides/generative-ai) <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-265"><a href="#cb27-265" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> generative AI (you should also  <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-266"><a href="#cb27-266" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> acknowledge and reference other <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-267"><a href="#cb27-267" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> sources of code such as         <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-268"><a href="#cb27-268" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> workshop, text books, online    <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-269"><a href="#cb27-269" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> tutorials)                      <span class="pp">|</span>                                 <span class="pp">|</span></span>
<span id="cb27-270"><a href="#cb27-270" aria-hidden="true" tabindex="-1"></a>+---------------------------------+---------------------------------+</span>
<span id="cb27-271"><a href="#cb27-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-272"><a href="#cb27-272" aria-hidden="true" tabindex="-1"></a><span class="fu"># Big data</span></span>
<span id="cb27-273"><a href="#cb27-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-274"><a href="#cb27-274" aria-hidden="true" tabindex="-1"></a>People argue about what quantity of data constitutes big data. In my</span>
<span id="cb27-275"><a href="#cb27-275" aria-hidden="true" tabindex="-1"></a>view, it is relative to what you are used to and it is not just about</span>
<span id="cb27-276"><a href="#cb27-276" aria-hidden="true" tabindex="-1"></a>the size of the data but also the complexity and the approach to</span>
<span id="cb27-277"><a href="#cb27-277" aria-hidden="true" tabindex="-1"></a>analysis.</span>
<span id="cb27-278"><a href="#cb27-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-279"><a href="#cb27-279" aria-hidden="true" tabindex="-1"></a>You have been used to working with data that has a few explanatory</span>
<span id="cb27-280"><a href="#cb27-280" aria-hidden="true" tabindex="-1"></a>variables, one or two response variables 10 to 100 observations.</span>
<span id="cb27-281"><a href="#cb27-281" aria-hidden="true" tabindex="-1"></a>Typically your approach has been to test one or two specific hypotheses.</span>
<span id="cb27-282"><a href="#cb27-282" aria-hidden="true" tabindex="-1"></a>There is usually a very direct relationship between the experiment you</span>
<span id="cb27-283"><a href="#cb27-283" aria-hidden="true" tabindex="-1"></a>design, the data you collect and the hypotheses you test.</span>
<span id="cb27-284"><a href="#cb27-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-285"><a href="#cb27-285" aria-hidden="true" tabindex="-1"></a>Example: You want to know the effect of an antimicrobial agent on the</span>
<span id="cb27-286"><a href="#cb27-286" aria-hidden="true" tabindex="-1"></a>growth of a bacterium. You design an experiment to measure the growth of</span>
<span id="cb27-287"><a href="#cb27-287" aria-hidden="true" tabindex="-1"></a>the bacterium in the presence and absence of the antimicrobial agent.</span>
<span id="cb27-288"><a href="#cb27-288" aria-hidden="true" tabindex="-1"></a>You collect the data and test the hypothesis that the growth of the</span>
<span id="cb27-289"><a href="#cb27-289" aria-hidden="true" tabindex="-1"></a>bacterium is different in the presence and absence of the antimicrobial</span>
<span id="cb27-290"><a href="#cb27-290" aria-hidden="true" tabindex="-1"></a>agent. You have a dataset with one response variable (growth) and one</span>
<span id="cb27-291"><a href="#cb27-291" aria-hidden="true" tabindex="-1"></a>explanatory variable (presence or absence of the antimicrobial agent)</span>
<span id="cb27-292"><a href="#cb27-292" aria-hidden="true" tabindex="-1"></a>and n observations (replicates of the experiment).</span>
<span id="cb27-293"><a href="#cb27-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-294"><a href="#cb27-294" aria-hidden="true" tabindex="-1"></a>When working with big data we often mean:</span>
<span id="cb27-295"><a href="#cb27-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-296"><a href="#cb27-296" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>having many variables (100s or 1000s), many observations or both</span>
<span id="cb27-297"><a href="#cb27-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-298"><a href="#cb27-298" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>visualising patterns in the data such as the relationship between</span>
<span id="cb27-299"><a href="#cb27-299" aria-hidden="true" tabindex="-1"></a>    variables or observations using "data reduction" techniques such as</span>
<span id="cb27-300"><a href="#cb27-300" aria-hidden="true" tabindex="-1"></a>    Principle Components Analysis (PCA)</span>
<span id="cb27-301"><a href="#cb27-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-302"><a href="#cb27-302" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>applying hundreds or thousands to tests to identify patterns or</span>
<span id="cb27-303"><a href="#cb27-303" aria-hidden="true" tabindex="-1"></a>    relationships amongst the significant results rather than drawing</span>
<span id="cb27-304"><a href="#cb27-304" aria-hidden="true" tabindex="-1"></a>    conclusions from individual tests</span>
<span id="cb27-305"><a href="#cb27-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-306"><a href="#cb27-306" aria-hidden="true" tabindex="-1"></a>Example: You want to know the effect of a drug on the expression of</span>
<span id="cb27-307"><a href="#cb27-307" aria-hidden="true" tabindex="-1"></a>genes in a cell. You design an experiment to measure the expression of</span>
<span id="cb27-308"><a href="#cb27-308" aria-hidden="true" tabindex="-1"></a>*all* the genes in the cell in the presence and absence of the drug. You</span>
<span id="cb27-309"><a href="#cb27-309" aria-hidden="true" tabindex="-1"></a>collect the data and test the hypothesis that the expression of any gene</span>
<span id="cb27-310"><a href="#cb27-310" aria-hidden="true" tabindex="-1"></a>is different in the presence and absence of the drug. You then look for</span>
<span id="cb27-311"><a href="#cb27-311" aria-hidden="true" tabindex="-1"></a>patterns in the data such as groups of genes that are co-expressed and</span>
<span id="cb27-312"><a href="#cb27-312" aria-hidden="true" tabindex="-1"></a>the relationship between the expression of genes and the effect of the</span>
<span id="cb27-313"><a href="#cb27-313" aria-hidden="true" tabindex="-1"></a>drug. You have a dataset with many response variables (expression of</span>
<span id="cb27-314"><a href="#cb27-314" aria-hidden="true" tabindex="-1"></a>genes) and one explanatory variable (presence or absence of the drug)</span>
<span id="cb27-315"><a href="#cb27-315" aria-hidden="true" tabindex="-1"></a>and n observations (replicates of the experiment).</span>
<span id="cb27-316"><a href="#cb27-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-317"><a href="#cb27-317" aria-hidden="true" tabindex="-1"></a><span class="fu"># Exercises</span></span>
<span id="cb27-318"><a href="#cb27-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-319"><a href="#cb27-319" aria-hidden="true" tabindex="-1"></a><span class="fu">## Set up a Project</span></span>
<span id="cb27-320"><a href="#cb27-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-321"><a href="#cb27-321" aria-hidden="true" tabindex="-1"></a>üé¨ Start RStudio from the Start menu</span>
<span id="cb27-322"><a href="#cb27-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-323"><a href="#cb27-323" aria-hidden="true" tabindex="-1"></a>üé¨ Make an RStudio project. Be deliberate about where you create it so</span>
<span id="cb27-324"><a href="#cb27-324" aria-hidden="true" tabindex="-1"></a>that it is a good place for you</span>
<span id="cb27-325"><a href="#cb27-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-326"><a href="#cb27-326" aria-hidden="true" tabindex="-1"></a>üé¨ Use the Files pane to make a folder for the data. I suggest</span>
<span id="cb27-327"><a href="#cb27-327" aria-hidden="true" tabindex="-1"></a><span class="in">`data-raw/`</span></span>
<span id="cb27-328"><a href="#cb27-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-329"><a href="#cb27-329" aria-hidden="true" tabindex="-1"></a>üé¨ Make a new script called <span class="in">`core-data-analysis.R`</span> to carry out the rest</span>
<span id="cb27-330"><a href="#cb27-330" aria-hidden="true" tabindex="-1"></a>of the work.</span>
<span id="cb27-331"><a href="#cb27-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-332"><a href="#cb27-332" aria-hidden="true" tabindex="-1"></a><span class="fu">## Load packages</span></span>
<span id="cb27-333"><a href="#cb27-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-334"><a href="#cb27-334" aria-hidden="true" tabindex="-1"></a>We need the following packages for this workshop:</span>
<span id="cb27-335"><a href="#cb27-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-336"><a href="#cb27-336" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**`tidyverse`** <span class="co">[</span><span class="ot">@tidyverse</span><span class="co">]</span>: importing the meta data which is in a</span>
<span id="cb27-337"><a href="#cb27-337" aria-hidden="true" tabindex="-1"></a>    text file, working with the data once it is in a dataframe to</span>
<span id="cb27-338"><a href="#cb27-338" aria-hidden="true" tabindex="-1"></a>    filter, summarise and plot.</span>
<span id="cb27-339"><a href="#cb27-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-340"><a href="#cb27-340" aria-hidden="true" tabindex="-1"></a>üé¨ Load **`tidyverse`**</span>
<span id="cb27-341"><a href="#cb27-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-344"><a href="#cb27-344" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-345"><a href="#cb27-345" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb27-346"><a href="#cb27-346" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-347"><a href="#cb27-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-348"><a href="#cb27-348" aria-hidden="true" tabindex="-1"></a><span class="fu">## Look at the data!</span></span>
<span id="cb27-349"><a href="#cb27-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-350"><a href="#cb27-350" aria-hidden="true" tabindex="-1"></a>It is almost always useful to take a look at your data (or at least</span>
<span id="cb27-351"><a href="#cb27-351" aria-hidden="true" tabindex="-1"></a>think about its format) before you attempt to read it in.</span>
<span id="cb27-352"><a href="#cb27-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-353"><a href="#cb27-353" aria-hidden="true" tabindex="-1"></a>We will be working with the following data files in this workshop</span>
<span id="cb27-354"><a href="#cb27-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-355"><a href="#cb27-355" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="co">[</span><span class="ot">biotech-cts.txt</span><span class="co">](data-raw/biotech-cts.txt)</span>. These are RNA-Seq data</span>
<span id="cb27-356"><a href="#cb27-356" aria-hidden="true" tabindex="-1"></a>    from three replicates for each of two of wheat varieties</span>
<span id="cb27-357"><a href="#cb27-357" aria-hidden="true" tabindex="-1"></a>    (<span class="co">[</span><span class="ot">S</span><span class="co">]</span>{.underline}usceptible and <span class="co">[</span><span class="ot">T</span><span class="co">]</span>{.underline}olerant) grown at two</span>
<span id="cb27-358"><a href="#cb27-358" aria-hidden="true" tabindex="-1"></a>    potassium conditions (<span class="co">[</span><span class="ot">C</span><span class="co">]</span>{.underline}ontrol <span class="co">[</span><span class="ot">K</span><span class="co">]</span>{.underline} and</span>
<span id="cb27-359"><a href="#cb27-359" aria-hidden="true" tabindex="-1"></a>    <span class="co">[</span><span class="ot">L</span><span class="co">]</span>{.underline}ow <span class="co">[</span><span class="ot">K</span><span class="co">]</span>{.underline}). The data are counts of the</span>
<span id="cb27-360"><a href="#cb27-360" aria-hidden="true" tabindex="-1"></a>    number of reads.</span>
<span id="cb27-361"><a href="#cb27-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-362"><a href="#cb27-362" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="co">[</span><span class="ot">cell-bio.tsv</span><span class="co">](data-raw/cell-bio.tsv)</span>. These are measurements from a</span>
<span id="cb27-363"><a href="#cb27-363" aria-hidden="true" tabindex="-1"></a>    Livecyte microscope, which performs live cell imaging, tracking</span>
<span id="cb27-364"><a href="#cb27-364" aria-hidden="true" tabindex="-1"></a>    individual cells, and measuring cell shape and size parameters for</span>
<span id="cb27-365"><a href="#cb27-365" aria-hidden="true" tabindex="-1"></a>    thousands of cells as they grow and divide. Measurements are taken</span>
<span id="cb27-366"><a href="#cb27-366" aria-hidden="true" tabindex="-1"></a>    for three replicates from each of two cell lines A and B.</span>
<span id="cb27-367"><a href="#cb27-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-368"><a href="#cb27-368" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="co">[</span><span class="ot">immuno.csv</span><span class="co">](data-raw/immuno.csv)</span>. These are flow cytometry data for</span>
<span id="cb27-369"><a href="#cb27-369" aria-hidden="true" tabindex="-1"></a>    three treatments with each of two antibodies. The measures are</span>
<span id="cb27-370"><a href="#cb27-370" aria-hidden="true" tabindex="-1"></a>    Forward scatter, side scatter, red fluorescence and green</span>
<span id="cb27-371"><a href="#cb27-371" aria-hidden="true" tabindex="-1"></a>    fluorescence.</span>
<span id="cb27-372"><a href="#cb27-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-373"><a href="#cb27-373" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="co">[</span><span class="ot">xlaevis_counts_S20.csv</span><span class="co">](data-raw/xlaevis_counts_S20.csv)</span>. These are</span>
<span id="cb27-374"><a href="#cb27-374" aria-hidden="true" tabindex="-1"></a>    RNA-Seq data from frogs. There are 2 siblings from each of three</span>
<span id="cb27-375"><a href="#cb27-375" aria-hidden="true" tabindex="-1"></a>    fertilisations and one sibling was treated with FGF and the other</span>
<span id="cb27-376"><a href="#cb27-376" aria-hidden="true" tabindex="-1"></a>    was a control. The data are counts of the number of reads.</span>
<span id="cb27-377"><a href="#cb27-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-378"><a href="#cb27-378" aria-hidden="true" tabindex="-1"></a>üé¨ Save these data files to the <span class="in">`data-raw/`</span> folder:</span>
<span id="cb27-379"><a href="#cb27-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-380"><a href="#cb27-380" aria-hidden="true" tabindex="-1"></a>üé¨ Consider the file extensions. What you do think the extensions</span>
<span id="cb27-381"><a href="#cb27-381" aria-hidden="true" tabindex="-1"></a>indicate?</span>
<span id="cb27-382"><a href="#cb27-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-383"><a href="#cb27-383" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER--- --&gt;</span></span>
<span id="cb27-384"><a href="#cb27-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-385"><a href="#cb27-385" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- txt usually indicates a plain text file, where the columns are  --&gt;</span></span>
<span id="cb27-386"><a href="#cb27-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-387"><a href="#cb27-387" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- often separated by a space but might also be separated by a tab,  --&gt;</span></span>
<span id="cb27-388"><a href="#cb27-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-389"><a href="#cb27-389" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- a backslash or forward slash, or some other character.  --&gt;</span></span>
<span id="cb27-390"><a href="#cb27-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-391"><a href="#cb27-391" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Plain text files can be opened in notepad or other similar --&gt;</span></span>
<span id="cb27-392"><a href="#cb27-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-393"><a href="#cb27-393" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- editor and still be readable. --&gt;</span></span>
<span id="cb27-394"><a href="#cb27-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-395"><a href="#cb27-395" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- csv stands for comma separated values and tsv stands for tab  --&gt;</span></span>
<span id="cb27-396"><a href="#cb27-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-397"><a href="#cb27-397" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- separated values. Both of these are plain text format. --&gt;</span></span>
<span id="cb27-398"><a href="#cb27-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-399"><a href="#cb27-399" aria-hidden="true" tabindex="-1"></a>üé¨ Go to the Files pane click on the <span class="in">`xlaevis_counts_S20.csv`</span> file and</span>
<span id="cb27-400"><a href="#cb27-400" aria-hidden="true" tabindex="-1"></a>choose View File<span class="ot">[^1]</span></span>
<span id="cb27-401"><a href="#cb27-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-402"><a href="#cb27-402" aria-hidden="true" tabindex="-1"></a><span class="ot">[^1]: </span>**Do not** be tempted to import data this way. Unless you are</span>
<span id="cb27-403"><a href="#cb27-403" aria-hidden="true" tabindex="-1"></a>    careful, your data import will not be scripted or will not be</span>
<span id="cb27-404"><a href="#cb27-404" aria-hidden="true" tabindex="-1"></a>    scripted correctly. This means your work will not be reproducible.</span>
<span id="cb27-405"><a href="#cb27-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-406"><a href="#cb27-406" aria-hidden="true" tabindex="-1"></a>![RStudio Files</span>
<span id="cb27-407"><a href="#cb27-407" aria-hidden="true" tabindex="-1"></a>Pane](images/rstudio-filepane-viewfile.png){fig-alt="Rstudio Files pane showing the data files and the View File option that appears when you click on the a particular file"}</span>
<span id="cb27-408"><a href="#cb27-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-409"><a href="#cb27-409" aria-hidden="true" tabindex="-1"></a>Any plain text file will open in the top left pane. Note that this is</span>
<span id="cb27-410"><a href="#cb27-410" aria-hidden="true" tabindex="-1"></a>NOT importing the data, it's just viewing the file.</span>
<span id="cb27-411"><a href="#cb27-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-412"><a href="#cb27-412" aria-hidden="true" tabindex="-1"></a>üé¨ Does it seem to be a csv file?</span>
<span id="cb27-413"><a href="#cb27-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-414"><a href="#cb27-414" aria-hidden="true" tabindex="-1"></a>üé¨ Try opening the other files from the Files pane. What happens?</span>
<span id="cb27-415"><a href="#cb27-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-416"><a href="#cb27-416" aria-hidden="true" tabindex="-1"></a>üé¨ Open each file in Excel. For the <span class="in">`.txt`</span> and <span class="in">`.tsv`</span> files you will</span>
<span id="cb27-417"><a href="#cb27-417" aria-hidden="true" tabindex="-1"></a>need to show "All files (*.*)" in the file type dropdown to see them and</span>
<span id="cb27-418"><a href="#cb27-418" aria-hidden="true" tabindex="-1"></a>you will need to work through the "Text Import Wizard" to open them.</span>
<span id="cb27-419"><a href="#cb27-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-420"><a href="#cb27-420" aria-hidden="true" tabindex="-1"></a>We are opening them each in Excel to see what they look like and to get</span>
<span id="cb27-421"><a href="#cb27-421" aria-hidden="true" tabindex="-1"></a>a sense of the structure of the data. We are not going to use Excel to</span>
<span id="cb27-422"><a href="#cb27-422" aria-hidden="true" tabindex="-1"></a>do anything.</span>
<span id="cb27-423"><a href="#cb27-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-424"><a href="#cb27-424" aria-hidden="true" tabindex="-1"></a>üé¨ What do you notice? Does it look like this data will be easy to</span>
<span id="cb27-425"><a href="#cb27-425" aria-hidden="true" tabindex="-1"></a>import and work with?</span>
<span id="cb27-426"><a href="#cb27-426" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-427"><a href="#cb27-427" aria-hidden="true" tabindex="-1"></a><span class="fu">## Import the data</span></span>
<span id="cb27-428"><a href="#cb27-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-429"><a href="#cb27-429" aria-hidden="true" tabindex="-1"></a>üé¨ Import each of the data files into R:</span>
<span id="cb27-430"><a href="#cb27-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-433"><a href="#cb27-433" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-434"><a href="#cb27-434" aria-hidden="true" tabindex="-1"></a>biotech <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data-raw/biotech-cts.txt"</span>)</span>
<span id="cb27-435"><a href="#cb27-435" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-436"><a href="#cb27-436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-439"><a href="#cb27-439" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-440"><a href="#cb27-440" aria-hidden="true" tabindex="-1"></a>cell_bio <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data-raw/cell-bio.tsv"</span>)</span>
<span id="cb27-441"><a href="#cb27-441" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-442"><a href="#cb27-442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-445"><a href="#cb27-445" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-446"><a href="#cb27-446" aria-hidden="true" tabindex="-1"></a>immuno <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data-raw/immuno.csv"</span>)</span>
<span id="cb27-447"><a href="#cb27-447" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-448"><a href="#cb27-448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-451"><a href="#cb27-451" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-452"><a href="#cb27-452" aria-hidden="true" tabindex="-1"></a>frogs <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data-raw/xlaevis_counts_S20.csv"</span>)</span>
<span id="cb27-453"><a href="#cb27-453" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-454"><a href="#cb27-454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-455"><a href="#cb27-455" aria-hidden="true" tabindex="-1"></a>üé¨ Click on each of the dataframes in the Environment pane to see what</span>
<span id="cb27-456"><a href="#cb27-456" aria-hidden="true" tabindex="-1"></a>they look like. An alternative to clicking is to use the <span class="in">`View()`</span></span>
<span id="cb27-457"><a href="#cb27-457" aria-hidden="true" tabindex="-1"></a>function:</span>
<span id="cb27-458"><a href="#cb27-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-461"><a href="#cb27-461" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-462"><a href="#cb27-462" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb27-463"><a href="#cb27-463" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(cell_bio)</span>
<span id="cb27-464"><a href="#cb27-464" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-465"><a href="#cb27-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-466"><a href="#cb27-466" aria-hidden="true" tabindex="-1"></a>üé¨ In each case, describe what is in the rows and the columns of the</span>
<span id="cb27-467"><a href="#cb27-467" aria-hidden="true" tabindex="-1"></a>dataframe. Where are the replicates/samples and where are the variables?</span>
<span id="cb27-468"><a href="#cb27-468" aria-hidden="true" tabindex="-1"></a>Are there any things you'd like to fix/change?</span>
<span id="cb27-469"><a href="#cb27-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-470"><a href="#cb27-470" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER--- --&gt;</span></span>
<span id="cb27-471"><a href="#cb27-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-472"><a href="#cb27-472" aria-hidden="true" tabindex="-1"></a>üé¨ Name of the first column in the <span class="in">`biotech`</span> dataframe:</span>
<span id="cb27-473"><a href="#cb27-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-476"><a href="#cb27-476" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-477"><a href="#cb27-477" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(biotech)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">"transcript"</span></span>
<span id="cb27-478"><a href="#cb27-478" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-479"><a href="#cb27-479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-480"><a href="#cb27-480" aria-hidden="true" tabindex="-1"></a>üé¨ Column names in the <span class="in">`cell_bio`</span> dataframe:</span>
<span id="cb27-481"><a href="#cb27-481" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-484"><a href="#cb27-484" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-485"><a href="#cb27-485" aria-hidden="true" tabindex="-1"></a>cell_bio <span class="ot">&lt;-</span> janitor<span class="sc">::</span><span class="fu">clean_names</span>(cell_bio)</span>
<span id="cb27-486"><a href="#cb27-486" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-487"><a href="#cb27-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-488"><a href="#cb27-488" aria-hidden="true" tabindex="-1"></a><span class="fu">## Getting an overview 1</span></span>
<span id="cb27-489"><a href="#cb27-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-490"><a href="#cb27-490" aria-hidden="true" tabindex="-1"></a><span class="fu">### `summary()`</span></span>
<span id="cb27-491"><a href="#cb27-491" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-492"><a href="#cb27-492" aria-hidden="true" tabindex="-1"></a>R's <span class="in">`summary()`</span> function is a quick way to get an overview of datasets.</span>
<span id="cb27-493"><a href="#cb27-493" aria-hidden="true" tabindex="-1"></a>It gives you a six number summary for every numeric variable (the</span>
<span id="cb27-494"><a href="#cb27-494" aria-hidden="true" tabindex="-1"></a>minimum, lower quartile, median, mean, upper quartile, and maximum). It</span>
<span id="cb27-495"><a href="#cb27-495" aria-hidden="true" tabindex="-1"></a>also gives you a count of the number of missing values.</span>
<span id="cb27-496"><a href="#cb27-496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-497"><a href="#cb27-497" aria-hidden="true" tabindex="-1"></a>üé¨ Use the <span class="in">`summary()`</span> function to get an overview of each dataframe. It</span>
<span id="cb27-498"><a href="#cb27-498" aria-hidden="true" tabindex="-1"></a>works well for these datasets. What types of dataset might this be less</span>
<span id="cb27-499"><a href="#cb27-499" aria-hidden="true" tabindex="-1"></a>useful for? Which dataframes and columns have missing values? Which</span>
<span id="cb27-500"><a href="#cb27-500" aria-hidden="true" tabindex="-1"></a>dataframes and variables seem very skewed? Which dataframe has more than</span>
<span id="cb27-501"><a href="#cb27-501" aria-hidden="true" tabindex="-1"></a>one character variable?</span>
<span id="cb27-502"><a href="#cb27-502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-503"><a href="#cb27-503" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER--- --&gt;</span></span>
<span id="cb27-504"><a href="#cb27-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-505"><a href="#cb27-505" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- It is less useful for datasets with very many variables  --&gt;</span></span>
<span id="cb27-506"><a href="#cb27-506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-507"><a href="#cb27-507" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- (columns) because we aren't able to take that in all  --&gt;</span></span>
<span id="cb27-508"><a href="#cb27-508" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-509"><a href="#cb27-509" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- at once. It fills up the screen buffer and we can't --&gt;</span></span>
<span id="cb27-510"><a href="#cb27-510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-511"><a href="#cb27-511" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- see all the output at once. --&gt;</span></span>
<span id="cb27-512"><a href="#cb27-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-513"><a href="#cb27-513" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- It is also less useful for datasets with many  --&gt;</span></span>
<span id="cb27-514"><a href="#cb27-514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-515"><a href="#cb27-515" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- character variables --&gt;</span></span>
<span id="cb27-516"><a href="#cb27-516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-517"><a href="#cb27-517" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- cell_bio has a perimeter column with 19948 NAs --&gt;</span></span>
<span id="cb27-518"><a href="#cb27-518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-519"><a href="#cb27-519" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- skew is revealed by the difference between the mean and  --&gt;</span></span>
<span id="cb27-520"><a href="#cb27-520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-521"><a href="#cb27-521" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- the median and a maximum that is much larger than all  --&gt;</span></span>
<span id="cb27-522"><a href="#cb27-522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-523"><a href="#cb27-523" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- the other values. The frogs and biotech dataframes have --&gt;</span></span>
<span id="cb27-524"><a href="#cb27-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-525"><a href="#cb27-525" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- skewed variables. Skew is typical in RNA-Seq counts. --&gt;</span></span>
<span id="cb27-526"><a href="#cb27-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-527"><a href="#cb27-527" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- immuno is the only dataset with more than one character --&gt;</span></span>
<span id="cb27-528"><a href="#cb27-528" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-529"><a href="#cb27-529" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- variable (it has three) --&gt;</span></span>
<span id="cb27-530"><a href="#cb27-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-531"><a href="#cb27-531" aria-hidden="true" tabindex="-1"></a><span class="fu">## Quality Control</span></span>
<span id="cb27-532"><a href="#cb27-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-533"><a href="#cb27-533" aria-hidden="true" tabindex="-1"></a><span class="fu">### Filtering rows</span></span>
<span id="cb27-534"><a href="#cb27-534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-535"><a href="#cb27-535" aria-hidden="true" tabindex="-1"></a>The filter function selects/drops a whole row on a condition. The</span>
<span id="cb27-536"><a href="#cb27-536" aria-hidden="true" tabindex="-1"></a>condition can be based on the value in one, or a combination, of</span>
<span id="cb27-537"><a href="#cb27-537" aria-hidden="true" tabindex="-1"></a>columns. We specify what we want to keep with the <span class="in">`filter()`</span> function.</span>
<span id="cb27-538"><a href="#cb27-538" aria-hidden="true" tabindex="-1"></a>This means you might often want to negate a condition with <span class="in">`!`</span>.</span>
<span id="cb27-539"><a href="#cb27-539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-540"><a href="#cb27-540" aria-hidden="true" tabindex="-1"></a>üé¨ To remove the rows in <span class="in">`cell_bio`</span> with missing values in the</span>
<span id="cb27-541"><a href="#cb27-541" aria-hidden="true" tabindex="-1"></a><span class="in">`perimeter`</span> column:</span>
<span id="cb27-542"><a href="#cb27-542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-545"><a href="#cb27-545" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-546"><a href="#cb27-546" aria-hidden="true" tabindex="-1"></a>cell_bio <span class="ot">&lt;-</span> cell_bio <span class="sc">|&gt;</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(perimeter))</span>
<span id="cb27-547"><a href="#cb27-547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-548"><a href="#cb27-548" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-549"><a href="#cb27-549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-550"><a href="#cb27-550" aria-hidden="true" tabindex="-1"></a><span class="in">`is.na(perimeter)`</span> returns a logical vector (a vector of <span class="in">`TRUE`</span>s and</span>
<span id="cb27-551"><a href="#cb27-551" aria-hidden="true" tabindex="-1"></a><span class="in">`FALSE`</span>s) of the same length as the column. The <span class="in">`!`</span> negates the logical</span>
<span id="cb27-552"><a href="#cb27-552" aria-hidden="true" tabindex="-1"></a>vector so the <span class="in">`TRUE`</span>s become <span class="in">`FALSE`</span>s and the <span class="in">`FALSE`</span>s become TRUE. This</span>
<span id="cb27-553"><a href="#cb27-553" aria-hidden="true" tabindex="-1"></a>means that the <span class="in">`filter()`</span> function *keeps* the rows where the perimeter</span>
<span id="cb27-554"><a href="#cb27-554" aria-hidden="true" tabindex="-1"></a>column is not missing.</span>
<span id="cb27-555"><a href="#cb27-555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-556"><a href="#cb27-556" aria-hidden="true" tabindex="-1"></a>üé¨ There's actually a tidyverse function that does the same thing as</span>
<span id="cb27-557"><a href="#cb27-557" aria-hidden="true" tabindex="-1"></a><span class="in">`filter(!is.na())`</span></span>
<span id="cb27-558"><a href="#cb27-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-561"><a href="#cb27-561" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-562"><a href="#cb27-562" aria-hidden="true" tabindex="-1"></a>cell_bio <span class="ot">&lt;-</span> cell_bio <span class="sc">|&gt;</span> <span class="fu">drop_na</span>(perimeter)</span>
<span id="cb27-563"><a href="#cb27-563" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-564"><a href="#cb27-564" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-565"><a href="#cb27-565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-566"><a href="#cb27-566" aria-hidden="true" tabindex="-1"></a>Sometimes you might want to work with a subset of data. For example, you</span>
<span id="cb27-567"><a href="#cb27-567" aria-hidden="true" tabindex="-1"></a>might want to examine just the Media treated cells in the <span class="in">`immuno`</span></span>
<span id="cb27-568"><a href="#cb27-568" aria-hidden="true" tabindex="-1"></a>dataset.</span>
<span id="cb27-569"><a href="#cb27-569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-570"><a href="#cb27-570" aria-hidden="true" tabindex="-1"></a>üé¨ Create a new dataframe called <span class="in">`immuno_media`</span> that contains only the</span>
<span id="cb27-571"><a href="#cb27-571" aria-hidden="true" tabindex="-1"></a>rows where <span class="in">`treatment`</span> is "MEDIA"</span>
<span id="cb27-572"><a href="#cb27-572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-575"><a href="#cb27-575" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-576"><a href="#cb27-576" aria-hidden="true" tabindex="-1"></a>immuno_media <span class="ot">&lt;-</span> immuno <span class="sc">|&gt;</span> </span>
<span id="cb27-577"><a href="#cb27-577" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(treatment <span class="sc">==</span> <span class="st">"MEDIA"</span>)</span>
<span id="cb27-578"><a href="#cb27-578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-579"><a href="#cb27-579" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-580"><a href="#cb27-580" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-581"><a href="#cb27-581" aria-hidden="true" tabindex="-1"></a>Or just the Media treated cells with a <span class="in">`FS_Lin`</span> between two values. You</span>
<span id="cb27-582"><a href="#cb27-582" aria-hidden="true" tabindex="-1"></a>can apply two filters and the <span class="in">`between()`</span> function to achieve this:</span>
<span id="cb27-583"><a href="#cb27-583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-584"><a href="#cb27-584" aria-hidden="true" tabindex="-1"></a>üé¨ Create a new dataframe called <span class="in">`immuno_media_live`</span> that contains only</span>
<span id="cb27-585"><a href="#cb27-585" aria-hidden="true" tabindex="-1"></a>the rows where <span class="in">`treatment`</span> is "MEDIA" and <span class="in">`FS_Lin`</span> is between 7500 and</span>
<span id="cb27-586"><a href="#cb27-586" aria-hidden="true" tabindex="-1"></a>28000:</span>
<span id="cb27-587"><a href="#cb27-587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-590"><a href="#cb27-590" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-591"><a href="#cb27-591" aria-hidden="true" tabindex="-1"></a>immuno_media_live <span class="ot">&lt;-</span> immuno <span class="sc">|&gt;</span></span>
<span id="cb27-592"><a href="#cb27-592" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(treatment <span class="sc">==</span> <span class="st">"MEDIA"</span>) <span class="sc">|&gt;</span></span>
<span id="cb27-593"><a href="#cb27-593" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">between</span>(FS_Lin, <span class="dv">7500</span>, <span class="dv">28000</span>))</span>
<span id="cb27-594"><a href="#cb27-594" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-595"><a href="#cb27-595" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-596"><a href="#cb27-596" aria-hidden="true" tabindex="-1"></a>üé¨ Now you try. Create a dataframe called <span class="in">`immuno_live`</span> that contains</span>
<span id="cb27-597"><a href="#cb27-597" aria-hidden="true" tabindex="-1"></a>only the rows where <span class="in">`FS_Lin`</span> is between 7500 and 28000 and <span class="in">`SS_Lin`</span> is</span>
<span id="cb27-598"><a href="#cb27-598" aria-hidden="true" tabindex="-1"></a>between 15000 and 35000:</span>
<span id="cb27-599"><a href="#cb27-599" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-602"><a href="#cb27-602" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-603"><a href="#cb27-603" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb27-604"><a href="#cb27-604" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-605"><a href="#cb27-605" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER---</span></span>
<span id="cb27-606"><a href="#cb27-606" aria-hidden="true" tabindex="-1"></a>immuno_live <span class="ot">&lt;-</span> immuno <span class="sc">|&gt;</span></span>
<span id="cb27-607"><a href="#cb27-607" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">between</span>(FS_Lin, <span class="dv">7500</span>, <span class="dv">28000</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb27-608"><a href="#cb27-608" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">between</span>(SS_Lin, <span class="dv">15000</span>, <span class="dv">35000</span>))</span>
<span id="cb27-609"><a href="#cb27-609" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-610"><a href="#cb27-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-611"><a href="#cb27-611" aria-hidden="true" tabindex="-1"></a><span class="fu">### Selecting columns</span></span>
<span id="cb27-612"><a href="#cb27-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-613"><a href="#cb27-613" aria-hidden="true" tabindex="-1"></a>Whilst we can always specify the columns we want to use when working</span>
<span id="cb27-614"><a href="#cb27-614" aria-hidden="true" tabindex="-1"></a>with data, sometimes we find it less overwhelming to create a new</span>
<span id="cb27-615"><a href="#cb27-615" aria-hidden="true" tabindex="-1"></a>dataframe with only the columns we want. The <span class="in">`select()`</span> function helps</span>
<span id="cb27-616"><a href="#cb27-616" aria-hidden="true" tabindex="-1"></a>us here.</span>
<span id="cb27-617"><a href="#cb27-617" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-618"><a href="#cb27-618" aria-hidden="true" tabindex="-1"></a>üé¨ Suppose we only want to work with the Susceptible varieties of wheat</span>
<span id="cb27-619"><a href="#cb27-619" aria-hidden="true" tabindex="-1"></a>in the <span class="in">`biotech`</span> dataframe. We can create a new dataframe with only the</span>
<span id="cb27-620"><a href="#cb27-620" aria-hidden="true" tabindex="-1"></a>columns we want:</span>
<span id="cb27-621"><a href="#cb27-621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-624"><a href="#cb27-624" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-625"><a href="#cb27-625" aria-hidden="true" tabindex="-1"></a>biotech_susceptible <span class="ot">&lt;-</span> biotech <span class="sc">|&gt;</span> </span>
<span id="cb27-626"><a href="#cb27-626" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SCK14_1,</span>
<span id="cb27-627"><a href="#cb27-627" aria-hidden="true" tabindex="-1"></a>         SCK14_2,</span>
<span id="cb27-628"><a href="#cb27-628" aria-hidden="true" tabindex="-1"></a>         SCK14_3,</span>
<span id="cb27-629"><a href="#cb27-629" aria-hidden="true" tabindex="-1"></a>         SLK14_1,</span>
<span id="cb27-630"><a href="#cb27-630" aria-hidden="true" tabindex="-1"></a>         SLK14_2,</span>
<span id="cb27-631"><a href="#cb27-631" aria-hidden="true" tabindex="-1"></a>         SLK14_3)</span>
<span id="cb27-632"><a href="#cb27-632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-633"><a href="#cb27-633" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-634"><a href="#cb27-634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-635"><a href="#cb27-635" aria-hidden="true" tabindex="-1"></a>In fact, there are a couple of alternatives to this. We can use the</span>
<span id="cb27-636"><a href="#cb27-636" aria-hidden="true" tabindex="-1"></a><span class="in">`starts_with()`</span> function to select all the columns that start with a</span>
<span id="cb27-637"><a href="#cb27-637" aria-hidden="true" tabindex="-1"></a>certain string:</span>
<span id="cb27-638"><a href="#cb27-638" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-639"><a href="#cb27-639" aria-hidden="true" tabindex="-1"></a>üé¨ Select columns starting with <span class="in">`S`</span></span>
<span id="cb27-640"><a href="#cb27-640" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-643"><a href="#cb27-643" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-644"><a href="#cb27-644" aria-hidden="true" tabindex="-1"></a>biotech_susceptible <span class="ot">&lt;-</span> biotech <span class="sc">|&gt;</span> </span>
<span id="cb27-645"><a href="#cb27-645" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"S"</span>))</span>
<span id="cb27-646"><a href="#cb27-646" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-647"><a href="#cb27-647" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-648"><a href="#cb27-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-649"><a href="#cb27-649" aria-hidden="true" tabindex="-1"></a>There is an <span class="in">`ends_with()`</span> function too!</span>
<span id="cb27-650"><a href="#cb27-650" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-651"><a href="#cb27-651" aria-hidden="true" tabindex="-1"></a>üé¨ The colon notation allows us to select a range of columns. Select</span>
<span id="cb27-652"><a href="#cb27-652" aria-hidden="true" tabindex="-1"></a>columns from <span class="in">`SCK14_1`</span> to <span class="in">`SLK14_3`</span></span>
<span id="cb27-653"><a href="#cb27-653" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-656"><a href="#cb27-656" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-657"><a href="#cb27-657" aria-hidden="true" tabindex="-1"></a>biotech_susceptible <span class="ot">&lt;-</span> biotech <span class="sc">|&gt;</span> </span>
<span id="cb27-658"><a href="#cb27-658" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SCK14_1<span class="sc">:</span>SLK14_3)</span>
<span id="cb27-659"><a href="#cb27-659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-660"><a href="#cb27-660" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-661"><a href="#cb27-661" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-662"><a href="#cb27-662" aria-hidden="true" tabindex="-1"></a>Note that you need to pay attention to the order of the columns in your</span>
<span id="cb27-663"><a href="#cb27-663" aria-hidden="true" tabindex="-1"></a>dataframe to use this.</span>
<span id="cb27-664"><a href="#cb27-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-665"><a href="#cb27-665" aria-hidden="true" tabindex="-1"></a>üé¨ You can use select and filter together. Try creating a dataframe from</span>
<span id="cb27-666"><a href="#cb27-666" aria-hidden="true" tabindex="-1"></a><span class="in">`frogs`</span> which has only the columns from sibling "<span class="sc">\_</span>A" and only the rows</span>
<span id="cb27-667"><a href="#cb27-667" aria-hidden="true" tabindex="-1"></a>where <span class="in">`S20_C_5`</span> is above 20.</span>
<span id="cb27-668"><a href="#cb27-668" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-671"><a href="#cb27-671" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-672"><a href="#cb27-672" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb27-673"><a href="#cb27-673" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-674"><a href="#cb27-674" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER---</span></span>
<span id="cb27-675"><a href="#cb27-675" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-676"><a href="#cb27-676" aria-hidden="true" tabindex="-1"></a>frogs_subset <span class="ot">&lt;-</span> frogs <span class="sc">|&gt;</span>   </span>
<span id="cb27-677"><a href="#cb27-677" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(S20_C_5 <span class="sc">&gt;</span> <span class="dv">20</span>) <span class="sc">|&gt;</span> </span>
<span id="cb27-678"><a href="#cb27-678" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">ends_with</span>(<span class="st">"_A"</span>))</span>
<span id="cb27-679"><a href="#cb27-679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-680"><a href="#cb27-680" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-681"><a href="#cb27-681" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-682"><a href="#cb27-682" aria-hidden="true" tabindex="-1"></a><span class="fu">### Visualisation</span></span>
<span id="cb27-683"><a href="#cb27-683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-684"><a href="#cb27-684" aria-hidden="true" tabindex="-1"></a>Histograms and density plots are useful for visualising the distribution</span>
<span id="cb27-685"><a href="#cb27-685" aria-hidden="true" tabindex="-1"></a>of variables when we have many observations. They allow you to see</span>
<span id="cb27-686"><a href="#cb27-686" aria-hidden="true" tabindex="-1"></a>features in the data that are not obvious from the summary statistics.</span>
<span id="cb27-687"><a href="#cb27-687" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-688"><a href="#cb27-688" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>A histogram shows the number of values in each range (bin), it is</span>
<span id="cb27-689"><a href="#cb27-689" aria-hidden="true" tabindex="-1"></a>    composed of touching bars. How a histogram looks depends on the</span>
<span id="cb27-690"><a href="#cb27-690" aria-hidden="true" tabindex="-1"></a>    number of bins used.</span>
<span id="cb27-691"><a href="#cb27-691" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-692"><a href="#cb27-692" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>A density plot shows the proportions of values in a range. The</span>
<span id="cb27-693"><a href="#cb27-693" aria-hidden="true" tabindex="-1"></a>    appearance of the distribution is not affected by the number of</span>
<span id="cb27-694"><a href="#cb27-694" aria-hidden="true" tabindex="-1"></a>    bins.</span>
<span id="cb27-695"><a href="#cb27-695" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-696"><a href="#cb27-696" aria-hidden="true" tabindex="-1"></a>üé¨ To plot a histogram of <span class="in">`S20_C_5`</span> in the <span class="in">`frogs`</span> dataframe:</span>
<span id="cb27-697"><a href="#cb27-697" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-700"><a href="#cb27-700" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-701"><a href="#cb27-701" aria-hidden="true" tabindex="-1"></a>frogs <span class="sc">|&gt;</span> </span>
<span id="cb27-702"><a href="#cb27-702" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> S20_C_5)) <span class="sc">+</span></span>
<span id="cb27-703"><a href="#cb27-703" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>() </span>
<span id="cb27-704"><a href="#cb27-704" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-705"><a href="#cb27-705" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-706"><a href="#cb27-706" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-707"><a href="#cb27-707" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-708"><a href="#cb27-708" aria-hidden="true" tabindex="-1"></a>This data is very skewed - there are so many low values that we can‚Äôt</span>
<span id="cb27-709"><a href="#cb27-709" aria-hidden="true" tabindex="-1"></a>see the tiny bars for the higher values. Logging the counts is a way to</span>
<span id="cb27-710"><a href="#cb27-710" aria-hidden="true" tabindex="-1"></a>make the distribution more visible.</span>
<span id="cb27-711"><a href="#cb27-711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-712"><a href="#cb27-712" aria-hidden="true" tabindex="-1"></a>üé¨ To plot the distribution of log <span class="in">`S20_C_5`</span> in the <span class="in">`frogs`</span> dataframe as</span>
<span id="cb27-713"><a href="#cb27-713" aria-hidden="true" tabindex="-1"></a>a histogram:</span>
<span id="cb27-714"><a href="#cb27-714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-717"><a href="#cb27-717" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-718"><a href="#cb27-718" aria-hidden="true" tabindex="-1"></a>frogs <span class="sc">|&gt;</span> </span>
<span id="cb27-719"><a href="#cb27-719" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log10</span>(S20_C_5))) <span class="sc">+</span></span>
<span id="cb27-720"><a href="#cb27-720" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>() </span>
<span id="cb27-721"><a href="#cb27-721" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-722"><a href="#cb27-722" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-723"><a href="#cb27-723" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-724"><a href="#cb27-724" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-725"><a href="#cb27-725" aria-hidden="true" tabindex="-1"></a>üé¨ Or a density plot:</span>
<span id="cb27-726"><a href="#cb27-726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-729"><a href="#cb27-729" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-730"><a href="#cb27-730" aria-hidden="true" tabindex="-1"></a>frogs <span class="sc">|&gt;</span> </span>
<span id="cb27-731"><a href="#cb27-731" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log10</span>(S20_C_5))) <span class="sc">+</span></span>
<span id="cb27-732"><a href="#cb27-732" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>()</span>
<span id="cb27-733"><a href="#cb27-733" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-734"><a href="#cb27-734" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-735"><a href="#cb27-735" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-736"><a href="#cb27-736" aria-hidden="true" tabindex="-1"></a>I‚Äôve used base 10 only because it is easy to convert to the original</span>
<span id="cb27-737"><a href="#cb27-737" aria-hidden="true" tabindex="-1"></a>scale (1 is 10, 2 is 100, 3 is 1000 etc). The warning about rows being</span>
<span id="cb27-738"><a href="#cb27-738" aria-hidden="true" tabindex="-1"></a>removed is expected - these are the counts of 0 since you can‚Äôt log a</span>
<span id="cb27-739"><a href="#cb27-739" aria-hidden="true" tabindex="-1"></a>value of 0. The peak at zero suggests quite a few counts of 1. We would</span>
<span id="cb27-740"><a href="#cb27-740" aria-hidden="true" tabindex="-1"></a>expect the distribution of counts to be roughly log normal because this</span>
<span id="cb27-741"><a href="#cb27-741" aria-hidden="true" tabindex="-1"></a>is expression of all the genes in the genome<span class="ot">[^2]</span>. That peak near the low</span>
<span id="cb27-742"><a href="#cb27-742" aria-hidden="true" tabindex="-1"></a>end is not expected for this distribution and suggests that these lower</span>
<span id="cb27-743"><a href="#cb27-743" aria-hidden="true" tabindex="-1"></a>counts might be anomalies.</span>
<span id="cb27-744"><a href="#cb27-744" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-745"><a href="#cb27-745" aria-hidden="true" tabindex="-1"></a><span class="ot">[^2]: </span>This a result of the&nbsp;[Central limit</span>
<span id="cb27-746"><a href="#cb27-746" aria-hidden="true" tabindex="-1"></a>    theorem](https://en.wikipedia.org/wiki/Central_limit_theorem), one</span>
<span id="cb27-747"><a href="#cb27-747" aria-hidden="true" tabindex="-1"></a>    consequence of which is that adding together lots of distributions -</span>
<span id="cb27-748"><a href="#cb27-748" aria-hidden="true" tabindex="-1"></a>    whatever distributions they are - will tend to a normal</span>
<span id="cb27-749"><a href="#cb27-749" aria-hidden="true" tabindex="-1"></a>    distribution.</span>
<span id="cb27-750"><a href="#cb27-750" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-751"><a href="#cb27-751" aria-hidden="true" tabindex="-1"></a>The excess number of low counts indicates we might want to create a cut</span>
<span id="cb27-752"><a href="#cb27-752" aria-hidden="true" tabindex="-1"></a>off for quality control. The removal of low counts is a common</span>
<span id="cb27-753"><a href="#cb27-753" aria-hidden="true" tabindex="-1"></a>processing step in ‚Äôomic data.</span>
<span id="cb27-754"><a href="#cb27-754" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-755"><a href="#cb27-755" aria-hidden="true" tabindex="-1"></a>If you have two groups of data, you can compare their distributions on</span>
<span id="cb27-756"><a href="#cb27-756" aria-hidden="true" tabindex="-1"></a>the same plot using the <span class="in">`fill`</span> aesthetic and making the fill transparent</span>
<span id="cb27-757"><a href="#cb27-757" aria-hidden="true" tabindex="-1"></a>with the <span class="in">`alpha`</span> argument.</span>
<span id="cb27-758"><a href="#cb27-758" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-759"><a href="#cb27-759" aria-hidden="true" tabindex="-1"></a>üé¨ To plot the distribution of <span class="in">`area`</span> in the <span class="in">`cell_bio`</span> dataframe as a</span>
<span id="cb27-760"><a href="#cb27-760" aria-hidden="true" tabindex="-1"></a>density plot, with the <span class="in">`clone`</span> variable as the fill:</span>
<span id="cb27-761"><a href="#cb27-761" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-764"><a href="#cb27-764" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-765"><a href="#cb27-765" aria-hidden="true" tabindex="-1"></a>cell_bio <span class="sc">|&gt;</span> </span>
<span id="cb27-766"><a href="#cb27-766" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> area, <span class="at">fill =</span> clone)) <span class="sc">+</span></span>
<span id="cb27-767"><a href="#cb27-767" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>)</span>
<span id="cb27-768"><a href="#cb27-768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-769"><a href="#cb27-769" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-770"><a href="#cb27-770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-771"><a href="#cb27-771" aria-hidden="true" tabindex="-1"></a>Boxplots and violin plots are also useful for visualising the</span>
<span id="cb27-772"><a href="#cb27-772" aria-hidden="true" tabindex="-1"></a>distribution of a variable, especially when there are more than about 5</span>
<span id="cb27-773"><a href="#cb27-773" aria-hidden="true" tabindex="-1"></a>groups.</span>
<span id="cb27-774"><a href="#cb27-774" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-775"><a href="#cb27-775" aria-hidden="true" tabindex="-1"></a>üé¨ To plot the distribution of <span class="in">`area`</span> in the <span class="in">`cell_bio`</span> dataframe as a</span>
<span id="cb27-776"><a href="#cb27-776" aria-hidden="true" tabindex="-1"></a>boxplot, with the <span class="in">`clone`</span> variable on the x-axis:</span>
<span id="cb27-777"><a href="#cb27-777" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-780"><a href="#cb27-780" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-781"><a href="#cb27-781" aria-hidden="true" tabindex="-1"></a>cell_bio <span class="sc">|&gt;</span> </span>
<span id="cb27-782"><a href="#cb27-782" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> clone, <span class="at">y =</span> area)) <span class="sc">+</span></span>
<span id="cb27-783"><a href="#cb27-783" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span>
<span id="cb27-784"><a href="#cb27-784" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-785"><a href="#cb27-785" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-786"><a href="#cb27-786" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-787"><a href="#cb27-787" aria-hidden="true" tabindex="-1"></a>Suppose you want to see the distribution of many similar variables such</span>
<span id="cb27-788"><a href="#cb27-788" aria-hidden="true" tabindex="-1"></a>as we have in <span class="in">`biotech`</span>. You cannot map a variable to <span class="in">`x`</span> or <span class="in">`fill`</span> as</span>
<span id="cb27-789"><a href="#cb27-789" aria-hidden="true" tabindex="-1"></a>we did in the two previous cases because the values are in *separate*</span>
<span id="cb27-790"><a href="#cb27-790" aria-hidden="true" tabindex="-1"></a>columns rather than than in *one* column with a second column giving a</span>
<span id="cb27-791"><a href="#cb27-791" aria-hidden="true" tabindex="-1"></a>group. However, we can use the <span class="in">`pivot_longer()`</span> function to reformat the</span>
<span id="cb27-792"><a href="#cb27-792" aria-hidden="true" tabindex="-1"></a>data and then pipe it into ggplot.</span>
<span id="cb27-793"><a href="#cb27-793" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-794"><a href="#cb27-794" aria-hidden="true" tabindex="-1"></a>üé¨ Plot the distribution of the logged counts in the <span class="in">`biotech`</span> dataframe</span>
<span id="cb27-795"><a href="#cb27-795" aria-hidden="true" tabindex="-1"></a>as a boxplot, the samples on the x axis:</span>
<span id="cb27-796"><a href="#cb27-796" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-799"><a href="#cb27-799" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-800"><a href="#cb27-800" aria-hidden="true" tabindex="-1"></a>biotech <span class="sc">|&gt;</span> </span>
<span id="cb27-801"><a href="#cb27-801" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>transcript,</span>
<span id="cb27-802"><a href="#cb27-802" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"sample"</span>,</span>
<span id="cb27-803"><a href="#cb27-803" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"count"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb27-804"><a href="#cb27-804" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> sample, <span class="at">y =</span> <span class="fu">log10</span>(count))) <span class="sc">+</span></span>
<span id="cb27-805"><a href="#cb27-805" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()</span>
<span id="cb27-806"><a href="#cb27-806" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-807"><a href="#cb27-807" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-808"><a href="#cb27-808" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-811"><a href="#cb27-811" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb27-812"><a href="#cb27-812" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb27-813"><a href="#cb27-813" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb27-814"><a href="#cb27-814" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-815"><a href="#cb27-815" aria-hidden="true" tabindex="-1"></a>This code takes the <span class="in">`biotech`</span> dataframe and transforms it dataset from</span>
<span id="cb27-816"><a href="#cb27-816" aria-hidden="true" tabindex="-1"></a>wide format to long format. It takes all columns except <span class="in">`transcript`</span></span>
<span id="cb27-817"><a href="#cb27-817" aria-hidden="true" tabindex="-1"></a>(<span class="in">`cols = -transcript`</span>) and gathers them into two new columns:</span>
<span id="cb27-818"><a href="#cb27-818" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-819"><a href="#cb27-819" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`"sample"`</span> ‚Üí Stores the original column names (<span class="in">`SCK14_1`</span>, <span class="in">`SCK14_2`</span>,</span>
<span id="cb27-820"><a href="#cb27-820" aria-hidden="true" tabindex="-1"></a>    etc.).</span>
<span id="cb27-821"><a href="#cb27-821" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`"count"`</span> ‚Üí Stores the values from those columns.</span>
<span id="cb27-822"><a href="#cb27-822" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-823"><a href="#cb27-823" aria-hidden="true" tabindex="-1"></a>Now, instead of multiple sample columns, we have a table like this:</span>
<span id="cb27-824"><a href="#cb27-824" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-825"><a href="#cb27-825" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> transcript <span class="pp">|</span> sample  <span class="pp">|</span> count <span class="pp">|</span></span>
<span id="cb27-826"><a href="#cb27-826" aria-hidden="true" tabindex="-1"></a><span class="pp">|------------|---------|-------|</span></span>
<span id="cb27-827"><a href="#cb27-827" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> MSTRG.1    <span class="pp">|</span> SCK14_1 <span class="pp">|</span> 560   <span class="pp">|</span></span>
<span id="cb27-828"><a href="#cb27-828" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> MSTRG.1    <span class="pp">|</span> SCK14_2 <span class="pp">|</span> 1526  <span class="pp">|</span></span>
<span id="cb27-829"><a href="#cb27-829" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> MSTRG.1    <span class="pp">|</span> SCK14_3 <span class="pp">|</span> 4168  <span class="pp">|</span></span>
<span id="cb27-830"><a href="#cb27-830" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> MSTRG.10   <span class="pp">|</span> SCK14_1 <span class="pp">|</span> 0     <span class="pp">|</span></span>
<span id="cb27-831"><a href="#cb27-831" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-832"><a href="#cb27-832" aria-hidden="true" tabindex="-1"></a>which is piped straight into ggplot. The nice thing about the pipe, is</span>
<span id="cb27-833"><a href="#cb27-833" aria-hidden="true" tabindex="-1"></a>that you need not create a new dataframe to do this. You can pipe the</span>
<span id="cb27-834"><a href="#cb27-834" aria-hidden="true" tabindex="-1"></a>output of <span class="in">`pivot_longer()`</span> directly into <span class="in">`ggplot()`</span>. This is very useful</span>
<span id="cb27-835"><a href="#cb27-835" aria-hidden="true" tabindex="-1"></a>when you are exploring the data and want to try out different</span>
<span id="cb27-836"><a href="#cb27-836" aria-hidden="true" tabindex="-1"></a>visualisations.</span>
<span id="cb27-837"><a href="#cb27-837" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-838"><a href="#cb27-838" aria-hidden="true" tabindex="-1"></a>You're finished!</span>
<span id="cb27-839"><a href="#cb27-839" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-840"><a href="#cb27-840" aria-hidden="true" tabindex="-1"></a><span class="fu"># ü•≥ Well Done! üéâ</span></span>
<span id="cb27-841"><a href="#cb27-841" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-842"><a href="#cb27-842" aria-hidden="true" tabindex="-1"></a><span class="fu"># Independent study following the workshop</span></span>
<span id="cb27-843"><a href="#cb27-843" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-844"><a href="#cb27-844" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">Consolidate</span><span class="co">](study_after_workshop.qmd)</span></span>
<span id="cb27-845"><a href="#cb27-845" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-846"><a href="#cb27-846" aria-hidden="true" tabindex="-1"></a><span class="fu"># The Code file</span></span>
<span id="cb27-847"><a href="#cb27-847" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-848"><a href="#cb27-848" aria-hidden="true" tabindex="-1"></a>All the answers (coding and thinking) are in the source code for the</span>
<span id="cb27-849"><a href="#cb27-849" aria-hidden="true" tabindex="-1"></a>workshop. You can view it using the <span class="in">`&lt;/&gt; Code`</span> button at the top of the</span>
<span id="cb27-850"><a href="#cb27-850" aria-hidden="true" tabindex="-1"></a>page. Coding and thinking answers are marked with <span class="in">`#---CODING ANSWER---`</span></span>
<span id="cb27-851"><a href="#cb27-851" aria-hidden="true" tabindex="-1"></a>and <span class="in">`#---THINKING ANSWER---`</span></span>
<span id="cb27-852"><a href="#cb27-852" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-853"><a href="#cb27-853" aria-hidden="true" tabindex="-1"></a>Pages made with R <span class="co">[</span><span class="ot">@R-core</span><span class="co">]</span>, Quarto <span class="co">[</span><span class="ot">@allaire2022</span><span class="co">]</span>, <span class="in">`knitr`</span> [@knitr1;</span>
<span id="cb27-854"><a href="#cb27-854" aria-hidden="true" tabindex="-1"></a>@knitr2; @knitr3], <span class="in">`kableExtra`</span> <span class="co">[</span><span class="ot">@kableExtra</span><span class="co">]</span></span>
<span id="cb27-855"><a href="#cb27-855" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-856"><a href="#cb27-856" aria-hidden="true" tabindex="-1"></a><span class="fu"># References</span></span>
</code></pre></div><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>Rand, E. (2025). Data Analysis in R for Becoming a Bioscientist (Version 0.2) <a href="https://github.com/3mmaRand/R4BABS" class="uri">https://github.com/3mmaRand/R4BABS</a></p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/3mmaRand/comp4biosci/edit/main/r4babs4/week-1/workshop.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li></ul></div></div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
<li class="nav-item">
    <a class="nav-link" href="https://github.com/3mmaRand">
<p>Emma on GitHub</p>
</a>
  </li>  
</ul>
</div>
  </div>
</footer><script>
document.addEventListener("DOMContentLoaded", function() {
  // grab all links under ‚ÄúOther Links‚Äù in the right-hand TOC
  const links = document.querySelectorAll('.quarto-other-links ul li a');
  links.forEach(function(a) {
    // find the one whose text is exactly ‚ÄúMake a suggestion‚Äù
    if (a.textContent.trim() === 'Make a suggestion') {
      const pageUrl   = encodeURIComponent(window.location.href);
      const pageTitle = encodeURIComponent(document.title);
      a.href = 'https://github.com/3mmaRand/R4BABS/issues/new'
             + '?title=' + pageTitle
             + '&body='  + pageUrl;
      a.target = '_blank';
    }
  });
});
</script><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>


</body></html>