<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.33">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Emma Rand">
<title>Workshop – Data Analysis in R</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-ea385d0e468b0dd5ea5bf0780b1290d9.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark-bc185b5c5bdbcb35c2eb49d8a876ef70.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-ea385d0e468b0dd5ea5bf0780b1290d9.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-72ac799c4abbddf31cb463dd8fc99f5d.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark-25ac7be22ffa75a8e8541e30a6dcc9a3.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../../site_libs/bootstrap/bootstrap-72ac799c4abbddf31cb463dd8fc99f5d.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-50FV9HPTRE"></script><script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-50FV9HPTRE', { 'anonymize_ip': true});
</script><script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script><script src="../../site_libs/kePrint-0.0.1/kePrint.js"></script><link href="../../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">
</head>
<body class="nav-sidebar floating nav-fixed quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Data Analysis in R</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Welcome!</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../r4babs1/r4babs1.html"> 
<span class="menu-text">BABS 1</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../r4babs2/r4babs2.html" aria-current="page"> 
<span class="menu-text">BABS 2</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../r4babs4/r4babs4.html"> 
<span class="menu-text">BABS 4</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../pgt52m/pgt52m.html"> 
<span class="menu-text">PGT 52M</span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
    <a href="https://github.com/3mmaRand" title="Emma on GitHub" class="quarto-navigation-tool px-1" aria-label="Emma on GitHub"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../r4babs2/week-3/overview.html">Week 3: Two-sample tests</a></li><li class="breadcrumb-item"><a href="../../r4babs2/week-3/workshop.html">Workshop</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/r4babs2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Analysis in R for BABS 2</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 1: The logic of hypothesis testing and CI</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-1/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-1/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-1/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-1/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 2: Introduction to statistical models: Single regression</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-2/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-2/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-2/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-2/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 3: Two-sample tests</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-3/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-3/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-3/workshop.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-3/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 4: One-way ANOVA and Kruskal-Wallis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-4/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-4/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-4/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-4/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 5: Two-way ANOVA</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-5/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-5/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-5/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-5/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="true">
 <span class="menu-text">Week 6: Association</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-6/overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">About</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-6/study_before_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Prepare!</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-6/workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../r4babs2/week-6/study_after_workshop.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Consolidate!</span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li>
<a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul class="collapse">
<li><a href="#session-overview" id="toc-session-overview" class="nav-link" data-scroll-target="#session-overview">Session overview</a></li>
  <li><a href="#philosophy" id="toc-philosophy" class="nav-link" data-scroll-target="#philosophy">Philosophy</a></li>
  </ul>
</li>
  <li><a href="#getting-started" id="toc-getting-started" class="nav-link" data-scroll-target="#getting-started">Getting started</a></li>
  <li>
<a href="#exercises" id="toc-exercises" class="nav-link" data-scroll-target="#exercises">Exercises</a>
  <ul class="collapse">
<li>
<a href="#adiponectin-secretion" id="toc-adiponectin-secretion" class="nav-link" data-scroll-target="#adiponectin-secretion">Adiponectin secretion</a>
  <ul class="collapse">
<li><a href="#summarising-the-data" id="toc-summarising-the-data" class="nav-link" data-scroll-target="#summarising-the-data">Summarising the data</a></li>
  <li><a href="#selecting-a-test" id="toc-selecting-a-test" class="nav-link" data-scroll-target="#selecting-a-test">Selecting a test</a></li>
  <li><a href="#applying-interpreting-and-reporting" id="toc-applying-interpreting-and-reporting" class="nav-link" data-scroll-target="#applying-interpreting-and-reporting">Applying, interpreting and reporting</a></li>
  <li><a href="#check-assumptions" id="toc-check-assumptions" class="nav-link" data-scroll-target="#check-assumptions">Check assumptions</a></li>
  <li><a href="#illustrating" id="toc-illustrating" class="nav-link" data-scroll-target="#illustrating">Illustrating</a></li>
  </ul>
</li>
  <li>
<a href="#grouse-parasites" id="toc-grouse-parasites" class="nav-link" data-scroll-target="#grouse-parasites">Grouse Parasites</a>
  <ul class="collapse">
<li><a href="#selecting" id="toc-selecting" class="nav-link" data-scroll-target="#selecting">Selecting</a></li>
  <li><a href="#applying-interpreting-and-reporting-1" id="toc-applying-interpreting-and-reporting-1" class="nav-link" data-scroll-target="#applying-interpreting-and-reporting-1">Applying, interpreting and reporting</a></li>
  <li><a href="#illustrating-1" id="toc-illustrating-1" class="nav-link" data-scroll-target="#illustrating-1">Illustrating</a></li>
  </ul>
</li>
  <li>
<a href="#gene-expression" id="toc-gene-expression" class="nav-link" data-scroll-target="#gene-expression">Gene Expression</a>
  <ul class="collapse">
<li><a href="#selecting-1" id="toc-selecting-1" class="nav-link" data-scroll-target="#selecting-1">Selecting</a></li>
  <li><a href="#applying-interpreting-and-reporting-2" id="toc-applying-interpreting-and-reporting-2" class="nav-link" data-scroll-target="#applying-interpreting-and-reporting-2">Applying, interpreting and reporting</a></li>
  </ul>
</li>
  <li><a href="#look-after-future-you" id="toc-look-after-future-you" class="nav-link" data-scroll-target="#look-after-future-you">Look after future you!</a></li>
  </ul>
</li>
  <li><a href="#well-done" id="toc-well-done" class="nav-link" data-scroll-target="#well-done">🥳 Well Done! 🎉</a></li>
  <li><a href="#independent-study-following-the-workshop" id="toc-independent-study-following-the-workshop" class="nav-link" data-scroll-target="#independent-study-following-the-workshop">Independent study following the workshop</a></li>
  <li><a href="#the-code-file" id="toc-the-code-file" class="nav-link" data-scroll-target="#the-code-file">The Code file</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/3mmaRand/comp4biosci/edit/main/r4babs2/week-3/workshop.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li></ul></div><div class="quarto-other-links"><h2>Other Links</h2><ul><li><a href="#"><i class="bi bi-chat-left-dots"></i>Make a suggestion</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../r4babs2/week-3/overview.html">Week 3: Two-sample tests</a></li><li class="breadcrumb-item"><a href="../../r4babs2/week-3/workshop.html">Workshop</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Workshop</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">Two-sample tests</p>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><section id="introduction" class="level1"><h1>Introduction</h1>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="images/how-much-i-know.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Artwork by @allison_horst: “How much I think I know about R”"><img src="images/how-much-i-know.png" class="img-fluid figure-img" alt="Illustrated line plot of 'How much I think I know about R' on the y-axis, and 'Time' on the x-axis. Along the line are emoji-style faces, showing the non-linear progression of R knowledge over time. At first, a nervous face becomes a happy face early on in learning, then a grimace face at an intermediate peak before a steep decline (with an exhausted face at the local minimum). Then, a determined face charges back up a hill, reaching another peak with a mind-blown face and text annotation 'join R community on twitter' followed by another decline, but this time the faces look happy even though their 'How much I think I know about R' value is declining." width="800"></a></p>
<figcaption>Artwork by <span class="citation" data-cites="allison_horst">Horst (<a href="#ref-allison_horst" role="doc-biblioref">2023</a>)</span>: “How much I think I know about R”</figcaption></figure>
</div>
<section id="session-overview" class="level2"><h2 class="anchored" data-anchor-id="session-overview">Session overview</h2>
<p>In this workshop you will get practice in choosing between, performing, and presenting the results of, two-sample tests and their non-parametric equivalents in R.</p>
</section><section id="philosophy" class="level2"><h2 class="anchored" data-anchor-id="philosophy">Philosophy</h2>
<p>Workshops are not a test. It is expected that you often don’t know how to start, make a lot of mistakes and need help. It is expected that you are familiar with independent study content before the workshop. However, you need not remember or understand every detail as the workshop should build and consolidate your understanding. Tips</p>
<ul>
<li>don’t worry about making mistakes</li>
<li>don’t let what you can not do interfere with what you can do</li>
<li>discussing code with your neighbours will help</li>
<li>look things up in the independent study material</li>
<li>look things up in your own code from earlier</li>
<li>there are no stupid questions</li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Key
</div>
</div>
<div class="callout-body-container callout-body">
<p>These four symbols are used at the beginning of each instruction so you know where to carry out the instruction.</p>
<p><img src="images/do_on_your_computer.png" class="img-fluid"> Something you need to do on your computer. It may be opening programs or documents or locating a file.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Something you should do in RStudio. It will often be typing a command or using the menus but might also be creating folders, locating or moving files.</p>
<p><img src="images/do_on_internet.png" class="img-fluid"> Something you should do in your browser on the internet. It may be searching for information, going to the VLE or downloading a file.</p>
<p><img src="images/answer.png" class="img-fluid"> A question for you to think about and answer. Record your answers in your script for future reference.</p>
</div>
</div>
</section></section><section id="getting-started" class="level1"><h1>Getting started</h1>
<p><img src="images/do_on_your_computer.png" class="img-fluid"> Start RStudio from the Start menu.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Make an RStudio project for this workshop by clicking on the drop-down menu on top right where it says <code>Project: (None)</code> and choosing New Project, then New Directory, then New Project. Navigate to the <code>data-analysis-in-r-2</code> folder and name the RStudio Project <code>week-3</code>.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Make new folders called <code>data-raw</code> and <code>figures</code>. You can do this on the Files Pane by clicking New Folder and typing into the box that appears.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Make a new script then save it with a name like <code>two-sample-tests.R</code> to carry out the rest of the work.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Add a comment to the script: <code># Two-sample tests</code> and load the <strong><code>tidyverse</code></strong> <span class="citation" data-cites="tidyverse">(<a href="#ref-tidyverse" role="doc-biblioref">Wickham et al. 2019</a>)</span> package</p>
</section><section id="exercises" class="level1"><h1>Exercises</h1>
<section id="adiponectin-secretion" class="level2"><h2 class="anchored" data-anchor-id="adiponectin-secretion">Adiponectin secretion</h2>
<p>Adiponectin is exclusively secreted from adipose tissue and modulates a number of metabolic processes. Nicotinic acid can affect adiponectin secretion. 3T3-L1 adipocytes were treated with nicotinic acid or with a control treatment and adiponectin concentration (pg/mL) measured in the medium. The data are in <a href="data-raw/adipocytes.txt">adipocytes.txt</a>. Each row represents an independent sample of adipocytes and the first column gives the concentration of adiponectin and the second column indicates whether they were treated with nicotinic acid or not (the control).</p>
<p><img src="images/do_on_your_computer.png" class="img-fluid"> Save a copy of <a href="data-raw/adipocytes.txt">adipocytes.txt</a> to <code>data-raw</code></p>
<p><img src="images/do_in_R.png" class="img-fluid"> Read in the data and check the structure. I used the name <code>adip</code> for the dataframe/tibble.</p>
<p>We have a tibble containing two variables: <code>adiponectin</code> is the response variable and is continuous; <code>treatment</code> is explanatory. <code>treatment</code> is categorical with two levels (groups). The first task is visualise the data to get an overview. For continuous response variables with categorical explanatory variables you could use <code>geom_point()</code>, <code>geom_boxplot()</code> or a variety of other geoms. I often use <code>geom_violin()</code> which allows us to see the distribution - the violin is fatter where there are more data points.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Do a quick plot of the data:</p>
<div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">adip</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">treatment</span>, y <span class="op">=</span> <span class="va">adiponectin</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_violin</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-3-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2"><img src="workshop_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<section id="summarising-the-data" class="level3"><h3 class="anchored" data-anchor-id="summarising-the-data">Summarising the data</h3>
<p>Summarising the data for each treatment group is the next sensible step. The most useful summary statistics are the means, standard deviations, sample sizes and standard errors.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Create a data frame called <code>adip_summary</code> that contains the means, standard deviations, sample sizes and standard errors for the control and nicotinic acid treated samples. You may need to read the <a href="../../r4babs1/week-9/workshop.html#summarise">Summarise section</a> from the Week 9 workshop of BABS1 <span class="citation" data-cites="rand2023">(<a href="#ref-rand2023" role="doc-biblioref">Rand 2023</a>)</span></p>
<p>You should get the following numbers:</p>
<div class="cell">
<div class="cell-output-display">
<table class="table caption-top table-sm table-striped small" data-quarto-postprocess="true">
<thead><tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">treatment</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">mean</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">std</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">n</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">se</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">control</td>
<td style="text-align: right;">5.546000</td>
<td style="text-align: right;">1.475247</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">0.3809072</td>
</tr>
<tr class="even">
<td style="text-align: left;">nicotinic</td>
<td style="text-align: right;">7.508667</td>
<td style="text-align: right;">1.793898</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">0.4631824</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<p>Remember that you can see the contents of a dataframe with any of these:</p>
<ul>
<li>type the name of the dataframe in the console: prints the dataframe to console</li>
<li>click on the dataframe name in the Environment window: opens a spreadsheet like view of the dataframe in the Code window</li>
<li>typing <code>View(dataframe)</code> in the console: opens a spreadsheet like view of the dataframe in the Code window</li>
</ul>
</div>
</div>
</section><section id="selecting-a-test" class="level3"><h3 class="anchored" data-anchor-id="selecting-a-test">Selecting a test</h3>
<p><img src="images/answer.png" class="img-fluid"> Do you think this is a paired-sample test or two-sample test?</p>
<!-- #---THINKING ANSWER---  -->
<!-- Two-sample - the samples are either control or nicotinic acid treated, not both.  -->
<!-- There is no link between the first control value and the first nicotinic acid  -->
<!-- value. The two samples are independent and it is a two-sample -->
</section><section id="applying-interpreting-and-reporting" class="level3"><h3 class="anchored" data-anchor-id="applying-interpreting-and-reporting">Applying, interpreting and reporting</h3>
<p><img src="images/do_in_R.png" class="img-fluid"> Create a two-sample model like this:</p>
<div class="cell">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">adip</span>,</span>
<span>          <span class="va">adiponectin</span> <span class="op">~</span> <span class="va">treatment</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><img src="images/do_in_R.png" class="img-fluid"> Examine the model with:</p>
<div class="cell">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = adiponectin ~ treatment, data = adip)

Residuals:
    Min      1Q  Median      3Q     Max 
-4.3787 -1.0967  0.1927  1.0245  3.1113 

Coefficients:
                   Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)          5.5460     0.4240  13.079  1.9e-13 ***
treatmentnicotinic   1.9627     0.5997   3.273  0.00283 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.642 on 28 degrees of freedom
Multiple R-squared:  0.2767,    Adjusted R-squared:  0.2509 
F-statistic: 10.71 on 1 and 28 DF,  p-value: 0.00283</code></pre>
</div>
</div>
<p><img src="images/answer.png" class="img-fluid"> What do you conclude from the test? Write your conclusion in a form suitable for a report.</p>
<!-- #---THINKING ANSWER---  -->
<!-- Adipocytes treated with nicotinic acid (mean +/- s.e.: 7.51 +/- 0.38 pg/mL)  -->
<!-- secrete significantly more adiponectin than controls (5.55  +/- 0.46 pg/mL)  -->
<!-- (t = 3.27; d.f. = 28; p = 0.003). -->
</section><section id="check-assumptions" class="level3"><h3 class="anchored" data-anchor-id="check-assumptions">Check assumptions</h3>
<p>The assumptions of the general linear model are that the residuals – the difference between predicted value (<em>i.e</em>., the group mean) and observed values - are normally distributed, and have homogeneous variance. To check these we can examine the <code>mod$residuals</code> variable. You may want to refer to <a href="../week-2/workshop.html#checking-assumptions">Checking assumptions</a> in the “Single regression” workshop.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Plot the model residuals against the fitted values.</p>
<p><img src="images/answer.png" class="img-fluid"> What to you conclude?</p>
<!-- #---THINKING ANSWER--- -->
<!-- The residuals look 'homogeneously' distributed. That is, the variance is the same  -->
<!-- for the three groups  -->
<p>To examine normality of the model residuals we can plot them as a histogram and do a normality test on them.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Plot a histogram of the residuals.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Use the <code><a href="https://rdrr.io/r/stats/shapiro.test.html">shapiro.test()</a></code> to test the normality of the model residuals</p>
<p><img src="images/answer.png" class="img-fluid"> What to you conclude?</p>
<!-- #---THINKING ANSWER--- -->
<!-- the residuals appear roughly normally distributed.  -->
<!-- The histogram is symmetrical and the shapiro test is NS. -->
<!-- Overall, it seems the assumptions are met -->
</section><section id="illustrating" class="level3"><h3 class="anchored" data-anchor-id="illustrating">Illustrating</h3>
<p><img src="images/do_in_R.png" class="img-fluid"> Create a figure like the one below. You may need to refer to <a href="../../r4babs1/week-9/workshop.html#visualise">Visualise</a> from the “Summarising data with several variables” workshop <span class="citation" data-cites="rand2023">(<a href="#ref-rand2023" role="doc-biblioref">Rand 2023</a>)</span></p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-11-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="workshop_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<p>We now need to annotate the figure with the results from the statistical test. This most commonly done with a line linking the means being compared and the <em>p</em>-value. The <code>annotate()</code> function can be used to draw the line and then to add the value. The line is a <code>segment</code> and the <em>p</em>-value is a <code>text</code>.</p>
<section id="sec-week-3-annotation-method-1" class="level4"><h4 class="anchored" data-anchor-id="sec-week-3-annotation-method-1">Annotation - method 1</h4>
<p>You can annotate the <em>p</em>-value manually by typing it in to the appropriate <code>annotate()</code> function.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Add annotation to the figure by adding:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">......</span> <span class="op">+</span></span>
<span><span class="fu">annotate</span><span class="op">(</span><span class="st">"segment"</span>, x <span class="op">=</span> <span class="fl">1</span>, xend <span class="op">=</span> <span class="fl">2</span>, </span>
<span>         y <span class="op">=</span> <span class="fl">11.3</span>, yend <span class="op">=</span> <span class="fl">11.3</span>,</span>
<span>         colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu">annotate</span><span class="op">(</span><span class="st">"text"</span>, x <span class="op">=</span> <span class="fl">1.5</span>,  y <span class="op">=</span> <span class="fl">11.7</span>, </span>
<span>         label <span class="op">=</span> <span class="st">"italic(p) == 0.0028"</span>,</span>
<span>         parse <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span> <span class="va">.....</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li>
<code>annotate("segment", ...)</code> adds a line to the figure starting at <em>x = 1</em> and <em>y = 11.3</em> and ending at <em>x = 2</em> and <em>y = 11.3</em>
</li>
<li>
<code>annotate("text", ...)</code> adds text to the figure at <em>x = 1.5</em> and <em>y = 11.7</em>
</li>
<li>the <code>label</code> is written in R’s plotmath syntax rather than being a normal string.
<ul>
<li>
<code>italic(p)</code> renders the letter <em>p</em> in italics</li>
<li>
<code>==</code> is the plotmath symbol for “=”</li>
</ul>
</li>
<li>
<code>parse = TRUE</code> tells ggplot2 to interpret that string as a plotmath expression rather than literally adding “italic(p) == 0.0028”.</li>
</ul>
<p>However, it is more reproducible to extract the <em>p</em>-value from the model object.</p>
</section><section id="sec-week-3-annotation-method-2" class="level4"><h4 class="anchored" data-anchor-id="sec-week-3-annotation-method-2">Annotation - method 2</h4>
<p><img src="images/do_in_R.png" class="img-fluid"> Extract the <em>p</em>-value from the model object:</p>
<div class="cell">
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p_value</span> <span class="op">&lt;-</span> <span class="fu">broom</span><span class="fu">::</span><span class="fu"><a href="https://generics.r-lib.org/reference/glance.html">glance</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>
<strong>`broom</strong> is a package that turns model outputs into more easy to use tidy formats</li>
<li>
<strong><code>glance(mod)</code></strong> puts information from mod including the <em>p</em>-value into a dataframe</li>
<li>
<code>pull(p.value)</code> extracts the value from <em>p</em>-value column in the dataframe</li>
<li>
<code>round(4)</code> rounds the <em>p</em>-value to 4 decimal places</li>
</ul>
<p>The segment and text can then be added to the figure.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Add annotation to the figure by adding:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>...... <span class="sc">+</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>, </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="fl">11.3</span>, <span class="at">yend =</span> <span class="fl">11.3</span>,</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>,  <span class="at">y =</span> <span class="fl">11.7</span>, </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"italic(p) == { p_value }"</span>),</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">parse =</span> <span class="cn">TRUE</span>)<span class="er">)</span> <span class="sc">+</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>.....</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li>The glue package allows us to use R objects in strings so their v values are printed.</li>
<li>We use curly braces <code><a href="https://rdrr.io/r/base/Paren.html">{}</a></code> a round the R object so it is understood as an R object and its value is written rather than literally writing “p_value”</li>
</ul>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-13-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4"><img src="workshop_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<p><img src="images/do_in_R.png" class="img-fluid"> Save your figure to your figures folder. Make sure you <a href="../../r4babs1/week-9/workshop.html#writing-figures-to-file">script figure saving with <code>ggsave()</code></a>.</p>
</section></section></section><section id="grouse-parasites" class="level2"><h2 class="anchored" data-anchor-id="grouse-parasites">Grouse Parasites</h2>
<p>Grouse livers were dissected and the number of individuals of a parasitic nematode were counted for two estates ‘Gordon’ and ‘Moss’. We want to know if the two estates have different infection rates. The data are in <a href="data-raw/grouse.csv">grouse.csv</a></p>
<p><img src="images/do_on_your_computer.png" class="img-fluid"> Save a copy of <a href="data-raw/grouse.csv">grouse.csv</a> to <code>data-raw</code></p>
<p><img src="images/do_in_R.png" class="img-fluid"> Read in the data and check the structure. I used the name <code>grouse</code> for the dataframe/tibble.</p>
<section id="selecting" class="level3"><h3 class="anchored" data-anchor-id="selecting">Selecting</h3>
<p><img src="images/answer.png" class="img-fluid"> Using your common sense, do these data look normally distributed?</p>
<!-- #---THINKING ANSWER---  -->
<!-- No and little need to test/plot - they're counts, skewed and  -->
<!-- have unequal variances -->
<p><img src="images/answer.png" class="img-fluid"> What test do you suggest?</p>
<!-- #---THINKING ANSWER---  -->
<!-- two-sample Wilcoxon (unpaired) AKA Mann-Whitney -->
</section><section id="applying-interpreting-and-reporting-1" class="level3"><h3 class="anchored" data-anchor-id="applying-interpreting-and-reporting-1">Applying, interpreting and reporting</h3>
<p><img src="images/do_in_R.png" class="img-fluid"> Summarise the data by finding the median of each group:</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Carry out a two-sample Wilcoxon test (also known as a Mann-Whitney):</p>
<div class="cell">
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/wilcox.test.html">wilcox.test</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">grouse</span>, <span class="va">nematodes</span> <span class="op">~</span> <span class="va">estate</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Wilcoxon rank sum exact test

data:  nematodes by estate
W = 78, p-value = 0.03546
alternative hypothesis: true location shift is not equal to 0</code></pre>
</div>
</div>
<p><img src="images/answer.png" class="img-fluid"> What do you conclude from the test? Write your conclusion in a form suitable for a report.</p>
<!-- #---THINKING ANSWER---  -->
<!-- Grouse on Gordon had significantly more nematodes (median = 13.5) than those  -->
<!-- on Moss (median = 5) (W = 78; n1 = 10, n2 = 10; p = 0.03546). -->
</section><section id="illustrating-1" class="level3"><h3 class="anchored" data-anchor-id="illustrating-1">Illustrating</h3>
<p>A box plot is a usually good choice for illustrating a two-sample Wilcoxon test because it shows the median and interquartile range.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> We can create a simple boxplot with:</p>
<div class="cell">
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span>data <span class="op">=</span> <span class="va">grouse</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">estate</span>, y <span class="op">=</span> <span class="va">nematodes</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_boxplot</span><span class="op">(</span><span class="op">)</span> </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><a href="workshop_files/figure-html/unnamed-chunk-19-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5"><img src="workshop_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="4200"></a></p>
</figure>
</div>
</div>
</div>
<p><img src="images/do_in_R.png" class="img-fluid"> Annotate and format the figure so it is more suitable for a report and save it to your figures folder.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<p>If you want to use method 2 for annotating the <em>p</em>-value, you can extract the <em>p</em>-value with</p>
<div class="cell">
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p_value</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/wilcox.test.html">wilcox.test</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">grouse</span>, <span class="va">nematodes</span> <span class="op">~</span> <span class="va">estate</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">broom</span><span class="fu">::</span><span class="fu"><a href="https://generics.r-lib.org/reference/tidy.html">tidy</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">p.value</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
</section></section><section id="gene-expression" class="level2"><h2 class="anchored" data-anchor-id="gene-expression">Gene Expression</h2>
<p>Bambara groundnut (<em>Vigna subterranea</em>) is an African legume with good nutritional value which can be influenced by low temperature stress. Researchers are interested in the expression levels of a particular set of 35 genes (<code>probe_id</code>) in response to temperature stress. They measure the expression of the genes at 23 and 18 degrees C (high and low temperature). These samples are <strong>not</strong> independent because we have two measure from one gene. The data are in <a href="data-raw/expr.xlsx">expr.xlsx</a>.</p>
<section id="selecting-1" class="level3"><h3 class="anchored" data-anchor-id="selecting-1">Selecting</h3>
<p><img src="images/answer.png" class="img-fluid"> What is the null hypothesis?</p>
<!-- #---THINKING ANSWER---  -->
<!-- expression is not affected by temperature / for each gene,  -->
<!-- the expression at the higher t   = expression at the lower t -->
<p><img src="images/do_in_R.png" class="img-fluid"> Save a copy of <a href="data-raw/expr.xlsx">expr.xlsx</a> and import the data. I named the dataframe <code>bambara</code></p>
<p><img src="images/answer.png" class="img-fluid"> What is the appropriate parametric test?</p>
<!-- #---THINKING ANSWER--- -->
<!-- the appropriate parametric test is the paired sample lm  -->
</section><section id="applying-interpreting-and-reporting-2" class="level3"><h3 class="anchored" data-anchor-id="applying-interpreting-and-reporting-2">Applying, interpreting and reporting</h3>
<p>A paired test requires account for the variation between genes.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> Create a paired-sample model like this:</p>
<div class="cell">
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">bambara</span>, </span>
<span>          <span class="va">expression</span> <span class="op">~</span> <span class="va">temperature</span> <span class="op">+</span> <span class="va">probe_id</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><img src="images/do_in_R.png" class="img-fluid"> Examine the model with:</p>
<div class="cell">
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = expression ~ temperature + probe_id, data = bambara)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.5345 -0.2094  0.0000  0.2094  0.5345 

Coefficients:
                                Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                      2.00019    0.28773   6.952 5.15e-08 ***
temperaturelowert               -0.30728    0.09591  -3.204  0.00294 ** 
probe_idGma.12117.1.S1_at        0.11320    0.40122   0.282  0.77955    
probe_idGma.12565.1.A1_at        1.31450    0.40122   3.276  0.00243 ** 
probe_idGma.12641.1.A1_at        0.41440    0.40122   1.033  0.30896    
probe_idGma.12911.1.A1_s_at      0.21940    0.40122   0.547  0.58806    
probe_idGma.13045.1.S1_at        0.02165    0.40122   0.054  0.95728    
probe_idGma.1679.1.A1_at         0.15610    0.40122   0.389  0.69965    
probe_idGma.17385.1.S1_at        0.35280    0.40122   0.879  0.38540    
probe_idGma.2266.1.S1_s_at      -0.27105    0.40122  -0.676  0.50388    
probe_idGma.2555.1.S1_s_at      -0.74210    0.40122  -1.850  0.07308 .  
probe_idGma.4452.1.A1_s_at      -0.39600    0.40122  -0.987  0.33062    
probe_idGma.5163.2.S1_at         0.69635    0.40122   1.736  0.09169 .  
probe_idGma.6423.1.S1_at         0.09230    0.40122   0.230  0.81943    
probe_idGma.8525.1.S1_s_at       0.87865    0.40122   2.190  0.03549 *  
probe_idGmaAffx.19961.1.S1_at    0.38790    0.40122   0.967  0.34047    
probe_idGmaAffx.24949.1.S1_at    0.17475    0.40122   0.436  0.66592    
probe_idGmaAffx.30443.1.S1_at    0.36625    0.40122   0.913  0.36775    
probe_idGmaAffx.33657.1.S1_at   -0.37655    0.40122  -0.939  0.35460    
probe_idGmaAffx.36315.1.S1_at    1.80255    0.40122   4.493 7.73e-05 ***
probe_idGmaAffx.37672.1.A1_at    0.60275    0.40122   1.502  0.14225    
probe_idGmaAffx.39830.1.S1_at    0.11055    0.40122   0.276  0.78457    
probe_idGmaAffx.44319.1.S1_at    0.45780    0.40122   1.141  0.26183    
probe_idGmaAffx.52122.1.S1_at    1.17505    0.40122   2.929  0.00604 ** 
probe_idGmaAffx.55864.1.S1_at   -0.40805    0.40122  -1.017  0.31632    
probe_idGmaAffx.5705.1.S1_at     0.81750    0.40122   2.038  0.04944 *  
probe_idGmaAffx.59856.1.S1_at   -0.22190    0.40122  -0.553  0.58384    
probe_idGmaAffx.67021.2.S1_at    0.22690    0.40122   0.566  0.57543    
probe_idGmaAffx.68483.1.S1_at    0.81930    0.40122   2.042  0.04897 *  
probe_idGmaAffx.7166.1.S1_at    -1.01450    0.40122  -2.529  0.01626 *  
probe_idGmaAffx.77709.1.S1_at    0.72625    0.40122   1.810  0.07913 .  
probe_idGmaAffx.77888.1.S1_at    0.17170    0.40122   0.428  0.67139    
probe_idGmaAffx.81915.1.S1_at    0.64390    0.40122   1.605  0.11777    
probe_idGmaAffx.8717.1.S1_at     0.93150    0.40122   2.322  0.02638 *  
probe_idGmaAffx.93493.1.S1_s_at  0.37840    0.40122   0.943  0.35227    
probe_idGmaAffx.9536.1.S1_at    -0.61725    0.40122  -1.538  0.13320    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.4012 on 34 degrees of freedom
Multiple R-squared:  0.8238,    Adjusted R-squared:  0.6424 
F-statistic: 4.542 on 35 and 34 DF,  p-value: 1.273e-05</code></pre>
</div>
</div>
<p>The “(Intercept) Estimate” will be the average of the higher temperature expression because <code>h</code> come before <code>l</code> in the alphabet. The “temperaturelowert” is how much the expression changes on average going from higher to lower temperatures. You can ignore the rest of the output.</p>
<p><img src="images/do_in_R.png" class="img-fluid"> If you prefer, you can use the <code><a href="https://rdrr.io/r/stats/anova.html">anova()</a></code> function on the model object:</p>
<div class="cell">
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Analysis of Variance Table

Response: expression
            Df  Sum Sq Mean Sq F value    Pr(&gt;F)    
temperature  1  1.6524 1.65240 10.2649  0.002943 ** 
probe_id    34 23.9354 0.70398  4.3732 2.079e-05 ***
Residuals   34  5.4732 0.16098                      
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>This presents you with less information. The <em>p</em>-value for the effect of temperate on gene expression will always be the same as in the larger table.</p>
<p><img src="images/answer.png" class="img-fluid"> State your conclusion from the test in a form suitable for including in a report. Make sure you give the direction of any significant effect.</p>
<!-- #---THINKING ANSWER--- -->
<!-- The expression of individual genes is significantly  -->
<!-- higher at higher temperatures than lower temperatures  -->
<!-- (t = 3.20, d.f. = 34, p = 0.0029) with an average difference -->
<!-- of 0.307 units. -->
<!-- You could instead quote (F = 10.3; d.f. = 1, 34, ; p = 0.0029) -->
</section></section><section id="look-after-future-you" class="level2"><h2 class="anchored" data-anchor-id="look-after-future-you">Look after future you!</h2>
<p>The code required to summarise, test, and plot data for a two-independent-samples test AND for a one-way ANOVA is <strong>exactly the same</strong> except for the names of the dataframe, variables and the axis labels and limits. Take some time to comment it your code so that you can make use of it next week.</p>
<p><a href="images/future_you.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6"><img src="images/future_you.png" class="img-fluid" alt="xxx" width="300"></a></p>
<p>You’re finished!</p>
</section></section><section id="well-done" class="level1"><h1>🥳 Well Done! 🎉</h1>
</section><section id="independent-study-following-the-workshop" class="level1"><h1>Independent study following the workshop</h1>
<p><a href="../../r4babs2/week-3/study_after_workshop.html">Consolidate</a></p>
</section><section id="the-code-file" class="level1"><h1>The Code file</h1>
<p>This contains all the code needed in the workshop even where it is not visible on the webpage.</p>
<p>The <code>workshop.qmd</code> file is the file I use to compile the practical. Qmd stands for Quarto markdown. It allows code and ordinary text to be interweaved to produce well-formatted reports including webpages. View the source code for this workshop using the <code>&lt;/&gt; Code</code> button at the top of the page. Coding and thinking answers are marked with <code>#---CODING ANSWER---</code> and <code>#---THINKING ANSWER---</code></p>
<p>Pages made with R <span class="citation" data-cites="R-core">(<a href="#ref-R-core" role="doc-biblioref">R Core Team 2024</a>)</span>, Quarto <span class="citation" data-cites="allaire2022">(<a href="#ref-allaire2022" role="doc-biblioref">Allaire et al. 2022</a>)</span>, <code>knitr</code> <span class="citation" data-cites="knitr1 knitr2 knitr3">(<a href="#ref-knitr1" role="doc-biblioref">Xie 2024</a>, <a href="#ref-knitr2" role="doc-biblioref">2015</a>, <a href="#ref-knitr3" role="doc-biblioref">2014</a>)</span>, <code>kableExtra</code> <span class="citation" data-cites="kableExtra">(<a href="#ref-kableExtra" role="doc-biblioref">Zhu 2024</a>)</span></p>
</section><section id="references" class="level1">


<!-- -->


</section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-allaire2022" class="csl-entry" role="listitem">
Allaire, J. J., Charles Teague, Carlos Scheidegger, Yihui Xie, and Christophe Dervieux. 2022. <em>Quarto</em>. <a href="https://doi.org/10.5281/zenodo.5960048">https://doi.org/10.5281/zenodo.5960048</a>.
</div>
<div id="ref-allison_horst" class="csl-entry" role="listitem">
Horst, Allison. 2023. <span>“Data Science Illustrations.”</span> <a href="https://allisonhorst.com/allison-horst">https://allisonhorst.com/allison-horst</a>.
</div>
<div id="ref-R-core" class="csl-entry" role="listitem">
R Core Team. 2024. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>.
</div>
<div id="ref-rand2023" class="csl-entry" role="listitem">
Rand, Emma. 2023. <em>Data Analysis in r for Becoming a Bioscientist</em>. <a href="https://3mmarand.github.io/R4BABS/">https://3mmarand.github.io/R4BABS/</a>.
</div>
<div id="ref-tidyverse" class="csl-entry" role="listitem">
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. <span>“Welcome to the <span></span>Tidyverse<span></span>”</span> 4: 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>.
</div>
<div id="ref-knitr3" class="csl-entry" role="listitem">
Xie, Yihui. 2014. <span>“Knitr: A Comprehensive Tool for Reproducible Research in <span>R</span>.”</span> In <em>Implementing Reproducible Computational Research</em>, edited by Victoria Stodden, Friedrich Leisch, and Roger D. Peng. Chapman; Hall/CRC.
</div>
<div id="ref-knitr2" class="csl-entry" role="listitem">
———. 2015. <em>Dynamic Documents with <span>R</span> and Knitr</em>. 2nd ed. Boca Raton, Florida: Chapman; Hall/CRC. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
<div id="ref-knitr1" class="csl-entry" role="listitem">
———. 2024. <em>Knitr: A General-Purpose Package for Dynamic Report Generation in r</em>. <a href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
<div id="ref-kableExtra" class="csl-entry" role="listitem">
Zhu, Hao. 2024. <em>kableExtra: Construct Complex Table with ’Kable’ and Pipe Syntax</em>. <a href="https://CRAN.R-project.org/package=kableExtra">https://CRAN.R-project.org/package=kableExtra</a>.
</div>
</div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("^(?:http:|https:)\/\/3mmarand\.github\.io\/R4BABS");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb17" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Workshop"</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Two-sample tests"</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="an">toc:</span><span class="co"> true</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="an">toc-location:</span><span class="co"> right</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a><span class="fu"># Introduction</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>![Artwork by @allison_horst: "How much I think I know about</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>R"](images/how-much-i-know.png){fig-alt="Illustrated line plot of 'How much I think I know about R' on the y-axis, and 'Time' on the x-axis. Along the line are emoji-style faces, showing the non-linear progression of R knowledge over time. At first, a nervous face becomes a happy face early on in learning, then a grimace face at an intermediate peak before a steep decline (with an exhausted face at the local minimum). Then, a determined face charges back up a hill, reaching another peak with a mind-blown face and text annotation 'join R community on twitter' followed by another decline, but this time the faces look happy even though their 'How much I think I know about R' value is declining."</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>width="800"}</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a><span class="fu">## Session overview</span></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>In this workshop you will get practice in choosing between, performing,</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>and presenting the results of, two-sample tests and their non-parametric</span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>equivalents in R.</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a><span class="fu">## Philosophy</span></span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>Workshops are not a test. It is expected that you often don't know how</span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>to start, make a lot of mistakes and need help. It is expected that you</span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>are familiar with independent study content before the workshop.</span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>However, you need not remember or understand every detail as the</span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>workshop should build and consolidate your understanding. Tips</span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>don't worry about making mistakes</span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>don't let what you can not do interfere with what you can do</span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>discussing code with your neighbours will help</span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>look things up in the independent study material</span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>look things up in your own code from earlier</span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>there are no stupid questions</span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a><span class="fu">## Key</span></span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a>These four symbols are used at the beginning of each instruction so you</span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a>know where to carry out the instruction.</span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_on_your_computer.png)</span> Something you need to do on your</span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a>computer. It may be opening programs or documents or locating a file.</span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Something you should do in RStudio. It will</span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a>often be typing a command or using the menus but might also be creating</span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a>folders, locating or moving files.</span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_on_internet.png)</span> Something you should do in your browser</span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a>on the internet. It may be searching for information, going to the VLE</span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a>or downloading a file.</span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-60"><a href="#cb17-60" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> A question for you to think about and answer.</span>
<span id="cb17-61"><a href="#cb17-61" aria-hidden="true" tabindex="-1"></a>Record your answers in your script for future reference.</span>
<span id="cb17-62"><a href="#cb17-62" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb17-63"><a href="#cb17-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-64"><a href="#cb17-64" aria-hidden="true" tabindex="-1"></a><span class="fu"># Getting started</span></span>
<span id="cb17-65"><a href="#cb17-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-66"><a href="#cb17-66" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_on_your_computer.png)</span> Start RStudio from the Start menu.</span>
<span id="cb17-67"><a href="#cb17-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-68"><a href="#cb17-68" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Make an RStudio project for this workshop by</span>
<span id="cb17-69"><a href="#cb17-69" aria-hidden="true" tabindex="-1"></a>clicking on the drop-down menu on top right where it says</span>
<span id="cb17-70"><a href="#cb17-70" aria-hidden="true" tabindex="-1"></a><span class="in">`Project: (None)`</span> and choosing New Project, then New Directory, then New</span>
<span id="cb17-71"><a href="#cb17-71" aria-hidden="true" tabindex="-1"></a>Project. Navigate to the <span class="in">`data-analysis-in-r-2`</span> folder and name the</span>
<span id="cb17-72"><a href="#cb17-72" aria-hidden="true" tabindex="-1"></a>RStudio Project <span class="in">`week-3`</span>.</span>
<span id="cb17-73"><a href="#cb17-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-74"><a href="#cb17-74" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Make new folders called <span class="in">`data-raw`</span> and</span>
<span id="cb17-75"><a href="#cb17-75" aria-hidden="true" tabindex="-1"></a><span class="in">`figures`</span>. You can do this on the Files Pane by clicking New Folder and</span>
<span id="cb17-76"><a href="#cb17-76" aria-hidden="true" tabindex="-1"></a>typing into the box that appears.</span>
<span id="cb17-77"><a href="#cb17-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-78"><a href="#cb17-78" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Make a new script then save it with a name like</span>
<span id="cb17-79"><a href="#cb17-79" aria-hidden="true" tabindex="-1"></a><span class="in">`two-sample-tests.R`</span> to carry out the rest of the work.</span>
<span id="cb17-80"><a href="#cb17-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-81"><a href="#cb17-81" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Add a comment to the script:</span>
<span id="cb17-82"><a href="#cb17-82" aria-hidden="true" tabindex="-1"></a><span class="in">`# Two-sample tests`</span> and load the **`tidyverse`** <span class="co">[</span><span class="ot">@tidyverse</span><span class="co">]</span> package</span>
<span id="cb17-83"><a href="#cb17-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-84"><a href="#cb17-84" aria-hidden="true" tabindex="-1"></a><span class="fu"># Exercises</span></span>
<span id="cb17-85"><a href="#cb17-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-86"><a href="#cb17-86" aria-hidden="true" tabindex="-1"></a><span class="fu">## Adiponectin secretion</span></span>
<span id="cb17-87"><a href="#cb17-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-88"><a href="#cb17-88" aria-hidden="true" tabindex="-1"></a>Adiponectin is exclusively secreted from adipose tissue and modulates a</span>
<span id="cb17-89"><a href="#cb17-89" aria-hidden="true" tabindex="-1"></a>number of metabolic processes. Nicotinic acid can affect adiponectin</span>
<span id="cb17-90"><a href="#cb17-90" aria-hidden="true" tabindex="-1"></a>secretion. 3T3-L1 adipocytes were treated with nicotinic acid or with a</span>
<span id="cb17-91"><a href="#cb17-91" aria-hidden="true" tabindex="-1"></a>control treatment and adiponectin concentration (pg/mL) measured in the</span>
<span id="cb17-92"><a href="#cb17-92" aria-hidden="true" tabindex="-1"></a>medium. The data are in <span class="co">[</span><span class="ot">adipocytes.txt</span><span class="co">](data-raw/adipocytes.txt)</span>. Each</span>
<span id="cb17-93"><a href="#cb17-93" aria-hidden="true" tabindex="-1"></a>row represents an independent sample of adipocytes and the first column</span>
<span id="cb17-94"><a href="#cb17-94" aria-hidden="true" tabindex="-1"></a>gives the concentration of adiponectin and the second column indicates</span>
<span id="cb17-95"><a href="#cb17-95" aria-hidden="true" tabindex="-1"></a>whether they were treated with nicotinic acid or not (the control).</span>
<span id="cb17-96"><a href="#cb17-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-97"><a href="#cb17-97" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_on_your_computer.png)</span> Save a copy of</span>
<span id="cb17-98"><a href="#cb17-98" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">adipocytes.txt</span><span class="co">](data-raw/adipocytes.txt)</span> to <span class="in">`data-raw`</span></span>
<span id="cb17-99"><a href="#cb17-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-100"><a href="#cb17-100" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Read in the data and check the structure. I used</span>
<span id="cb17-101"><a href="#cb17-101" aria-hidden="true" tabindex="-1"></a>the name <span class="in">`adip`</span> for the dataframe/tibble.</span>
<span id="cb17-102"><a href="#cb17-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-105"><a href="#cb17-105" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-106"><a href="#cb17-106" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false  </span></span>
<span id="cb17-107"><a href="#cb17-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-108"><a href="#cb17-108" aria-hidden="true" tabindex="-1"></a><span class="co">#| #---CODING ANSWER---</span></span>
<span id="cb17-109"><a href="#cb17-109" aria-hidden="true" tabindex="-1"></a>adip  <span class="ot">&lt;-</span>  <span class="fu">read_table</span>(<span class="st">"data-raw/adipocytes.txt"</span>)</span>
<span id="cb17-110"><a href="#cb17-110" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(adip)</span>
<span id="cb17-111"><a href="#cb17-111" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-112"><a href="#cb17-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-113"><a href="#cb17-113" aria-hidden="true" tabindex="-1"></a>We have a tibble containing two variables: <span class="in">`adiponectin`</span> is the response</span>
<span id="cb17-114"><a href="#cb17-114" aria-hidden="true" tabindex="-1"></a>variable and is continuous; <span class="in">`treatment`</span> is explanatory. <span class="in">`treatment`</span> is</span>
<span id="cb17-115"><a href="#cb17-115" aria-hidden="true" tabindex="-1"></a>categorical with two levels (groups). The first task is visualise the</span>
<span id="cb17-116"><a href="#cb17-116" aria-hidden="true" tabindex="-1"></a>data to get an overview. For continuous response variables with</span>
<span id="cb17-117"><a href="#cb17-117" aria-hidden="true" tabindex="-1"></a>categorical explanatory variables you could use <span class="in">`geom_point()`</span>,</span>
<span id="cb17-118"><a href="#cb17-118" aria-hidden="true" tabindex="-1"></a><span class="in">`geom_boxplot()`</span> or a variety of other geoms. I often use</span>
<span id="cb17-119"><a href="#cb17-119" aria-hidden="true" tabindex="-1"></a><span class="in">`geom_violin()`</span> which allows us to see the distribution - the violin is</span>
<span id="cb17-120"><a href="#cb17-120" aria-hidden="true" tabindex="-1"></a>fatter where there are more data points.</span>
<span id="cb17-121"><a href="#cb17-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-122"><a href="#cb17-122" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Do a quick plot of the data:</span>
<span id="cb17-123"><a href="#cb17-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-126"><a href="#cb17-126" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-127"><a href="#cb17-127" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> adip, <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> adiponectin)) <span class="sc">+</span></span>
<span id="cb17-128"><a href="#cb17-128" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>()</span>
<span id="cb17-129"><a href="#cb17-129" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-130"><a href="#cb17-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-131"><a href="#cb17-131" aria-hidden="true" tabindex="-1"></a><span class="fu">### Summarising the data</span></span>
<span id="cb17-132"><a href="#cb17-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-133"><a href="#cb17-133" aria-hidden="true" tabindex="-1"></a>Summarising the data for each treatment group is the next sensible step.</span>
<span id="cb17-134"><a href="#cb17-134" aria-hidden="true" tabindex="-1"></a>The most useful summary statistics are the means, standard deviations,</span>
<span id="cb17-135"><a href="#cb17-135" aria-hidden="true" tabindex="-1"></a>sample sizes and standard errors.</span>
<span id="cb17-136"><a href="#cb17-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-137"><a href="#cb17-137" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Create a data frame called <span class="in">`adip_summary`</span> that</span>
<span id="cb17-138"><a href="#cb17-138" aria-hidden="true" tabindex="-1"></a>contains the means, standard deviations, sample sizes and standard</span>
<span id="cb17-139"><a href="#cb17-139" aria-hidden="true" tabindex="-1"></a>errors for the control and nicotinic acid treated samples. You may need</span>
<span id="cb17-140"><a href="#cb17-140" aria-hidden="true" tabindex="-1"></a>to read the [Summarise</span>
<span id="cb17-141"><a href="#cb17-141" aria-hidden="true" tabindex="-1"></a>section](../../r4babs1/week-9/workshop.html#summarise) from the Week 9</span>
<span id="cb17-142"><a href="#cb17-142" aria-hidden="true" tabindex="-1"></a>workshop of BABS1 <span class="co">[</span><span class="ot">@rand2023</span><span class="co">]</span></span>
<span id="cb17-143"><a href="#cb17-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-146"><a href="#cb17-146" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-147"><a href="#cb17-147" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false  </span></span>
<span id="cb17-148"><a href="#cb17-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-149"><a href="#cb17-149" aria-hidden="true" tabindex="-1"></a><span class="co">#| #---CODING ANSWER---</span></span>
<span id="cb17-150"><a href="#cb17-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-151"><a href="#cb17-151" aria-hidden="true" tabindex="-1"></a>adip_summary <span class="ot">&lt;-</span> adip <span class="sc">|&gt;</span></span>
<span id="cb17-152"><a href="#cb17-152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(treatment) <span class="sc">|&gt;</span></span>
<span id="cb17-153"><a href="#cb17-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean =</span> <span class="fu">mean</span>(adiponectin),</span>
<span id="cb17-154"><a href="#cb17-154" aria-hidden="true" tabindex="-1"></a>            <span class="at">std =</span> <span class="fu">sd</span>(adiponectin),</span>
<span id="cb17-155"><a href="#cb17-155" aria-hidden="true" tabindex="-1"></a>            <span class="at">n =</span> <span class="fu">length</span>(adiponectin),</span>
<span id="cb17-156"><a href="#cb17-156" aria-hidden="true" tabindex="-1"></a>            <span class="at">se =</span> std<span class="sc">/</span><span class="fu">sqrt</span>(n))</span>
<span id="cb17-157"><a href="#cb17-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-158"><a href="#cb17-158" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-159"><a href="#cb17-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-160"><a href="#cb17-160" aria-hidden="true" tabindex="-1"></a>You should get the following numbers:</span>
<span id="cb17-161"><a href="#cb17-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-164"><a href="#cb17-164" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-165"><a href="#cb17-165" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb17-166"><a href="#cb17-166" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(adip_summary) <span class="sc">|&gt;</span> </span>
<span id="cb17-167"><a href="#cb17-167" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>()</span>
<span id="cb17-168"><a href="#cb17-168" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-169"><a href="#cb17-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-170"><a href="#cb17-170" aria-hidden="true" tabindex="-1"></a>::: callout-tip</span>
<span id="cb17-171"><a href="#cb17-171" aria-hidden="true" tabindex="-1"></a>Remember that you can see the contents of a dataframe with any of these:</span>
<span id="cb17-172"><a href="#cb17-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-173"><a href="#cb17-173" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>type the name of the dataframe in the console: prints the dataframe</span>
<span id="cb17-174"><a href="#cb17-174" aria-hidden="true" tabindex="-1"></a>    to console</span>
<span id="cb17-175"><a href="#cb17-175" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>click on the dataframe name in the Environment window: opens a</span>
<span id="cb17-176"><a href="#cb17-176" aria-hidden="true" tabindex="-1"></a>    spreadsheet like view of the dataframe in the Code window</span>
<span id="cb17-177"><a href="#cb17-177" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>typing <span class="in">`View(dataframe)`</span> in the console: opens a spreadsheet like</span>
<span id="cb17-178"><a href="#cb17-178" aria-hidden="true" tabindex="-1"></a>    view of the dataframe in the Code window</span>
<span id="cb17-179"><a href="#cb17-179" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb17-180"><a href="#cb17-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-181"><a href="#cb17-181" aria-hidden="true" tabindex="-1"></a><span class="fu">### Selecting a test</span></span>
<span id="cb17-182"><a href="#cb17-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-183"><a href="#cb17-183" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> Do you think this is a paired-sample test or</span>
<span id="cb17-184"><a href="#cb17-184" aria-hidden="true" tabindex="-1"></a>two-sample test?</span>
<span id="cb17-185"><a href="#cb17-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-186"><a href="#cb17-186" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER---  --&gt;</span></span>
<span id="cb17-187"><a href="#cb17-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-188"><a href="#cb17-188" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Two-sample - the samples are either control or nicotinic acid treated, not both.  --&gt;</span></span>
<span id="cb17-189"><a href="#cb17-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-190"><a href="#cb17-190" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- There is no link between the first control value and the first nicotinic acid  --&gt;</span></span>
<span id="cb17-191"><a href="#cb17-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-192"><a href="#cb17-192" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- value. The two samples are independent and it is a two-sample --&gt;</span></span>
<span id="cb17-193"><a href="#cb17-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-194"><a href="#cb17-194" aria-hidden="true" tabindex="-1"></a><span class="fu">### Applying, interpreting and reporting</span></span>
<span id="cb17-195"><a href="#cb17-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-196"><a href="#cb17-196" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Create a two-sample model like this:</span>
<span id="cb17-197"><a href="#cb17-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-200"><a href="#cb17-200" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-201"><a href="#cb17-201" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="at">data =</span> adip,</span>
<span id="cb17-202"><a href="#cb17-202" aria-hidden="true" tabindex="-1"></a>          adiponectin <span class="sc">~</span> treatment)</span>
<span id="cb17-203"><a href="#cb17-203" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-204"><a href="#cb17-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-205"><a href="#cb17-205" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Examine the model with:</span>
<span id="cb17-206"><a href="#cb17-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-209"><a href="#cb17-209" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-210"><a href="#cb17-210" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(mod)</span>
<span id="cb17-211"><a href="#cb17-211" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-212"><a href="#cb17-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-213"><a href="#cb17-213" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> What do you conclude from the test? Write your</span>
<span id="cb17-214"><a href="#cb17-214" aria-hidden="true" tabindex="-1"></a>conclusion in a form suitable for a report.</span>
<span id="cb17-215"><a href="#cb17-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-216"><a href="#cb17-216" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER---  --&gt;</span></span>
<span id="cb17-217"><a href="#cb17-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-218"><a href="#cb17-218" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Adipocytes treated with nicotinic acid (mean +/- s.e.: 7.51 +/- 0.38 pg/mL)  --&gt;</span></span>
<span id="cb17-219"><a href="#cb17-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-220"><a href="#cb17-220" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- secrete significantly more adiponectin than controls (5.55  +/- 0.46 pg/mL)  --&gt;</span></span>
<span id="cb17-221"><a href="#cb17-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-222"><a href="#cb17-222" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- (t = 3.27; d.f. = 28; p = 0.003). --&gt;</span></span>
<span id="cb17-223"><a href="#cb17-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-224"><a href="#cb17-224" aria-hidden="true" tabindex="-1"></a><span class="fu">### Check assumptions</span></span>
<span id="cb17-225"><a href="#cb17-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-226"><a href="#cb17-226" aria-hidden="true" tabindex="-1"></a>The assumptions of the general linear model are that the residuals – the</span>
<span id="cb17-227"><a href="#cb17-227" aria-hidden="true" tabindex="-1"></a>difference between predicted value (*i.e*., the group mean) and observed</span>
<span id="cb17-228"><a href="#cb17-228" aria-hidden="true" tabindex="-1"></a>values - are normally distributed, and have homogeneous variance. To</span>
<span id="cb17-229"><a href="#cb17-229" aria-hidden="true" tabindex="-1"></a>check these we can examine the <span class="in">`mod$residuals`</span> variable. You may want to</span>
<span id="cb17-230"><a href="#cb17-230" aria-hidden="true" tabindex="-1"></a>refer to [Checking</span>
<span id="cb17-231"><a href="#cb17-231" aria-hidden="true" tabindex="-1"></a>assumptions](../week-2/workshop.html#checking-assumptions) in the</span>
<span id="cb17-232"><a href="#cb17-232" aria-hidden="true" tabindex="-1"></a>"Single regression" workshop.</span>
<span id="cb17-233"><a href="#cb17-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-234"><a href="#cb17-234" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Plot the model residuals against the fitted</span>
<span id="cb17-235"><a href="#cb17-235" aria-hidden="true" tabindex="-1"></a>values.</span>
<span id="cb17-236"><a href="#cb17-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-239"><a href="#cb17-239" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-240"><a href="#cb17-240" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb17-241"><a href="#cb17-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-242"><a href="#cb17-242" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER-</span></span>
<span id="cb17-243"><a href="#cb17-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-244"><a href="#cb17-244" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mod, <span class="at">which =</span> <span class="dv">1</span>)</span>
<span id="cb17-245"><a href="#cb17-245" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-246"><a href="#cb17-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-247"><a href="#cb17-247" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> What to you conclude?</span>
<span id="cb17-248"><a href="#cb17-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-249"><a href="#cb17-249" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER--- --&gt;</span></span>
<span id="cb17-250"><a href="#cb17-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-251"><a href="#cb17-251" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- The residuals look 'homogeneously' distributed. That is, the variance is the same  --&gt;</span></span>
<span id="cb17-252"><a href="#cb17-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-253"><a href="#cb17-253" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- for the three groups  --&gt;</span></span>
<span id="cb17-254"><a href="#cb17-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-255"><a href="#cb17-255" aria-hidden="true" tabindex="-1"></a>To examine normality of the model residuals we can plot them as a</span>
<span id="cb17-256"><a href="#cb17-256" aria-hidden="true" tabindex="-1"></a>histogram and do a normality test on them.</span>
<span id="cb17-257"><a href="#cb17-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-258"><a href="#cb17-258" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Plot a histogram of the residuals.</span>
<span id="cb17-259"><a href="#cb17-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-262"><a href="#cb17-262" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-263"><a href="#cb17-263" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb17-264"><a href="#cb17-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-265"><a href="#cb17-265" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER-</span></span>
<span id="cb17-266"><a href="#cb17-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-267"><a href="#cb17-267" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> mod<span class="sc">$</span>residuals)) <span class="sc">+</span> </span>
<span id="cb17-268"><a href="#cb17-268" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">10</span>)</span>
<span id="cb17-269"><a href="#cb17-269" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-270"><a href="#cb17-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-271"><a href="#cb17-271" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Use the <span class="in">`shapiro.test()`</span> to test the normality</span>
<span id="cb17-272"><a href="#cb17-272" aria-hidden="true" tabindex="-1"></a>of the model residuals</span>
<span id="cb17-273"><a href="#cb17-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-276"><a href="#cb17-276" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-277"><a href="#cb17-277" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb17-278"><a href="#cb17-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-279"><a href="#cb17-279" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER-</span></span>
<span id="cb17-280"><a href="#cb17-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-281"><a href="#cb17-281" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(mod<span class="sc">$</span>residuals)</span>
<span id="cb17-282"><a href="#cb17-282" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-283"><a href="#cb17-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-284"><a href="#cb17-284" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> What to you conclude?</span>
<span id="cb17-285"><a href="#cb17-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-286"><a href="#cb17-286" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER--- --&gt;</span></span>
<span id="cb17-287"><a href="#cb17-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-288"><a href="#cb17-288" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- the residuals appear roughly normally distributed.  --&gt;</span></span>
<span id="cb17-289"><a href="#cb17-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-290"><a href="#cb17-290" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- The histogram is symmetrical and the shapiro test is NS. --&gt;</span></span>
<span id="cb17-291"><a href="#cb17-291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-292"><a href="#cb17-292" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Overall, it seems the assumptions are met --&gt;</span></span>
<span id="cb17-293"><a href="#cb17-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-294"><a href="#cb17-294" aria-hidden="true" tabindex="-1"></a><span class="fu">### Illustrating</span></span>
<span id="cb17-295"><a href="#cb17-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-296"><a href="#cb17-296" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Create a figure like the one below. You may need</span>
<span id="cb17-297"><a href="#cb17-297" aria-hidden="true" tabindex="-1"></a>to refer to <span class="co">[</span><span class="ot">Visualise</span><span class="co">](../../r4babs1/week-9/workshop.html#visualise)</span></span>
<span id="cb17-298"><a href="#cb17-298" aria-hidden="true" tabindex="-1"></a>from the "Summarising data with several variables" workshop <span class="co">[</span><span class="ot">@rand2023</span><span class="co">]</span></span>
<span id="cb17-299"><a href="#cb17-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-302"><a href="#cb17-302" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-303"><a href="#cb17-303" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb17-304"><a href="#cb17-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-305"><a href="#cb17-305" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER---</span></span>
<span id="cb17-306"><a href="#cb17-306" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb17-307"><a href="#cb17-307" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> adip, <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> adiponectin),</span>
<span id="cb17-308"><a href="#cb17-308" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">height =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb17-309"><a href="#cb17-309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> adip_summary, </span>
<span id="cb17-310"><a href="#cb17-310" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">ymin =</span> mean <span class="sc">-</span> se, <span class="at">ymax =</span> mean <span class="sc">+</span> se),</span>
<span id="cb17-311"><a href="#cb17-311" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb17-312"><a href="#cb17-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> adip_summary, </span>
<span id="cb17-313"><a href="#cb17-313" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">ymin =</span> mean, <span class="at">ymax =</span> mean),</span>
<span id="cb17-314"><a href="#cb17-314" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb17-315"><a href="#cb17-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Adiponectin (pg/mL)"</span>, </span>
<span id="cb17-316"><a href="#cb17-316" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>), </span>
<span id="cb17-317"><a href="#cb17-317" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb17-318"><a href="#cb17-318" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Treatment"</span>, </span>
<span id="cb17-319"><a href="#cb17-319" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Control"</span>, <span class="st">"Nicotinic acid"</span>)) <span class="sc">+</span></span>
<span id="cb17-320"><a href="#cb17-320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb17-321"><a href="#cb17-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-322"><a href="#cb17-322" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-323"><a href="#cb17-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-324"><a href="#cb17-324" aria-hidden="true" tabindex="-1"></a>We now need to annotate the figure with the results from the statistical</span>
<span id="cb17-325"><a href="#cb17-325" aria-hidden="true" tabindex="-1"></a>test. This most commonly done with a line linking the means being</span>
<span id="cb17-326"><a href="#cb17-326" aria-hidden="true" tabindex="-1"></a>compared and the *p*-value. The <span class="in">`annotate()`</span> function can be used to</span>
<span id="cb17-327"><a href="#cb17-327" aria-hidden="true" tabindex="-1"></a>draw the line and then to add the value. The line is a <span class="in">`segment`</span> and the</span>
<span id="cb17-328"><a href="#cb17-328" aria-hidden="true" tabindex="-1"></a>*p*-value is a <span class="in">`text`</span>.</span>
<span id="cb17-329"><a href="#cb17-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-330"><a href="#cb17-330" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Annotation - method 1 {#sec-week-3-annotation-method-1}</span></span>
<span id="cb17-331"><a href="#cb17-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-332"><a href="#cb17-332" aria-hidden="true" tabindex="-1"></a>You can annotate the *p*-value manually by typing it in to the</span>
<span id="cb17-333"><a href="#cb17-333" aria-hidden="true" tabindex="-1"></a>appropriate <span class="in">`annotate()`</span> function.</span>
<span id="cb17-334"><a href="#cb17-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-335"><a href="#cb17-335" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Add annotation to the figure by adding:</span>
<span id="cb17-336"><a href="#cb17-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-337"><a href="#cb17-337" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb17-338"><a href="#cb17-338" aria-hidden="true" tabindex="-1"></a>...... <span class="sc">+</span></span>
<span id="cb17-339"><a href="#cb17-339" aria-hidden="true" tabindex="-1"></a><span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>, </span>
<span id="cb17-340"><a href="#cb17-340" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fl">11.3</span>, <span class="at">yend =</span> <span class="fl">11.3</span>,</span>
<span id="cb17-341"><a href="#cb17-341" aria-hidden="true" tabindex="-1"></a>         <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb17-342"><a href="#cb17-342" aria-hidden="true" tabindex="-1"></a><span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>,  <span class="at">y =</span> <span class="fl">11.7</span>, </span>
<span id="cb17-343"><a href="#cb17-343" aria-hidden="true" tabindex="-1"></a>         <span class="at">label =</span> <span class="st">"italic(p) == 0.0028"</span>,</span>
<span id="cb17-344"><a href="#cb17-344" aria-hidden="true" tabindex="-1"></a>         <span class="at">parse =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-345"><a href="#cb17-345" aria-hidden="true" tabindex="-1"></a> .....</span>
<span id="cb17-346"><a href="#cb17-346" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-347"><a href="#cb17-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-348"><a href="#cb17-348" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`annotate("segment", ...)`</span> adds a line to the figure starting at *x</span>
<span id="cb17-349"><a href="#cb17-349" aria-hidden="true" tabindex="-1"></a>    = 1* and *y = 11.3* and ending at *x = 2* and *y = 11.3*</span>
<span id="cb17-350"><a href="#cb17-350" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`annotate("text", ...)`</span> adds text to the figure at *x = 1.5* and *y</span>
<span id="cb17-351"><a href="#cb17-351" aria-hidden="true" tabindex="-1"></a>    = 11.7*</span>
<span id="cb17-352"><a href="#cb17-352" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>the <span class="in">`label`</span> is written in R’s plotmath syntax rather than being a</span>
<span id="cb17-353"><a href="#cb17-353" aria-hidden="true" tabindex="-1"></a>    normal string.</span>
<span id="cb17-354"><a href="#cb17-354" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span><span class="in">`italic(p)`</span> renders the letter *p* in italics</span>
<span id="cb17-355"><a href="#cb17-355" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span><span class="in">`==`</span> is the plotmath symbol for "="</span>
<span id="cb17-356"><a href="#cb17-356" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`parse = TRUE`</span> tells ggplot2 to interpret that string as a plotmath</span>
<span id="cb17-357"><a href="#cb17-357" aria-hidden="true" tabindex="-1"></a>    expression rather than literally adding "italic(p) == 0.0028".</span>
<span id="cb17-358"><a href="#cb17-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-359"><a href="#cb17-359" aria-hidden="true" tabindex="-1"></a>However, it is more reproducible to extract the *p*-value from the model</span>
<span id="cb17-360"><a href="#cb17-360" aria-hidden="true" tabindex="-1"></a>object.</span>
<span id="cb17-361"><a href="#cb17-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-362"><a href="#cb17-362" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Annotation - method 2 {#sec-week-3-annotation-method-2}</span></span>
<span id="cb17-363"><a href="#cb17-363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-364"><a href="#cb17-364" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Extract the *p*-value from the model object:</span>
<span id="cb17-365"><a href="#cb17-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-368"><a href="#cb17-368" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-369"><a href="#cb17-369" aria-hidden="true" tabindex="-1"></a>p_value <span class="ot">&lt;-</span> broom<span class="sc">::</span><span class="fu">glance</span>(mod) <span class="sc">|&gt;</span> <span class="fu">pull</span>(p.value) <span class="sc">|&gt;</span> <span class="fu">round</span>(<span class="dv">4</span>)</span>
<span id="cb17-370"><a href="#cb17-370" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-371"><a href="#cb17-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-372"><a href="#cb17-372" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**\`broom** is a package that turns model outputs into more easy to</span>
<span id="cb17-373"><a href="#cb17-373" aria-hidden="true" tabindex="-1"></a>    use tidy formats</span>
<span id="cb17-374"><a href="#cb17-374" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**`glance(mod)`** puts information from mod including the *p*-value</span>
<span id="cb17-375"><a href="#cb17-375" aria-hidden="true" tabindex="-1"></a>    into a dataframe</span>
<span id="cb17-376"><a href="#cb17-376" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`pull(p.value)`</span> extracts the value from *p*-value column in the</span>
<span id="cb17-377"><a href="#cb17-377" aria-hidden="true" tabindex="-1"></a>    dataframe</span>
<span id="cb17-378"><a href="#cb17-378" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`round(4)`</span> rounds the *p*-value to 4 decimal places</span>
<span id="cb17-379"><a href="#cb17-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-380"><a href="#cb17-380" aria-hidden="true" tabindex="-1"></a>The segment and text can then be added to the figure.</span>
<span id="cb17-381"><a href="#cb17-381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-382"><a href="#cb17-382" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Add annotation to the figure by adding:</span>
<span id="cb17-383"><a href="#cb17-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-384"><a href="#cb17-384" aria-hidden="true" tabindex="-1"></a><span class="in">``` r</span></span>
<span id="cb17-385"><a href="#cb17-385" aria-hidden="true" tabindex="-1"></a>...... <span class="sc">+</span></span>
<span id="cb17-386"><a href="#cb17-386" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>, </span>
<span id="cb17-387"><a href="#cb17-387" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="fl">11.3</span>, <span class="at">yend =</span> <span class="fl">11.3</span>,</span>
<span id="cb17-388"><a href="#cb17-388" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb17-389"><a href="#cb17-389" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>,  <span class="at">y =</span> <span class="fl">11.7</span>, </span>
<span id="cb17-390"><a href="#cb17-390" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"italic(p) == { p_value }"</span>),</span>
<span id="cb17-391"><a href="#cb17-391" aria-hidden="true" tabindex="-1"></a>                              <span class="at">parse =</span> <span class="cn">TRUE</span>)<span class="er">)</span> <span class="sc">+</span></span>
<span id="cb17-392"><a href="#cb17-392" aria-hidden="true" tabindex="-1"></a>.....</span>
<span id="cb17-393"><a href="#cb17-393" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-394"><a href="#cb17-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-395"><a href="#cb17-395" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The glue package allows us to use R objects in strings so their v</span>
<span id="cb17-396"><a href="#cb17-396" aria-hidden="true" tabindex="-1"></a>    values are printed.</span>
<span id="cb17-397"><a href="#cb17-397" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>We use curly braces <span class="in">`{}`</span> a round the R object so it is understood as</span>
<span id="cb17-398"><a href="#cb17-398" aria-hidden="true" tabindex="-1"></a>    an R object and its value is written rather than literally writing</span>
<span id="cb17-399"><a href="#cb17-399" aria-hidden="true" tabindex="-1"></a>    "p_value"</span>
<span id="cb17-400"><a href="#cb17-400" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-403"><a href="#cb17-403" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-404"><a href="#cb17-404" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb17-405"><a href="#cb17-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-406"><a href="#cb17-406" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb17-407"><a href="#cb17-407" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> adip, <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> adiponectin),</span>
<span id="cb17-408"><a href="#cb17-408" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">height =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb17-409"><a href="#cb17-409" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> adip_summary, </span>
<span id="cb17-410"><a href="#cb17-410" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">ymin =</span> mean <span class="sc">-</span> se, <span class="at">ymax =</span> mean <span class="sc">+</span> se),</span>
<span id="cb17-411"><a href="#cb17-411" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb17-412"><a href="#cb17-412" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> adip_summary, </span>
<span id="cb17-413"><a href="#cb17-413" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">ymin =</span> mean, <span class="at">ymax =</span> mean),</span>
<span id="cb17-414"><a href="#cb17-414" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb17-415"><a href="#cb17-415" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Adiponectin (pg/mL)"</span>, </span>
<span id="cb17-416"><a href="#cb17-416" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>), </span>
<span id="cb17-417"><a href="#cb17-417" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb17-418"><a href="#cb17-418" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Treatment"</span>, </span>
<span id="cb17-419"><a href="#cb17-419" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Control"</span>, <span class="st">"Nicotinic acid"</span>)) <span class="sc">+</span></span>
<span id="cb17-420"><a href="#cb17-420" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>, </span>
<span id="cb17-421"><a href="#cb17-421" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="fl">11.3</span>, <span class="at">yend =</span> <span class="fl">11.3</span>,</span>
<span id="cb17-422"><a href="#cb17-422" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb17-423"><a href="#cb17-423" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>,  <span class="at">y =</span> <span class="fl">11.7</span>,</span>
<span id="cb17-424"><a href="#cb17-424" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"italic(p) == { p_value }"</span>),</span>
<span id="cb17-425"><a href="#cb17-425" aria-hidden="true" tabindex="-1"></a>                              <span class="at">parse =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb17-426"><a href="#cb17-426" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb17-427"><a href="#cb17-427" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-428"><a href="#cb17-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-429"><a href="#cb17-429" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Save your figure to your figures folder. Make</span>
<span id="cb17-430"><a href="#cb17-430" aria-hidden="true" tabindex="-1"></a>sure you [script figure saving with</span>
<span id="cb17-431"><a href="#cb17-431" aria-hidden="true" tabindex="-1"></a><span class="in">`ggsave()`</span>](../../r4babs1/week-9/workshop.html#writing-figures-to-file).</span>
<span id="cb17-432"><a href="#cb17-432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-435"><a href="#cb17-435" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-436"><a href="#cb17-436" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb17-437"><a href="#cb17-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-438"><a href="#cb17-438" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER---</span></span>
<span id="cb17-439"><a href="#cb17-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-440"><a href="#cb17-440" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign the figure to a variable</span></span>
<span id="cb17-441"><a href="#cb17-441" aria-hidden="true" tabindex="-1"></a>fig1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb17-442"><a href="#cb17-442" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> adip, <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">y =</span> adiponectin),</span>
<span id="cb17-443"><a href="#cb17-443" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">height =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb17-444"><a href="#cb17-444" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> adip_summary, </span>
<span id="cb17-445"><a href="#cb17-445" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">ymin =</span> mean <span class="sc">-</span> se, <span class="at">ymax =</span> mean <span class="sc">+</span> se),</span>
<span id="cb17-446"><a href="#cb17-446" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb17-447"><a href="#cb17-447" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">data =</span> adip_summary, </span>
<span id="cb17-448"><a href="#cb17-448" aria-hidden="true" tabindex="-1"></a>                <span class="fu">aes</span>(<span class="at">x =</span> treatment, <span class="at">ymin =</span> mean, <span class="at">ymax =</span> mean),</span>
<span id="cb17-449"><a href="#cb17-449" aria-hidden="true" tabindex="-1"></a>                <span class="at">width =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb17-450"><a href="#cb17-450" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Adiponectin (pg/mL)"</span>, </span>
<span id="cb17-451"><a href="#cb17-451" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>), </span>
<span id="cb17-452"><a href="#cb17-452" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb17-453"><a href="#cb17-453" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Treatment"</span>, </span>
<span id="cb17-454"><a href="#cb17-454" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Control"</span>, <span class="st">"Nicotinic acid"</span>)) <span class="sc">+</span></span>
<span id="cb17-455"><a href="#cb17-455" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>, </span>
<span id="cb17-456"><a href="#cb17-456" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="fl">11.3</span>, <span class="at">yend =</span> <span class="fl">11.3</span>,</span>
<span id="cb17-457"><a href="#cb17-457" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb17-458"><a href="#cb17-458" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>,  <span class="at">y =</span> <span class="fl">11.7</span>, </span>
<span id="cb17-459"><a href="#cb17-459" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"italic(p) == { p_value }"</span>),</span>
<span id="cb17-460"><a href="#cb17-460" aria-hidden="true" tabindex="-1"></a>                              <span class="at">parse =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb17-461"><a href="#cb17-461" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb17-462"><a href="#cb17-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-463"><a href="#cb17-463" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-464"><a href="#cb17-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-467"><a href="#cb17-467" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-468"><a href="#cb17-468" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb17-469"><a href="#cb17-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-470"><a href="#cb17-470" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER---</span></span>
<span id="cb17-471"><a href="#cb17-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-472"><a href="#cb17-472" aria-hidden="true" tabindex="-1"></a><span class="co"># save figure to figures/adipo.png</span></span>
<span id="cb17-473"><a href="#cb17-473" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"figures/adipo.png"</span>,</span>
<span id="cb17-474"><a href="#cb17-474" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> fig1,</span>
<span id="cb17-475"><a href="#cb17-475" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="fl">3.5</span>,</span>
<span id="cb17-476"><a href="#cb17-476" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="fl">3.5</span>,</span>
<span id="cb17-477"><a href="#cb17-477" aria-hidden="true" tabindex="-1"></a>       <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb17-478"><a href="#cb17-478" aria-hidden="true" tabindex="-1"></a>       <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb17-479"><a href="#cb17-479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-480"><a href="#cb17-480" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-481"><a href="#cb17-481" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-482"><a href="#cb17-482" aria-hidden="true" tabindex="-1"></a><span class="fu">## Grouse Parasites</span></span>
<span id="cb17-483"><a href="#cb17-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-484"><a href="#cb17-484" aria-hidden="true" tabindex="-1"></a>Grouse livers were dissected and the number of individuals of a</span>
<span id="cb17-485"><a href="#cb17-485" aria-hidden="true" tabindex="-1"></a>parasitic nematode were counted for two estates 'Gordon' and 'Moss'. We</span>
<span id="cb17-486"><a href="#cb17-486" aria-hidden="true" tabindex="-1"></a>want to know if the two estates have different infection rates. The data</span>
<span id="cb17-487"><a href="#cb17-487" aria-hidden="true" tabindex="-1"></a>are in <span class="co">[</span><span class="ot">grouse.csv</span><span class="co">](data-raw/grouse.csv)</span></span>
<span id="cb17-488"><a href="#cb17-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-489"><a href="#cb17-489" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_on_your_computer.png)</span> Save a copy of</span>
<span id="cb17-490"><a href="#cb17-490" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">grouse.csv</span><span class="co">](data-raw/grouse.csv)</span> to <span class="in">`data-raw`</span></span>
<span id="cb17-491"><a href="#cb17-491" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-492"><a href="#cb17-492" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Read in the data and check the structure. I used</span>
<span id="cb17-493"><a href="#cb17-493" aria-hidden="true" tabindex="-1"></a>the name <span class="in">`grouse`</span> for the dataframe/tibble.</span>
<span id="cb17-494"><a href="#cb17-494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-497"><a href="#cb17-497" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-498"><a href="#cb17-498" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false  </span></span>
<span id="cb17-499"><a href="#cb17-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-500"><a href="#cb17-500" aria-hidden="true" tabindex="-1"></a><span class="co">#| #---CODING ANSWER---</span></span>
<span id="cb17-501"><a href="#cb17-501" aria-hidden="true" tabindex="-1"></a>grouse  <span class="ot">&lt;-</span>  <span class="fu">read_csv</span>(<span class="st">"data-raw/grouse.csv"</span>)</span>
<span id="cb17-502"><a href="#cb17-502" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(grouse)</span>
<span id="cb17-503"><a href="#cb17-503" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-504"><a href="#cb17-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-505"><a href="#cb17-505" aria-hidden="true" tabindex="-1"></a><span class="fu">### Selecting</span></span>
<span id="cb17-506"><a href="#cb17-506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-507"><a href="#cb17-507" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> Using your common sense, do these data look</span>
<span id="cb17-508"><a href="#cb17-508" aria-hidden="true" tabindex="-1"></a>normally distributed?</span>
<span id="cb17-509"><a href="#cb17-509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-510"><a href="#cb17-510" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER---  --&gt;</span></span>
<span id="cb17-511"><a href="#cb17-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-512"><a href="#cb17-512" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- No and little need to test/plot - they're counts, skewed and  --&gt;</span></span>
<span id="cb17-513"><a href="#cb17-513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-514"><a href="#cb17-514" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- have unequal variances --&gt;</span></span>
<span id="cb17-515"><a href="#cb17-515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-516"><a href="#cb17-516" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> What test do you suggest?</span>
<span id="cb17-517"><a href="#cb17-517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-518"><a href="#cb17-518" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER---  --&gt;</span></span>
<span id="cb17-519"><a href="#cb17-519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-520"><a href="#cb17-520" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- two-sample Wilcoxon (unpaired) AKA Mann-Whitney --&gt;</span></span>
<span id="cb17-521"><a href="#cb17-521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-522"><a href="#cb17-522" aria-hidden="true" tabindex="-1"></a><span class="fu">### Applying, interpreting and reporting</span></span>
<span id="cb17-523"><a href="#cb17-523" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-524"><a href="#cb17-524" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Summarise the data by finding the median of each</span>
<span id="cb17-525"><a href="#cb17-525" aria-hidden="true" tabindex="-1"></a>group:</span>
<span id="cb17-526"><a href="#cb17-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-529"><a href="#cb17-529" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-530"><a href="#cb17-530" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false  </span></span>
<span id="cb17-531"><a href="#cb17-531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-532"><a href="#cb17-532" aria-hidden="true" tabindex="-1"></a><span class="co"># ---CODING ANSWER---</span></span>
<span id="cb17-533"><a href="#cb17-533" aria-hidden="true" tabindex="-1"></a>grouse <span class="sc">|&gt;</span> </span>
<span id="cb17-534"><a href="#cb17-534" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(estate) <span class="sc">|&gt;</span> </span>
<span id="cb17-535"><a href="#cb17-535" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">median</span>(nematodes))</span>
<span id="cb17-536"><a href="#cb17-536" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-537"><a href="#cb17-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-538"><a href="#cb17-538" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Carry out a two-sample Wilcoxon test (also known</span>
<span id="cb17-539"><a href="#cb17-539" aria-hidden="true" tabindex="-1"></a>as a Mann-Whitney):</span>
<span id="cb17-540"><a href="#cb17-540" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-543"><a href="#cb17-543" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-544"><a href="#cb17-544" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(<span class="at">data =</span> grouse, nematodes <span class="sc">~</span> estate)</span>
<span id="cb17-545"><a href="#cb17-545" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-546"><a href="#cb17-546" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-547"><a href="#cb17-547" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> What do you conclude from the test? Write your</span>
<span id="cb17-548"><a href="#cb17-548" aria-hidden="true" tabindex="-1"></a>conclusion in a form suitable for a report.</span>
<span id="cb17-549"><a href="#cb17-549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-550"><a href="#cb17-550" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER---  --&gt;</span></span>
<span id="cb17-551"><a href="#cb17-551" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-552"><a href="#cb17-552" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- Grouse on Gordon had significantly more nematodes (median = 13.5) than those  --&gt;</span></span>
<span id="cb17-553"><a href="#cb17-553" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-554"><a href="#cb17-554" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- on Moss (median = 5) (W = 78; n1 = 10, n2 = 10; p = 0.03546). --&gt;</span></span>
<span id="cb17-555"><a href="#cb17-555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-556"><a href="#cb17-556" aria-hidden="true" tabindex="-1"></a><span class="fu">### Illustrating</span></span>
<span id="cb17-557"><a href="#cb17-557" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-558"><a href="#cb17-558" aria-hidden="true" tabindex="-1"></a>A box plot is a usually good choice for illustrating a two-sample</span>
<span id="cb17-559"><a href="#cb17-559" aria-hidden="true" tabindex="-1"></a>Wilcoxon test because it shows the median and interquartile range.</span>
<span id="cb17-560"><a href="#cb17-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-561"><a href="#cb17-561" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> We can create a simple boxplot with:</span>
<span id="cb17-562"><a href="#cb17-562" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-565"><a href="#cb17-565" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-566"><a href="#cb17-566" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> grouse, <span class="fu">aes</span>(<span class="at">x =</span> estate, <span class="at">y =</span> nematodes) ) <span class="sc">+</span></span>
<span id="cb17-567"><a href="#cb17-567" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() </span>
<span id="cb17-568"><a href="#cb17-568" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-569"><a href="#cb17-569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-570"><a href="#cb17-570" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Annotate and format the figure so it is more</span>
<span id="cb17-571"><a href="#cb17-571" aria-hidden="true" tabindex="-1"></a>suitable for a report and save it to your figures folder.</span>
<span id="cb17-572"><a href="#cb17-572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-573"><a href="#cb17-573" aria-hidden="true" tabindex="-1"></a>::: callout-tip</span>
<span id="cb17-574"><a href="#cb17-574" aria-hidden="true" tabindex="-1"></a>If you want to use method 2 for annotating the *p*-value, you can</span>
<span id="cb17-575"><a href="#cb17-575" aria-hidden="true" tabindex="-1"></a>extract the *p*-value with</span>
<span id="cb17-576"><a href="#cb17-576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-579"><a href="#cb17-579" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-580"><a href="#cb17-580" aria-hidden="true" tabindex="-1"></a>p_value <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(<span class="at">data =</span> grouse, nematodes <span class="sc">~</span> estate) <span class="sc">|&gt;</span> </span>
<span id="cb17-581"><a href="#cb17-581" aria-hidden="true" tabindex="-1"></a>  broom<span class="sc">::</span><span class="fu">tidy</span>() <span class="sc">|&gt;</span> </span>
<span id="cb17-582"><a href="#cb17-582" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(p.value) <span class="sc">|&gt;</span></span>
<span id="cb17-583"><a href="#cb17-583" aria-hidden="true" tabindex="-1"></a>  <span class="fu">round</span>(<span class="dv">4</span>)</span>
<span id="cb17-584"><a href="#cb17-584" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-585"><a href="#cb17-585" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-586"><a href="#cb17-586" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb17-587"><a href="#cb17-587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-590"><a href="#cb17-590" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-591"><a href="#cb17-591" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false  </span></span>
<span id="cb17-592"><a href="#cb17-592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-593"><a href="#cb17-593" aria-hidden="true" tabindex="-1"></a><span class="co"># ---CODING ANSWER---</span></span>
<span id="cb17-594"><a href="#cb17-594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-595"><a href="#cb17-595" aria-hidden="true" tabindex="-1"></a>fig2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> grouse, <span class="fu">aes</span>(<span class="at">x =</span> estate, <span class="at">y =</span> nematodes) ) <span class="sc">+</span></span>
<span id="cb17-596"><a href="#cb17-596" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb17-597"><a href="#cb17-597" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">name =</span> <span class="st">"Estate"</span>, </span>
<span id="cb17-598"><a href="#cb17-598" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Gordon"</span>, <span class="st">"Moss"</span>)) <span class="sc">+</span></span>
<span id="cb17-599"><a href="#cb17-599" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Number of nematodes"</span>,</span>
<span id="cb17-600"><a href="#cb17-600" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>),</span>
<span id="cb17-601"><a href="#cb17-601" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">75</span>)) <span class="sc">+</span></span>
<span id="cb17-602"><a href="#cb17-602" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"segment"</span>, <span class="at">x =</span> <span class="dv">1</span>, <span class="at">xend =</span> <span class="dv">2</span>, </span>
<span id="cb17-603"><a href="#cb17-603" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="dv">67</span>, <span class="at">yend =</span> <span class="dv">67</span>,</span>
<span id="cb17-604"><a href="#cb17-604" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb17-605"><a href="#cb17-605" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>,  <span class="at">y =</span> <span class="dv">69</span>, </span>
<span id="cb17-606"><a href="#cb17-606" aria-hidden="true" tabindex="-1"></a>          <span class="at">label =</span> glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"italic(p) == { p_value }"</span>),</span>
<span id="cb17-607"><a href="#cb17-607" aria-hidden="true" tabindex="-1"></a>          <span class="at">parse =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb17-608"><a href="#cb17-608" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb17-609"><a href="#cb17-609" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-610"><a href="#cb17-610" aria-hidden="true" tabindex="-1"></a><span class="co"># save figure to figures/grouse.png</span></span>
<span id="cb17-611"><a href="#cb17-611" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"figures/grouse.png"</span>,</span>
<span id="cb17-612"><a href="#cb17-612" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> fig2,</span>
<span id="cb17-613"><a href="#cb17-613" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">3</span>,</span>
<span id="cb17-614"><a href="#cb17-614" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">3</span>,</span>
<span id="cb17-615"><a href="#cb17-615" aria-hidden="true" tabindex="-1"></a>       <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb17-616"><a href="#cb17-616" aria-hidden="true" tabindex="-1"></a>       <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb17-617"><a href="#cb17-617" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-618"><a href="#cb17-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-619"><a href="#cb17-619" aria-hidden="true" tabindex="-1"></a><span class="fu">## Gene Expression</span></span>
<span id="cb17-620"><a href="#cb17-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-621"><a href="#cb17-621" aria-hidden="true" tabindex="-1"></a>Bambara groundnut (*Vigna subterranea*) is an African legume with good</span>
<span id="cb17-622"><a href="#cb17-622" aria-hidden="true" tabindex="-1"></a>nutritional value which can be influenced by low temperature stress.</span>
<span id="cb17-623"><a href="#cb17-623" aria-hidden="true" tabindex="-1"></a>Researchers are interested in the expression levels of a particular set</span>
<span id="cb17-624"><a href="#cb17-624" aria-hidden="true" tabindex="-1"></a>of 35 genes (<span class="in">`probe_id`</span>) in response to temperature stress. They measure</span>
<span id="cb17-625"><a href="#cb17-625" aria-hidden="true" tabindex="-1"></a>the expression of the genes at 23 and 18 degrees C (high and low</span>
<span id="cb17-626"><a href="#cb17-626" aria-hidden="true" tabindex="-1"></a>temperature). These samples are **not** independent because we have two</span>
<span id="cb17-627"><a href="#cb17-627" aria-hidden="true" tabindex="-1"></a>measure from one gene. The data are in <span class="co">[</span><span class="ot">expr.xlsx</span><span class="co">](data-raw/expr.xlsx)</span>.</span>
<span id="cb17-628"><a href="#cb17-628" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-629"><a href="#cb17-629" aria-hidden="true" tabindex="-1"></a><span class="fu">### Selecting</span></span>
<span id="cb17-630"><a href="#cb17-630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-631"><a href="#cb17-631" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> What is the null hypothesis?</span>
<span id="cb17-632"><a href="#cb17-632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-633"><a href="#cb17-633" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER---  --&gt;</span></span>
<span id="cb17-634"><a href="#cb17-634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-635"><a href="#cb17-635" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- expression is not affected by temperature / for each gene,  --&gt;</span></span>
<span id="cb17-636"><a href="#cb17-636" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-637"><a href="#cb17-637" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- the expression at the higher t   = expression at the lower t --&gt;</span></span>
<span id="cb17-638"><a href="#cb17-638" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-639"><a href="#cb17-639" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Save a copy of <span class="co">[</span><span class="ot">expr.xlsx</span><span class="co">](data-raw/expr.xlsx)</span></span>
<span id="cb17-640"><a href="#cb17-640" aria-hidden="true" tabindex="-1"></a>and import the data. I named the dataframe <span class="in">`bambara`</span></span>
<span id="cb17-641"><a href="#cb17-641" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-644"><a href="#cb17-644" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-645"><a href="#cb17-645" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false  </span></span>
<span id="cb17-646"><a href="#cb17-646" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-647"><a href="#cb17-647" aria-hidden="true" tabindex="-1"></a><span class="co">#---CODING ANSWER---</span></span>
<span id="cb17-648"><a href="#cb17-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-649"><a href="#cb17-649" aria-hidden="true" tabindex="-1"></a><span class="co"># we need the readxl package</span></span>
<span id="cb17-650"><a href="#cb17-650" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb17-651"><a href="#cb17-651" aria-hidden="true" tabindex="-1"></a>bambara  <span class="ot">&lt;-</span>  <span class="fu">read_excel</span>(<span class="st">"data-raw/expr.xlsx"</span>)</span>
<span id="cb17-652"><a href="#cb17-652" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-653"><a href="#cb17-653" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-654"><a href="#cb17-654" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> What is the appropriate parametric test?</span>
<span id="cb17-655"><a href="#cb17-655" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-656"><a href="#cb17-656" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER--- --&gt;</span></span>
<span id="cb17-657"><a href="#cb17-657" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-658"><a href="#cb17-658" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- the appropriate parametric test is the paired sample lm  --&gt;</span></span>
<span id="cb17-659"><a href="#cb17-659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-660"><a href="#cb17-660" aria-hidden="true" tabindex="-1"></a><span class="fu">### Applying, interpreting and reporting</span></span>
<span id="cb17-661"><a href="#cb17-661" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-662"><a href="#cb17-662" aria-hidden="true" tabindex="-1"></a>A paired test requires account for the variation between genes.</span>
<span id="cb17-663"><a href="#cb17-663" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-664"><a href="#cb17-664" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Create a paired-sample model like this:</span>
<span id="cb17-665"><a href="#cb17-665" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-668"><a href="#cb17-668" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-669"><a href="#cb17-669" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="at">data =</span> bambara, </span>
<span id="cb17-670"><a href="#cb17-670" aria-hidden="true" tabindex="-1"></a>          expression <span class="sc">~</span> temperature <span class="sc">+</span> probe_id)</span>
<span id="cb17-671"><a href="#cb17-671" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-672"><a href="#cb17-672" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-673"><a href="#cb17-673" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> Examine the model with:</span>
<span id="cb17-674"><a href="#cb17-674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-677"><a href="#cb17-677" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-678"><a href="#cb17-678" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(mod)</span>
<span id="cb17-679"><a href="#cb17-679" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-680"><a href="#cb17-680" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-681"><a href="#cb17-681" aria-hidden="true" tabindex="-1"></a>The "(Intercept) Estimate" will be the average of the higher temperature</span>
<span id="cb17-682"><a href="#cb17-682" aria-hidden="true" tabindex="-1"></a>expression because <span class="in">`h`</span> come before <span class="in">`l`</span> in the alphabet. The</span>
<span id="cb17-683"><a href="#cb17-683" aria-hidden="true" tabindex="-1"></a>"temperaturelowert" is how much the expression changes on average going</span>
<span id="cb17-684"><a href="#cb17-684" aria-hidden="true" tabindex="-1"></a>from higher to lower temperatures. You can ignore the rest of the</span>
<span id="cb17-685"><a href="#cb17-685" aria-hidden="true" tabindex="-1"></a>output.</span>
<span id="cb17-686"><a href="#cb17-686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-687"><a href="#cb17-687" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/do_in_R.png)</span> If you prefer, you can use the <span class="in">`anova()`</span></span>
<span id="cb17-688"><a href="#cb17-688" aria-hidden="true" tabindex="-1"></a>function on the model object:</span>
<span id="cb17-689"><a href="#cb17-689" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-692"><a href="#cb17-692" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-693"><a href="#cb17-693" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(mod)</span>
<span id="cb17-694"><a href="#cb17-694" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-695"><a href="#cb17-695" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-696"><a href="#cb17-696" aria-hidden="true" tabindex="-1"></a>This presents you with less information. The *p*-value for the effect of</span>
<span id="cb17-697"><a href="#cb17-697" aria-hidden="true" tabindex="-1"></a>temperate on gene expression will always be the same as in the larger</span>
<span id="cb17-698"><a href="#cb17-698" aria-hidden="true" tabindex="-1"></a>table.</span>
<span id="cb17-699"><a href="#cb17-699" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-700"><a href="#cb17-700" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/answer.png)</span> State your conclusion from the test in a form</span>
<span id="cb17-701"><a href="#cb17-701" aria-hidden="true" tabindex="-1"></a>suitable for including in a report. Make sure you give the direction of</span>
<span id="cb17-702"><a href="#cb17-702" aria-hidden="true" tabindex="-1"></a>any significant effect.</span>
<span id="cb17-703"><a href="#cb17-703" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-704"><a href="#cb17-704" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- #---THINKING ANSWER--- --&gt;</span></span>
<span id="cb17-705"><a href="#cb17-705" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-706"><a href="#cb17-706" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- The expression of individual genes is significantly  --&gt;</span></span>
<span id="cb17-707"><a href="#cb17-707" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-708"><a href="#cb17-708" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- higher at higher temperatures than lower temperatures  --&gt;</span></span>
<span id="cb17-709"><a href="#cb17-709" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-710"><a href="#cb17-710" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- (t = 3.20, d.f. = 34, p = 0.0029) with an average difference --&gt;</span></span>
<span id="cb17-711"><a href="#cb17-711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-712"><a href="#cb17-712" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- of 0.307 units. --&gt;</span></span>
<span id="cb17-713"><a href="#cb17-713" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-714"><a href="#cb17-714" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- You could instead quote (F = 10.3; d.f. = 1, 34, ; p = 0.0029) --&gt;</span></span>
<span id="cb17-715"><a href="#cb17-715" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-716"><a href="#cb17-716" aria-hidden="true" tabindex="-1"></a><span class="fu">## Look after future you!</span></span>
<span id="cb17-717"><a href="#cb17-717" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-718"><a href="#cb17-718" aria-hidden="true" tabindex="-1"></a>The code required to summarise, test, and plot data for a</span>
<span id="cb17-719"><a href="#cb17-719" aria-hidden="true" tabindex="-1"></a>two-independent-samples test AND for a one-way ANOVA is **exactly the</span>
<span id="cb17-720"><a href="#cb17-720" aria-hidden="true" tabindex="-1"></a>same** except for the names of the dataframe, variables and the axis</span>
<span id="cb17-721"><a href="#cb17-721" aria-hidden="true" tabindex="-1"></a>labels and limits. Take some time to comment it your code so that you</span>
<span id="cb17-722"><a href="#cb17-722" aria-hidden="true" tabindex="-1"></a>can make use of it next week.</span>
<span id="cb17-723"><a href="#cb17-723" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-724"><a href="#cb17-724" aria-hidden="true" tabindex="-1"></a><span class="al">![](images/future_you.png)</span>{fig-alt="xxx" width="300"}</span>
<span id="cb17-725"><a href="#cb17-725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-726"><a href="#cb17-726" aria-hidden="true" tabindex="-1"></a>You're finished!</span>
<span id="cb17-727"><a href="#cb17-727" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-728"><a href="#cb17-728" aria-hidden="true" tabindex="-1"></a><span class="fu"># 🥳 Well Done! 🎉</span></span>
<span id="cb17-729"><a href="#cb17-729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-730"><a href="#cb17-730" aria-hidden="true" tabindex="-1"></a><span class="fu"># Independent study following the workshop</span></span>
<span id="cb17-731"><a href="#cb17-731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-732"><a href="#cb17-732" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">Consolidate</span><span class="co">](study_after_workshop.qmd)</span></span>
<span id="cb17-733"><a href="#cb17-733" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-734"><a href="#cb17-734" aria-hidden="true" tabindex="-1"></a><span class="fu"># The Code file</span></span>
<span id="cb17-735"><a href="#cb17-735" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-736"><a href="#cb17-736" aria-hidden="true" tabindex="-1"></a>This contains all the code needed in the workshop even where it is not</span>
<span id="cb17-737"><a href="#cb17-737" aria-hidden="true" tabindex="-1"></a>visible on the webpage.</span>
<span id="cb17-738"><a href="#cb17-738" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-739"><a href="#cb17-739" aria-hidden="true" tabindex="-1"></a>The <span class="in">`workshop.qmd`</span> file is the file I use to compile the practical. Qmd</span>
<span id="cb17-740"><a href="#cb17-740" aria-hidden="true" tabindex="-1"></a>stands for Quarto markdown. It allows code and ordinary text to be</span>
<span id="cb17-741"><a href="#cb17-741" aria-hidden="true" tabindex="-1"></a>interweaved to produce well-formatted reports including webpages. View</span>
<span id="cb17-742"><a href="#cb17-742" aria-hidden="true" tabindex="-1"></a>the source code for this workshop using the <span class="in">`&lt;/&gt; Code`</span> button at the top</span>
<span id="cb17-743"><a href="#cb17-743" aria-hidden="true" tabindex="-1"></a>of the page. Coding and thinking answers are marked with</span>
<span id="cb17-744"><a href="#cb17-744" aria-hidden="true" tabindex="-1"></a><span class="in">`#---CODING ANSWER---`</span> and <span class="in">`#---THINKING ANSWER---`</span></span>
<span id="cb17-745"><a href="#cb17-745" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-746"><a href="#cb17-746" aria-hidden="true" tabindex="-1"></a>Pages made with R <span class="co">[</span><span class="ot">@R-core</span><span class="co">]</span>, Quarto <span class="co">[</span><span class="ot">@allaire2022</span><span class="co">]</span>, <span class="in">`knitr`</span> [@knitr1;</span>
<span id="cb17-747"><a href="#cb17-747" aria-hidden="true" tabindex="-1"></a>@knitr2; @knitr3], <span class="in">`kableExtra`</span> <span class="co">[</span><span class="ot">@kableExtra</span><span class="co">]</span></span>
<span id="cb17-748"><a href="#cb17-748" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-749"><a href="#cb17-749" aria-hidden="true" tabindex="-1"></a><span class="fu"># References</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>Rand, E. (2025). Data Analysis in R for Becoming a Bioscientist (Version 0.2) <a href="https://github.com/3mmaRand/R4BABS" class="uri">https://github.com/3mmaRand/R4BABS</a></p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/3mmaRand/comp4biosci/edit/main/r4babs2/week-3/workshop.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li></ul></div></div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
<li class="nav-item">
    <a class="nav-link" href="https://github.com/3mmaRand">
<p>Emma on GitHub</p>
</a>
  </li>  
</ul>
</div>
  </div>
</footer><script>
document.addEventListener("DOMContentLoaded", function() {
  // grab all links under “Other Links” in the right-hand TOC
  const links = document.querySelectorAll('.quarto-other-links ul li a');
  links.forEach(function(a) {
    // find the one whose text is exactly “Make a suggestion”
    if (a.textContent.trim() === 'Make a suggestion') {
      const pageUrl   = encodeURIComponent(window.location.href);
      const pageTitle = encodeURIComponent(document.title);
      a.href = 'https://github.com/3mmaRand/R4BABS/issues/new'
             + '?title=' + pageTitle
             + '&body='  + pageUrl;
      a.target = '_blank';
    }
  });
});
</script><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>


</body></html>